diff -ruN llvm-project-16.0.0.src/clang/lib/Driver/Driver.cpp llvm-project-16.0.0.src-patch/clang/lib/Driver/Driver.cpp
--- llvm-project-16.0.0.src/clang/lib/Driver/Driver.cpp	2023-03-17 21:26:07.000000000 +0900
+++ llvm-project-16.0.0.src-patch/clang/lib/Driver/Driver.cpp	2023-03-18 21:21:07.478638487 +0900
@@ -184,6 +184,11 @@
     P = llvm::sys::path::parent_path(Dir);
     llvm::sys::path::append(P, CLANG_INSTALL_LIBDIR_BASENAME, "clang",
                             CLANG_VERSION_MAJOR_STRING);
+    if (!llvm::sys::fs::exists(P)) {
+        P = Dir;
+        llvm::sys::path::append(P, CLANG_INSTALL_LIBDIR_BASENAME, "clang",
+                                CLANG_VERSION_MAJOR_STRING);
+    }
   }
 
   return std::string(P.str());
