diff -ruN llvm-project-22.1.0-rc2.src/clang/lib/Options/OptionUtils.cpp llvm-project-22.1.0-rc2.src-patch/clang/lib/Options/OptionUtils.cpp
--- llvm-project-22.1.0-rc2.src/clang/lib/Options/OptionUtils.cpp	2026-01-27 17:33:32.000000000 +0900
+++ llvm-project-22.1.0-rc2.src-patch/clang/lib/Options/OptionUtils.cpp	2026-01-30 21:46:58.316490564 +0900
@@ -234,6 +234,11 @@
     // changes here also needs to happen in lld/COFF/Driver.cpp
     llvm::sys::path::append(P, CLANG_INSTALL_LIBDIR_BASENAME, "clang",
                             CLANG_VERSION_MAJOR_STRING);
+    if (!llvm::sys::fs::exists(P)) {
+        P = Dir;
+        llvm::sys::path::append(P, CLANG_INSTALL_LIBDIR_BASENAME, "clang",
+                                CLANG_VERSION_MAJOR_STRING);
+    }
   }
 
   return std::string(P);
