diff -ruN -a apache-arrow-5.0.0/cpp/cmake_modules/ThirdpartyToolchain.cmake apache-arrow-5.0.0-patch/cpp/cmake_modules/ThirdpartyToolchain.cmake
--- apache-arrow-5.0.0/cpp/cmake_modules/ThirdpartyToolchain.cmake	2021-07-22 21:40:24.000000000 +0900
+++ apache-arrow-5.0.0-patch/cpp/cmake_modules/ThirdpartyToolchain.cmake	2021-08-12 22:51:49.430865500 +0900
@@ -1108,6 +1108,7 @@
   set(GLOG_CMAKE_ARGS
       ${EP_COMMON_CMAKE_ARGS}
       "-DCMAKE_INSTALL_PREFIX=${GLOG_BUILD_DIR}"
+      -DCMAKE_INSTALL_LIBDIR=lib
       -DBUILD_SHARED_LIBS=OFF
       -DBUILD_TESTING=OFF
       -DWITH_GFLAGS=OFF
diff -ruN -a apache-arrow-5.0.0/cpp/src/arrow/compute/kernel.h apache-arrow-5.0.0-patch/cpp/src/arrow/compute/kernel.h
--- apache-arrow-5.0.0/cpp/src/arrow/compute/kernel.h	2021-07-22 21:40:24.000000000 +0900
+++ apache-arrow-5.0.0-patch/cpp/src/arrow/compute/kernel.h	2021-08-12 22:51:49.446989100 +0900
@@ -608,7 +608,7 @@
         finalize(std::move(finalize)) {}
 
   VectorKernel(std::shared_ptr<KernelSignature> sig, ArrayKernelExec exec,
-               KernelInit init = NULLPTR, VectorFinalize finalize = NULLPTR)
+               KernelInit init, VectorFinalize finalize = NULLPTR)
       : ArrayKernel(std::move(sig), std::move(exec), std::move(init)),
         finalize(std::move(finalize)) {}
 
diff -ruN -a apache-arrow-5.0.0/cpp/src/arrow/flight/client_header_internal.cc apache-arrow-5.0.0-patch/cpp/src/arrow/flight/client_header_internal.cc
--- apache-arrow-5.0.0/cpp/src/arrow/flight/client_header_internal.cc	2021-07-22 21:40:24.000000000 +0900
+++ apache-arrow-5.0.0-patch/cpp/src/arrow/flight/client_header_internal.cc	2021-08-12 22:50:56.061107300 +0900
@@ -18,10 +18,10 @@
 // Interfaces for defining middleware for Flight clients. Currently
 // experimental.
 
+#include "arrow/flight/platform.h"
 #include "arrow/flight/client_header_internal.h"
 #include "arrow/flight/client.h"
 #include "arrow/flight/client_auth.h"
-#include "arrow/flight/platform.h"
 #include "arrow/util/base64.h"
 #include "arrow/util/make_unique.h"
 #include "arrow/util/string.h"
diff -ruN -a apache-arrow-5.0.0/cpp/src/arrow/vendored/datetime/date.h apache-arrow-5.0.0-patch/cpp/src/arrow/vendored/datetime/date.h
--- apache-arrow-5.0.0/cpp/src/arrow/vendored/datetime/date.h	2021-07-22 21:40:24.000000000 +0900
+++ apache-arrow-5.0.0-patch/cpp/src/arrow/vendored/datetime/date.h	2021-08-12 22:51:49.446989100 +0900
@@ -109,7 +109,7 @@
 #    define CONSTDATA const
 #    define CONSTCD11
 #    define CONSTCD14
-#    define NOEXCEPT _NOEXCEPT
+#    define NOEXCEPT
 #  else
 //   VS2017 and later
 #    define CONSTDATA constexpr const
