diff -ruN tesseract-3.05.01/api/baseapi.cpp tesseract-3.05.01-android/api/baseapi.cpp
--- tesseract-3.05.01/api/baseapi.cpp	2017-06-02 03:06:24.000000000 +0900
+++ tesseract-3.05.01-android/api/baseapi.cpp	2018-02-25 01:38:07.026294918 +0900
@@ -42,6 +42,10 @@
 #include <string.h>
 #endif  // _WIN32
 
+#include <stdio.h>
+int fgetpos(FILE* __fp, fpos_t* __pos);
+int fsetpos(FILE* __fp, const fpos_t* __pos);
+
 #include <iostream>
 #include <string>
 #include <iterator>
diff -ruN tesseract-3.05.01/api/Makefile.am tesseract-3.05.01-android/api/Makefile.am
--- tesseract-3.05.01/api/Makefile.am	2017-06-02 03:06:24.000000000 +0900
+++ tesseract-3.05.01-android/api/Makefile.am	2018-02-25 01:36:15.815499281 +0900
@@ -20,7 +20,7 @@
 noinst_LTLIBRARIES = libtesseract_api.la
 else
 lib_LTLIBRARIES += libtesseract_api.la
-libtesseract_api_la_LDFLAGS = -version-info $(GENERIC_LIBRARY_VERSION)
+libtesseract_api_la_LDFLAGS = -avoid-version
 libtesseract_api_la_LIBADD = \
     ../ccmain/libtesseract_main.la \
     ../textord/libtesseract_textord.la \
@@ -67,7 +67,7 @@
         ../neural_networks/runtime/libtesseract_neural.la
 endif
 
-libtesseract_la_LDFLAGS += -version-info $(GENERIC_LIBRARY_VERSION) -no-undefined
+libtesseract_la_LDFLAGS += -avoid-version -no-undefined
 
 bin_PROGRAMS = tesseract
 tesseract_SOURCES = tesseractmain.cpp
