diff -ruN tesseract-3.05.00/ccstruct/imagedata.cpp tesseract-3.05.00-windows/ccstruct/imagedata.cpp
--- tesseract-3.05.00/ccstruct/imagedata.cpp	2017-02-17 02:59:48.000000000 +0900
+++ tesseract-3.05.00-windows/ccstruct/imagedata.cpp	2017-04-11 22:44:49.603028200 +0900
@@ -453,13 +453,7 @@
     if (needs_loading) LoadPageInBackground(index);
     // We can't directly load the page, or the background load will delete it
     // while the caller is using it, so give it a chance to work.
-#if __cplusplus > 199711L
-    std::this_thread::sleep_for(std::chrono::seconds(1));
-#elif _WIN32  // MSVS
     Sleep(1000);
-#else
-    sleep(1);
-#endif
   }
   return page;
 }
