diff -ruN tesseract-4.1.0-rc2/configure tesseract-4.1.0-rc2-android/configure
--- tesseract-4.1.0-rc2/configure	2019-06-09 20:29:42.873630849 +0900
+++ tesseract-4.1.0-rc2-android/configure	2019-06-09 20:31:53.716960166 +0900
@@ -4219,16 +4219,8 @@
   avx=false
 fi
 
-if $avx; then
-     if true; then
-  AVX_OPT_TRUE=
-  AVX_OPT_FALSE='#'
-else
   AVX_OPT_TRUE='#'
   AVX_OPT_FALSE=
-fi
-
-fi
 
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether C++ compiler accepts -mavx2" >&5
 $as_echo_n "checking whether C++ compiler accepts -mavx2... " >&6; }
@@ -4265,16 +4257,8 @@
   avx2=false
 fi
 
-if $avx2; then
-     if true; then
-  AVX2_OPT_TRUE=
-  AVX2_OPT_FALSE='#'
-else
   AVX2_OPT_TRUE='#'
   AVX2_OPT_FALSE=
-fi
-
-fi
 
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether C++ compiler accepts -msse4.1" >&5
 $as_echo_n "checking whether C++ compiler accepts -msse4.1... " >&6; }
@@ -4311,16 +4295,8 @@
   sse41=false
 fi
 
-if $sse41; then
-     if true; then
-  SSE41_OPT_TRUE=
-  SSE41_OPT_FALSE='#'
-else
   SSE41_OPT_TRUE='#'
   SSE41_OPT_FALSE=
-fi
-
-fi
 
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether C++ compiler accepts -march=native" >&5
 $as_echo_n "checking whether C++ compiler accepts -march=native... " >&6; }
diff -ruN tesseract-4.1.0-rc2/src/api/baseapi.cpp tesseract-4.1.0-rc2-android/src/api/baseapi.cpp
--- tesseract-4.1.0-rc2/src/api/baseapi.cpp	2019-06-09 20:23:25.221794239 +0900
+++ tesseract-4.1.0-rc2-android/src/api/baseapi.cpp	2019-06-09 20:21:28.167558385 +0900
@@ -41,6 +41,10 @@
 #include <unistd.h>
 #endif  // _WIN32
 
+#include <stdio.h>
+int fgetpos(FILE* __fp, fpos_t* __pos);
+int fsetpos(FILE* __fp, const fpos_t* __pos);
+
 #include <clocale>             // for LC_ALL, LC_CTYPE, LC_NUMERIC
 #include <cmath>               // for round, M_PI
 #include <cstdint>             // for int32_t
diff -ruN tesseract-4.1.0-rc2/src/api/Makefile.am tesseract-4.1.0-rc2-android/src/api/Makefile.am
--- tesseract-4.1.0-rc2/src/api/Makefile.am	2019-06-09 20:23:25.221794239 +0900
+++ tesseract-4.1.0-rc2-android/src/api/Makefile.am	2019-06-09 20:21:28.168558387 +0900
@@ -72,7 +72,7 @@
 libtesseract_la_LIBADD += ../arch/libtesseract_sse.la
 endif
 
-libtesseract_la_LDFLAGS += -version-info $(GENERIC_LIBRARY_VERSION) $(NOUNDEFINED)
+libtesseract_la_LDFLAGS += -avoid-version $(NOUNDEFINED)
 
 bin_PROGRAMS = tesseract
 tesseract_SOURCES = tesseractmain.cpp
@@ -95,5 +95,5 @@
 libtesseract_la_LDFLAGS += -no-undefined -Wl,--as-needed -lws2_32
 endif
 if ADD_RT
-tesseract_LDADD += -lrt
+tesseract_LDADD += 
 endif
diff -ruN tesseract-4.1.0-rc2/src/api/Makefile.in tesseract-4.1.0-rc2-android/src/api/Makefile.in
--- tesseract-4.1.0-rc2/src/api/Makefile.in	2019-06-09 20:23:25.221794239 +0900
+++ tesseract-4.1.0-rc2-android/src/api/Makefile.in	2019-06-09 20:22:25.734674378 +0900
@@ -90,7 +90,7 @@
 @VISIBILITY_TRUE@am__append_7 = -DTESS_IMPORTS
 @T_WIN_TRUE@am__append_8 = -lws2_32
 @T_WIN_TRUE@am__append_9 = -no-undefined -Wl,--as-needed -lws2_32
-@ADD_RT_TRUE@am__append_10 = -lrt
+@ADD_RT_TRUE@am__append_10 =
 subdir = src/api
 DIST_COMMON = $(srcdir)/Makefile.in $(srcdir)/Makefile.am \
 	$(srcdir)/tess_version.h.in $(top_srcdir)/config/depcomp \
@@ -411,7 +411,7 @@
 	hocrrenderer.cpp lstmboxrenderer.cpp pdfrenderer.cpp \
 	wordstrboxrenderer.cpp renderer.cpp
 libtesseract_la_LDFLAGS = $(LEPTONICA_LIBS) $(OPENCL_LDFLAGS) \
-	$(libarchive_LIBS) -version-info $(GENERIC_LIBRARY_VERSION) \
+	$(libarchive_LIBS) -avoid-version \
 	$(NOUNDEFINED) $(am__append_9)
 libtesseract_la_SOURCES = 
 # Dummy C++ source to cause C++ linking.
diff -ruN tesseract-4.1.0-rc2/src/ccutil/fileio.cpp tesseract-4.1.0-rc2-android/src/ccutil/fileio.cpp
--- tesseract-4.1.0-rc2/src/ccutil/fileio.cpp	2019-05-01 17:58:56.000000000 +0900
+++ tesseract-4.1.0-rc2-android/src/ccutil/fileio.cpp	2019-06-09 20:21:28.168558387 +0900
@@ -20,7 +20,6 @@
 #include <io.h>
 #endif
 #else
-#include <glob.h>
 #include <unistd.h>
 #endif
 
@@ -115,16 +114,7 @@
 }
 #else
 bool File::DeleteMatchingFiles(const char* pattern) {
-  glob_t pglob;
-  char **paths;
-  bool all_deleted = true;
-  if (glob(pattern, 0, nullptr, &pglob) == 0) {
-    for (paths = pglob.gl_pathv; *paths != nullptr; paths++) {
-      all_deleted &= File::Delete(*paths);
-    }
-    globfree(&pglob);
-  }
-  return all_deleted;
+    return false;
 }
 #endif
 
diff -ruN tesseract-4.1.0-rc2/src/lstm/lstm.cpp tesseract-4.1.0-rc2-android/src/lstm/lstm.cpp
--- tesseract-4.1.0-rc2/src/lstm/lstm.cpp	2019-05-01 17:58:56.000000000 +0900
+++ tesseract-4.1.0-rc2-android/src/lstm/lstm.cpp	2019-06-09 20:21:28.169558389 +0900
@@ -33,6 +33,10 @@
 #include "networkscratch.h"
 #include "tprintf.h"
 
+#if (defined(_MSC_VER) && _MSC_VER < 1900) || defined(ANDROID)
+extern double log2(double n);
+#endif  // _MSC_VER
+
 // Macros for openmp code if it is available, otherwise empty macros.
 #ifdef _OPENMP
 #define PARALLEL_IF_OPENMP(__num_threads)                                  \
diff -ruN tesseract-4.1.0-rc2/src/lstm/weightmatrix.cpp tesseract-4.1.0-rc2-android/src/lstm/weightmatrix.cpp
--- tesseract-4.1.0-rc2/src/lstm/weightmatrix.cpp	2019-05-01 17:58:56.000000000 +0900
+++ tesseract-4.1.0-rc2-android/src/lstm/weightmatrix.cpp	2019-06-09 20:21:28.169558389 +0900
@@ -23,6 +23,10 @@
 #include "statistc.h"
 #include "tprintf.h"
 
+#if (defined(_MSC_VER) && _MSC_VER < 1900) || defined(ANDROID)
+extern double log2(double n);
+#endif  // _MSC_VER
+
 namespace tesseract {
 
 #if defined(ANDROID)
