diff -ruN tesseract-3.05.00/ccmain/thresholder.cpp tesseract-3.05.00-linux/ccmain/thresholder.cpp
--- tesseract-3.05.00/ccmain/thresholder.cpp	2017-02-17 02:59:48.000000000 +0900
+++ tesseract-3.05.00-linux/ccmain/thresholder.cpp	2017-04-12 07:51:30.014191009 +0900
@@ -184,7 +184,7 @@
     // We have a binary image, but it still has to be copied, as this API
     // allows the caller to modify the output.
     Pix* original = GetPixRect();
-    *pix = pixCopy(nullptr, original);
+    *pix = pixCopy(NULL, original);
     pixDestroy(&original);
   } else {
     OtsuThresholdRectToPix(pix_, pix);
diff -ruN tesseract-3.05.00/training/stringrenderer.cpp tesseract-3.05.00-linux/training/stringrenderer.cpp
--- tesseract-3.05.00/training/stringrenderer.cpp	2017-02-17 02:59:48.000000000 +0900
+++ tesseract-3.05.00-linux/training/stringrenderer.cpp	2017-04-12 07:51:41.142435264 +0900
@@ -244,7 +244,7 @@
   int offset = 0;
   TRand rand;
   bool started_underline = false;
-  PangoAttribute* und_attr = nullptr;
+  PangoAttribute* und_attr = NULL;
 
   while (offset < page_text.length()) {
     offset += SpanUTF8Whitespace(text + offset);
@@ -263,7 +263,7 @@
         // previous word.
         pango_attr_list_insert(attr_list, und_attr);
         started_underline = false;
-        und_attr = nullptr;
+        und_attr = NULL;
       }
     }
     if (!started_underline && RandBool(underline_start_prob_, &rand)) {
