diff -ruN tesseract-4.0.0-beta.3/src/api/baseapi.cpp tesseract-4.0.0-beta.3-android/src/api/baseapi.cpp
--- tesseract-4.0.0-beta.3/src/api/baseapi.cpp	2018-06-20 19:21:25.000000000 +0900
+++ tesseract-4.0.0-beta.3-android/src/api/baseapi.cpp	2018-06-25 12:24:31.630344792 +0900
@@ -42,6 +42,10 @@
 #include <unistd.h>
 #endif  // _WIN32
 
+#include <stdio.h>
+int fgetpos(FILE* __fp, fpos_t* __pos);
+int fsetpos(FILE* __fp, const fpos_t* __pos);
+
 #include <algorithm>
 #include <clocale>
 #include <fstream>
diff -ruN tesseract-4.0.0-beta.3/src/api/Makefile.am tesseract-4.0.0-beta.3-android/src/api/Makefile.am
--- tesseract-4.0.0-beta.3/src/api/Makefile.am	2018-06-20 19:21:25.000000000 +0900
+++ tesseract-4.0.0-beta.3-android/src/api/Makefile.am	2018-06-25 12:23:35.824330876 +0900
@@ -54,7 +54,7 @@
     ../ccutil/libtesseract_ccutil.la \
     ../opencl/libtesseract_opencl.la 
 
-libtesseract_la_LDFLAGS += -version-info $(GENERIC_LIBRARY_VERSION) -no-undefined
+libtesseract_la_LDFLAGS += -avoid-version -no-undefined
 
 bin_PROGRAMS = tesseract
 tesseract_SOURCES = tesseractmain.cpp
diff -ruN tesseract-4.0.0-beta.3/src/lstm/lstm.cpp tesseract-4.0.0-beta.3-android/src/lstm/lstm.cpp
--- tesseract-4.0.0-beta.3/src/lstm/lstm.cpp	2018-06-20 19:21:25.000000000 +0900
+++ tesseract-4.0.0-beta.3-android/src/lstm/lstm.cpp	2018-06-25 12:23:35.825330876 +0900
@@ -33,6 +33,10 @@
 #include "networkscratch.h"
 #include "tprintf.h"
 
+#if (defined(_MSC_VER) && _MSC_VER < 1900) || defined(ANDROID)
+extern double log2(double n);
+#endif  // _MSC_VER
+
 // Macros for openmp code if it is available, otherwise empty macros.
 #ifdef _OPENMP
 #define PARALLEL_IF_OPENMP(__num_threads)                                  \
diff -ruN tesseract-4.0.0-beta.3/src/lstm/weightmatrix.cpp tesseract-4.0.0-beta.3-android/src/lstm/weightmatrix.cpp
--- tesseract-4.0.0-beta.3/src/lstm/weightmatrix.cpp	2018-06-20 19:21:25.000000000 +0900
+++ tesseract-4.0.0-beta.3-android/src/lstm/weightmatrix.cpp	2018-06-25 12:23:35.826330877 +0900
@@ -25,6 +25,10 @@
 #include "statistc.h"
 #include "tprintf.h"
 
+#if (defined(_MSC_VER) && _MSC_VER < 1900) || defined(ANDROID)
+extern double log2(double n);
+#endif  // _MSC_VER
+
 namespace tesseract {
 
 #if (defined(_MSC_VER) && _MSC_VER < 1900) || defined(ANDROID)
