diff -ruN tesseract-4.0.0/src/api/baseapi.cpp tesseract-4.0.0-android/src/api/baseapi.cpp
--- tesseract-4.0.0/src/api/baseapi.cpp	2018-10-29 17:53:12.000000000 +0900
+++ tesseract-4.0.0-android/src/api/baseapi.cpp	2018-11-05 13:45:44.925599392 +0900
@@ -41,6 +41,10 @@
 #include <unistd.h>
 #endif  // _WIN32
 
+#include <stdio.h>
+int fgetpos(FILE* __fp, fpos_t* __pos);
+int fsetpos(FILE* __fp, const fpos_t* __pos);
+
 #include <clocale>             // for LC_ALL, LC_CTYPE, LC_NUMERIC
 #include <cmath>               // for round, M_PI
 #include <cstdint>             // for int32_t
diff -ruN tesseract-4.0.0/src/api/Makefile.am tesseract-4.0.0-android/src/api/Makefile.am
--- tesseract-4.0.0/src/api/Makefile.am	2018-10-29 17:53:12.000000000 +0900
+++ tesseract-4.0.0-android/src/api/Makefile.am	2018-11-05 13:47:48.275392451 +0900
@@ -58,7 +58,7 @@
     ../ccutil/libtesseract_ccutil.la \
     ../opencl/libtesseract_opencl.la
 
-libtesseract_la_LDFLAGS += -version-info $(GENERIC_LIBRARY_VERSION) $(NOUNDEFINED)
+libtesseract_la_LDFLAGS += -avoid-version $(NOUNDEFINED)
 
 bin_PROGRAMS = tesseract
 tesseract_SOURCES = tesseractmain.cpp
diff -ruN tesseract-4.0.0/src/ccutil/fileio.cpp tesseract-4.0.0-android/src/ccutil/fileio.cpp
--- tesseract-4.0.0/src/ccutil/fileio.cpp	2018-10-29 17:53:12.000000000 +0900
+++ tesseract-4.0.0-android/src/ccutil/fileio.cpp	2018-11-05 13:45:44.925599392 +0900
@@ -19,7 +19,6 @@
 #include <io.h>
 #endif
 #else
-#include <glob.h>
 #include <unistd.h>
 #endif
 
@@ -110,16 +109,7 @@
 }
 #else
 bool File::DeleteMatchingFiles(const char* pattern) {
-  glob_t pglob;
-  char **paths;
-  bool all_deleted = true;
-  if (glob(pattern, 0, nullptr, &pglob) == 0) {
-    for (paths = pglob.gl_pathv; *paths != nullptr; paths++) {
-      all_deleted &= File::Delete(*paths);
-    }
-    globfree(&pglob);
-  }
-  return all_deleted;
+    return false;
 }
 #endif
 
diff -ruN tesseract-4.0.0/src/lstm/lstm.cpp tesseract-4.0.0-android/src/lstm/lstm.cpp
--- tesseract-4.0.0/src/lstm/lstm.cpp	2018-10-29 17:53:12.000000000 +0900
+++ tesseract-4.0.0-android/src/lstm/lstm.cpp	2018-11-05 13:45:44.926599391 +0900
@@ -33,6 +33,10 @@
 #include "networkscratch.h"
 #include "tprintf.h"
 
+#if (defined(_MSC_VER) && _MSC_VER < 1900) || defined(ANDROID)
+extern double log2(double n);
+#endif  // _MSC_VER
+
 // Macros for openmp code if it is available, otherwise empty macros.
 #ifdef _OPENMP
 #define PARALLEL_IF_OPENMP(__num_threads)                                  \
diff -ruN tesseract-4.0.0/src/lstm/weightmatrix.cpp tesseract-4.0.0-android/src/lstm/weightmatrix.cpp
--- tesseract-4.0.0/src/lstm/weightmatrix.cpp	2018-10-29 17:53:12.000000000 +0900
+++ tesseract-4.0.0-android/src/lstm/weightmatrix.cpp	2018-11-05 13:45:44.926599391 +0900
@@ -25,6 +25,10 @@
 #include "statistc.h"
 #include "tprintf.h"
 
+#if (defined(_MSC_VER) && _MSC_VER < 1900) || defined(ANDROID)
+extern double log2(double n);
+#endif  // _MSC_VER
+
 namespace tesseract {
 
 #if defined(ANDROID)
