diff -ruN libfreenect-0.5.2/fakenect/CMakeLists.txt libfreenect-0.5.2-macosx/fakenect/CMakeLists.txt
--- libfreenect-0.5.2/fakenect/CMakeLists.txt	2015-01-27 12:01:37.000000000 +0900
+++ libfreenect-0.5.2-macosx/fakenect/CMakeLists.txt	2015-04-14 21:26:08.896900306 +0900
@@ -4,6 +4,7 @@
 SET(EXECUTABLE_OUTPUT_PATH ${CMAKE_BINARY_DIR}/bin)
 SET(LIBRARY_OUTPUT_PATH ${CMAKE_BINARY_DIR}/lib/fakenect)
 add_library (fakenect SHARED fakenect.c)
+set_target_properties(fakenect PROPERTIES INSTALL_NAME_DIR "@rpath")
 set_target_properties ( fakenect PROPERTIES
   VERSION ${PROJECT_VER}
   SOVERSION ${PROJECT_APIVER}
diff -ruN libfreenect-0.5.2/OpenNI2-FreenectDriver/CMakeLists.txt libfreenect-0.5.2-macosx/OpenNI2-FreenectDriver/CMakeLists.txt
--- libfreenect-0.5.2/OpenNI2-FreenectDriver/CMakeLists.txt	2015-01-27 12:01:37.000000000 +0900
+++ libfreenect-0.5.2-macosx/OpenNI2-FreenectDriver/CMakeLists.txt	2015-04-14 21:24:42.548226107 +0900
@@ -5,6 +5,7 @@
 file(GLOB HEADERS src/*.hpp src/*.h)
 file(GLOB SOURCES src/*.cpp)
 add_library(FreenectDriver SHARED ${HEADERS} ${SOURCES})
+set_target_properties(FreenectDriver PROPERTIES INSTALL_NAME_DIR "@rpath")
 
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-unused-function")
 
diff -ruN libfreenect-0.5.2/src/CMakeLists.txt libfreenect-0.5.2-macosx/src/CMakeLists.txt
--- libfreenect-0.5.2/src/CMakeLists.txt	2015-01-27 12:01:37.000000000 +0900
+++ libfreenect-0.5.2-macosx/src/CMakeLists.txt	2015-04-14 21:24:04.039479468 +0900
@@ -22,6 +22,7 @@
 LIST(APPEND SRC core.c tilt.c cameras.c flags.c usb_libusb10.c registration.c audio.c loader.c)
 
 add_library (freenect SHARED ${SRC})
+set_target_properties(freenect PROPERTIES INSTALL_NAME_DIR "@rpath")
 set_target_properties ( freenect PROPERTIES
   VERSION ${PROJECT_VER}
   SOVERSION ${PROJECT_APIVER})	
diff -ruN libfreenect-0.5.2/wrappers/c_sync/CMakeLists.txt libfreenect-0.5.2-macosx/wrappers/c_sync/CMakeLists.txt
--- libfreenect-0.5.2/wrappers/c_sync/CMakeLists.txt	2015-01-27 12:01:37.000000000 +0900
+++ libfreenect-0.5.2-macosx/wrappers/c_sync/CMakeLists.txt	2015-04-14 21:23:32.973877142 +0900
@@ -7,6 +7,7 @@
 include_directories(${THREADS_PTHREADS_INCLUDE_DIR})
 
 add_library (freenect_sync SHARED libfreenect_sync.c)
+set_target_properties(freenect_sync PROPERTIES INSTALL_NAME_DIR "@rpath")
 add_library (freenect_sync_static STATIC libfreenect_sync.c)
 set_target_properties (freenect_sync_static PROPERTIES OUTPUT_NAME freenect_sync)
 
diff -ruN libfreenect-0.5.2/wrappers/opencv/CMakeLists.txt libfreenect-0.5.2-macosx/wrappers/opencv/CMakeLists.txt
--- libfreenect-0.5.2/wrappers/opencv/CMakeLists.txt	2015-01-27 12:01:37.000000000 +0900
+++ libfreenect-0.5.2-macosx/wrappers/opencv/CMakeLists.txt	2015-04-14 21:25:36.721276459 +0900
@@ -3,6 +3,7 @@
 ######################################################################################
 find_package(OpenCV REQUIRED)
 add_library (freenect_cv SHARED libfreenect_cv.c)
+set_target_properties(freenect_cv PROPERTIES INSTALL_NAME_DIR "@rpath")
 set_target_properties (freenect_cv PROPERTIES
   VERSION ${PROJECT_VER}
   SOVERSION ${PROJECT_APIVER})
