diff -ruN libfreenect-0.4.0/CMakeLists.txt libfreenect-0.4.0-windows/CMakeLists.txt
--- libfreenect-0.4.0/CMakeLists.txt	2014-02-16 00:49:20.000000000 +0900
+++ libfreenect-0.4.0-windows/CMakeLists.txt	2014-03-21 23:21:37.216455090 +0900
@@ -118,6 +118,7 @@
 
 if(WIN32)
   include_directories("${CMAKE_CURRENT_SOURCE_DIR}/platform/windows")
+  include_directories("${CMAKE_CURRENT_SOURCE_DIR}/platform/windows/libusb10emu")
   include_directories("${CMAKE_CURRENT_SOURCE_DIR}/platform/windows/libusb10emu/libusb-1.0")
 endif()
 
diff -ruN libfreenect-0.4.0/src/CMakeLists.txt libfreenect-0.4.0-windows/src/CMakeLists.txt
--- libfreenect-0.4.0/src/CMakeLists.txt	2014-02-16 00:49:20.000000000 +0900
+++ libfreenect-0.4.0-windows/src/CMakeLists.txt	2014-03-21 23:20:01.691782389 +0900
@@ -36,16 +36,16 @@
 install (TARGETS freenect
   DESTINATION "${PROJECT_LIBRARY_INSTALL_DIR}")
 
-add_library (freenectstatic STATIC ${SRC})
-set_target_properties (freenectstatic PROPERTIES OUTPUT_NAME freenect)
-IF(UNIX AND NOT APPLE)
-  SET_TARGET_PROPERTIES (freenectstatic PROPERTIES COMPILE_FLAGS "-fPIC")
-ENDIF()
-install (TARGETS freenectstatic
-  DESTINATION "${PROJECT_LIBRARY_INSTALL_DIR}")
+# add_library (freenectstatic STATIC ${SRC})
+# set_target_properties (freenectstatic PROPERTIES OUTPUT_NAME freenect)
+# IF(UNIX AND NOT APPLE)
+#  SET_TARGET_PROPERTIES (freenectstatic PROPERTIES COMPILE_FLAGS "-fPIC")
+#ENDIF()
+#install (TARGETS freenectstatic
+#  DESTINATION "${PROJECT_LIBRARY_INSTALL_DIR}")
 
 target_link_libraries (freenect ${LIBUSB_1_LIBRARIES})
-target_link_libraries (freenectstatic ${LIBUSB_1_LIBRARIES})
+#target_link_libraries (freenectstatic ${LIBUSB_1_LIBRARIES})
 
 # Install the header files
 install (FILES "../include/libfreenect.h" "../include/libfreenect_registration.h"
diff -ruN libfreenect-0.4.0/wrappers/c_sync/CMakeLists.txt libfreenect-0.4.0-windows/wrappers/c_sync/CMakeLists.txt
--- libfreenect-0.4.0/wrappers/c_sync/CMakeLists.txt	2014-02-16 00:49:20.000000000 +0900
+++ libfreenect-0.4.0-windows/wrappers/c_sync/CMakeLists.txt	2014-03-21 23:21:01.320826667 +0900
@@ -9,19 +9,19 @@
 find_package(Threads REQUIRED)
 include_directories(${THREADS_PTHREADS_INCLUDE_DIR})
 
-add_library (freenect_sync SHARED libfreenect_sync.c)
+# add_library (freenect_sync SHARED libfreenect_sync.c)
 add_library (freenect_sync_static STATIC libfreenect_sync.c)
 set_target_properties (freenect_sync_static PROPERTIES OUTPUT_NAME freenect_sync)
 
-set_target_properties (freenect_sync PROPERTIES
-  VERSION ${PROJECT_VER}
-  SOVERSION ${PROJECT_APIVER})
+# set_target_properties (freenect_sync PROPERTIES
+#  VERSION ${PROJECT_VER}
+#  SOVERSION ${PROJECT_APIVER})
 
-target_link_libraries (freenect_sync freenect ${CMAKE_THREAD_LIBS_INIT})
+# target_link_libraries (freenect_sync freenect ${CMAKE_THREAD_LIBS_INIT})
 target_link_libraries (freenect_sync_static freenect ${CMAKE_THREAD_LIBS_INIT})
 
-install (TARGETS freenect_sync
-  DESTINATION "${PROJECT_LIBRARY_INSTALL_DIR}")
+# install (TARGETS freenect_sync
+#   DESTINATION "${PROJECT_LIBRARY_INSTALL_DIR}")
 install (TARGETS freenect_sync_static
   DESTINATION "${PROJECT_LIBRARY_INSTALL_DIR}")
 install (FILES "libfreenect_sync.h"
