diff -ruN flandmark-master/libflandmark/msvc-compat.h flandmark-master-windows/libflandmark/msvc-compat.h
--- flandmark-master/libflandmark/msvc-compat.h	2016-06-08 03:04:49.000000000 +0900
+++ flandmark-master-windows/libflandmark/msvc-compat.h	2017-04-08 22:21:01.187633700 +0900
@@ -1,18 +1 @@
-#ifndef FLANDMAR_MSVC_COMPAT
-#define FLANDMAR_MSVC_COMPAT
-#ifdef _MSC_VER
-
-typedef unsigned char uint8_t;
-typedef char int8_t;
-typedef unsigned __int16 uint16_t;
-typedef __int16 int16_t;
-typedef unsigned __int32 uint32_t;
-typedef __int32 int32_t;
-typedef unsigned __int64 uint64_t;
-typedef __int64 int64_t;
-
-#else
-#	include <stdint.h>
-#endif
-
-#endif
+#include <stdint.h>
