diff -ruN cminpack-1.3.11/cminpack.h cminpack-1.3.11-patch/cminpack.h
--- cminpack-1.3.11/cminpack.h	2024-09-13 18:04:15.000000000 +0900
+++ cminpack-1.3.11-patch/cminpack.h	2024-10-27 09:09:37.257671690 +0900
@@ -36,38 +36,7 @@
 extern "C" {
 #endif /* __cplusplus */
 
-/* Cmake will define cminpack_EXPORTS on Windows when it
-configures to build a shared library. If you are going to use
-another build system on windows or create the visual studio
-projects by hand you need to define cminpack_EXPORTS when
-building a DLL on windows.
-*/
-#if defined (__GNUC__)
-#define CMINPACK_DECLSPEC_EXPORT  __declspec(__dllexport__)
-#define CMINPACK_DECLSPEC_IMPORT  __declspec(__dllimport__)
-#endif
-#if defined (_MSC_VER) || defined (__BORLANDC__)
-#define CMINPACK_DECLSPEC_EXPORT  __declspec(dllexport)
-#define CMINPACK_DECLSPEC_IMPORT  __declspec(dllimport)
-#endif
-#ifdef __WATCOMC__
-#define CMINPACK_DECLSPEC_EXPORT  __export
-#define CMINPACK_DECLSPEC_IMPORT  __import
-#endif
-#ifdef __IBMC__
-#define CMINPACK_DECLSPEC_EXPORT  _Export
-#define CMINPACK_DECLSPEC_IMPORT  _Import
-#endif
-
-#if !defined(CMINPACK_NO_DLL) && defined(_WIN32)
-  #if defined(CMINPACK_DLL_EXPORTS)
-    #define  CMINPACK_EXPORT CMINPACK_DECLSPEC_EXPORT
-  #else
-    #define  CMINPACK_EXPORT CMINPACK_DECLSPEC_IMPORT
-  #endif /* cminpack_EXPORTS */
-#else /* defined (_WIN32) */
  #define CMINPACK_EXPORT
-#endif
 
 #if defined(__CUDA_ARCH__) || defined(__CUDACC__)
 #define __cminpack_attr__ __device__
diff -ruN cminpack-1.3.11/cminpackP.h cminpack-1.3.11-patch/cminpackP.h
--- cminpack-1.3.11/cminpackP.h	2024-09-13 18:04:15.000000000 +0900
+++ cminpack-1.3.11-patch/cminpackP.h	2024-10-27 09:10:43.208509151 +0900
@@ -11,10 +11,10 @@
 #endif
 
 #ifdef USE_BLAS
-#ifdef __APPLE__
-#include <Accelerate/Accelerate.h>
-#define __cminpack_blasint__ int
-#else /* !__APPLE__ */
+//#ifdef __APPLE__
+//#include <Accelerate/Accelerate.h>
+//#define __cminpack_blasint__ int
+//#else /* !__APPLE__ */
 #ifndef __cminpack_blasint__
 #define __cminpack_blasint__ int
 #endif
@@ -34,13 +34,13 @@
   const char *TransA, const char *Diag,
   const __cminpack_blasint__ *N, const __cminpack_real__ *A, const __cminpack_blasint__ *lda, __cminpack_real__ *X,
   const __cminpack_blasint__ *incX);
-#endif /* !__APPLE__ */
+//#endif /* !__APPLE__ */
 #endif /* USE_BLAS */
 
 #ifdef USE_LAPACK
-#ifdef __APPLE__
-#include <Accelerate/Accelerate.h>
-#else /* !__APPLE__ */
+//#ifdef __APPLE__
+//#include <Accelerate/Accelerate.h>
+//#else /* !__APPLE__ */
 #if defined(__LP64__) /* In LP64 match sizes with the 32 bit ABI */
 typedef int 		__CLPK_integer;
 typedef int 		__CLPK_logical;
@@ -62,7 +62,7 @@
 void __cminpack_lapack__(geqrf_)(
   __CLPK_integer *m, __CLPK_integer *n, __cminpack_real__ *a, __CLPK_integer * lda,
   __cminpack_real__ *tau, __cminpack_real__ *work, __CLPK_integer *lwork, __CLPK_integer *info);
-#endif /* !__APPLE__ */
+//#endif /* !__APPLE__ */
 #endif
 
 #include "minpackP.h"
diff -ruN cminpack-1.3.11/Makefile cminpack-1.3.11-patch/Makefile
--- cminpack-1.3.11/Makefile	2024-09-13 18:04:15.000000000 +0900
+++ cminpack-1.3.11-patch/Makefile	2024-10-27 09:09:37.259671695 +0900
@@ -11,7 +11,7 @@
 #LIBSUFFIX=s
 #CFLAGS="-O3 -g -Wall -Wextra -DUSE_BLAS -DUSE_LAPACK"
 CFLAGS_L=$(CFLAGS) -DUSE_BLAS -DUSE_LAPACK
-LDADD_L=-framework Accelerate
+#LDADD_L=-framework Accelerate
 
 ### configuration for the long double version:
 ## make LIBSUFFIX=s CFLAGS="-O3 -g -Wall -Wextra -D__cminpack_long_double__"
@@ -91,7 +91,7 @@
 install: $(LIB)
 	cp $(LIB) ${DESTDIR}/lib
 	chmod 644 ${DESTDIR}/lib/$(LIB)
-	$(RANLIB) -t ${DESTDIR}/lib/$(LIB) # might be unnecessary
+#	$(RANLIB) -t ${DESTDIR}/lib/$(LIB) # might be unnecessary
 	cp minpack.h ${DESTDIR}/include
 	chmod 644 ${DESTDIR}/include/minpack.h
 	cp cminpack.h ${DESTDIR}/include
