diff -ruN cminpack-1.3.6/cminpack.h cminpack-1.3.6-patch/cminpack.h
--- cminpack-1.3.6/cminpack.h	2017-02-25 00:09:31.000000000 +0900
+++ cminpack-1.3.6-patch/cminpack.h	2018-03-03 20:19:34.897627900 +0900
@@ -36,38 +36,7 @@
 extern "C" {
 #endif /* __cplusplus */
 
-/* Cmake will define cminpack_EXPORTS on Windows when it
-configures to build a shared library. If you are going to use
-another build system on windows or create the visual studio
-projects by hand you need to define cminpack_EXPORTS when
-building a DLL on windows.
-*/
-#if defined (__GNUC__)
-#define CMINPACK_DECLSPEC_EXPORT  __declspec(__dllexport__)
-#define CMINPACK_DECLSPEC_IMPORT  __declspec(__dllimport__)
-#endif
-#if defined (_MSC_VER) || defined (__BORLANDC__)
-#define CMINPACK_DECLSPEC_EXPORT  __declspec(dllexport)
-#define CMINPACK_DECLSPEC_IMPORT  __declspec(dllimport)
-#endif
-#ifdef __WATCOMC__
-#define CMINPACK_DECLSPEC_EXPORT  __export
-#define CMINPACK_DECLSPEC_IMPORT  __import
-#endif
-#ifdef __IBMC__
-#define CMINPACK_DECLSPEC_EXPORT  _Export
-#define CMINPACK_DECLSPEC_IMPORT  _Import
-#endif
-
-#if !defined(CMINPACK_NO_DLL) && (defined(__WIN32__) || defined(WIN32) || defined (_WIN32))
-#if defined(cminpack_EXPORTS) || defined(CMINPACK_EXPORTS) || defined(CMINPACK_DLL_EXPORTS)
-    #define  CMINPACK_EXPORT CMINPACK_DECLSPEC_EXPORT
-  #else
-    #define  CMINPACK_EXPORT CMINPACK_DECLSPEC_IMPORT
-  #endif /* cminpack_EXPORTS */
-#else /* defined (_WIN32) */
  #define CMINPACK_EXPORT
-#endif
 
 #if defined(__CUDA_ARCH__) || defined(__CUDACC__)
 #define __cminpack_attr__ __device__
diff -ruN cminpack-1.3.6/cminpackP.h cminpack-1.3.6-patch/cminpackP.h
--- cminpack-1.3.6/cminpackP.h	2017-02-25 00:09:31.000000000 +0900
+++ cminpack-1.3.6-patch/cminpackP.h	2018-03-03 20:19:08.831974800 +0900
@@ -11,20 +11,20 @@
 #endif
 
 #ifdef USE_CBLAS
-#ifdef __APPLE__
-#include <Accelerate/Accelerate.h>
-#else
+//#ifdef __APPLE__
+//#include <Accelerate/Accelerate.h>
+//#else
 #include <cblas.h>
-#endif
+//#endif
 #define __cminpack_enorm__(n,x) cblas_dnrm2(n,x,1)
 #else
 #define __cminpack_enorm__(n,x) __cminpack_func__(enorm)(n,x)
 #endif
 
 #ifdef USE_LAPACK
-#ifdef __APPLE__
-#include <Accelerate/Accelerate.h>
-#else
+//#ifdef __APPLE__
+//#include <Accelerate/Accelerate.h>
+//#else
 #if defined(__LP64__) /* In LP64 match sizes with the 32 bit ABI */
 typedef int 		__CLPK_integer;
 typedef int 		__CLPK_logical;
@@ -51,7 +51,7 @@
 int dgeqrf_(__CLPK_integer *m, __CLPK_integer *n, __CLPK_doublereal *a, __CLPK_integer *
             lda, __CLPK_doublereal *tau, __CLPK_doublereal *work, __CLPK_integer *lwork, __CLPK_integer *info);
 #endif
-#endif
+//#endif
 
 #include "minpackP.h"
 
diff -ruN cminpack-1.3.6/Makefile cminpack-1.3.6-patch/Makefile
--- cminpack-1.3.6/Makefile	2017-02-25 00:09:31.000000000 +0900
+++ cminpack-1.3.6-patch/Makefile	2018-03-03 20:19:08.847604700 +0900
@@ -11,7 +11,7 @@
 #LIBSUFFIX=s
 #CFLAGS="-O3 -g -Wall -Wextra -DUSE_CBLAS -DUSE_LAPACK"
 CFLAGS_L=$(CFLAGS) -DUSE_CBLAS -DUSE_LAPACK
-LDADD_L=-framework Accelerate
+#LDADD_L=-framework Accelerate
 
 ### configuration for the long double version:
 ## make LIBSUFFIX=s CFLAGS="-O3 -g -Wall -Wextra -D__cminpack_long_double__"
@@ -109,7 +109,7 @@
 install: libcminpack$(LIBSUFFIX).a
 	cp libcminpack$(LIBSUFFIX).a ${DESTDIR}/lib
 	chmod 644 ${DESTDIR}/lib/libcminpack$(LIBSUFFIX).a
-	ranlib -t ${DESTDIR}/lib/libcminpack$(LIBSUFFIX).a # might be unnecessary
+#	ranlib -t ${DESTDIR}/lib/libcminpack$(LIBSUFFIX).a # might be unnecessary
 	cp minpack.h ${DESTDIR}/include
 	chmod 644 ${DESTDIR}/include/minpack.h
 	cp cminpack.h ${DESTDIR}/include
