diff -ruN mkl-dnn-1.0/include/mkldnn.hpp mkl-dnn-1.0-patch/include/mkldnn.hpp
--- mkl-dnn-1.0/include/mkldnn.hpp	2019-07-11 02:03:46.000000000 +0900
+++ mkl-dnn-1.0-patch/include/mkldnn.hpp	2019-07-27 22:47:33.629745898 +0900
@@ -73,6 +73,8 @@
         if (status != mkldnn_success)
             throw error(status, message);
     }
+
+    const std::string& toStdString() { return std::string(message).append(": status = ").append(std::to_string(status)); }
 };
 
 /// A class that provides the destructor for an Intel(R) MKL-DNN C handle
diff -ruN mkl-dnn-1.0/src/cpu/cpu_engine.cpp mkl-dnn-1.0-patch/src/cpu/cpu_engine.cpp
--- mkl-dnn-1.0/src/cpu/cpu_engine.cpp	2019-07-11 02:03:46.000000000 +0900
+++ mkl-dnn-1.0-patch/src/cpu/cpu_engine.cpp	2019-07-23 21:11:44.409275262 +0900
@@ -75,6 +75,23 @@
 namespace impl {
 namespace cpu {
 
+template struct jit_uni_dw_conv_fwd_kernel<avx512_core, data_type::bf16>;
+template struct jit_uni_dw_conv_fwd_kernel<avx512_common, data_type::f32>;
+template struct jit_uni_dw_conv_fwd_kernel<avx2, data_type::f32>;
+template struct jit_uni_dw_conv_fwd_kernel<sse41, data_type::f32>;
+
+template struct jit_uni_dw_conv_bwd_data_kernel<avx512_core, data_type::bf16>;
+template struct jit_uni_dw_conv_bwd_data_kernel<avx512_common, data_type::f32>;
+template struct jit_uni_dw_conv_bwd_data_kernel<avx2, data_type::f32>;
+template struct jit_uni_dw_conv_bwd_data_kernel<sse41, data_type::f32>;
+
+template struct jit_uni_dw_conv_bwd_weights_kernel<avx512_core,
+        data_type::bf16>;
+template struct jit_uni_dw_conv_bwd_weights_kernel<avx512_common,
+        data_type::f32>;
+template struct jit_uni_dw_conv_bwd_weights_kernel<avx2, data_type::f32>;
+template struct jit_uni_dw_conv_bwd_weights_kernel<sse41, data_type::f32>;
+
 status_t cpu_engine_t::create_memory_storage(
         memory_storage_t **storage, unsigned flags, size_t size, void *handle) {
     return safe_ptr_assign<memory_storage_t>(
diff -ruN mkl-dnn-1.0/src/cpu/jit_uni_dw_conv_kernel_utils.hpp mkl-dnn-1.0-patch/src/cpu/jit_uni_dw_conv_kernel_utils.hpp
--- mkl-dnn-1.0/src/cpu/jit_uni_dw_conv_kernel_utils.hpp	2019-07-11 02:03:46.000000000 +0900
+++ mkl-dnn-1.0-patch/src/cpu/jit_uni_dw_conv_kernel_utils.hpp	2019-07-23 21:11:39.279269441 +0900
@@ -202,11 +202,6 @@
         scratchpad.book(key_conv_padded_bias, sizeof(float) * jcp.oc);
 }
 
-template struct jit_uni_dw_conv_fwd_kernel<avx512_core, data_type::bf16>;
-template struct jit_uni_dw_conv_fwd_kernel<avx512_common, data_type::f32>;
-template struct jit_uni_dw_conv_fwd_kernel<avx2, data_type::f32>;
-template struct jit_uni_dw_conv_fwd_kernel<sse41, data_type::f32>;
-
 template <cpu_isa_t isa, data_type_t kernel_dt>
 struct jit_uni_dw_conv_bwd_data_kernel {
 
@@ -341,11 +336,6 @@
     UNUSED(jcp);
 }
 
-template struct jit_uni_dw_conv_bwd_data_kernel<avx512_core, data_type::bf16>;
-template struct jit_uni_dw_conv_bwd_data_kernel<avx512_common, data_type::f32>;
-template struct jit_uni_dw_conv_bwd_data_kernel<avx2, data_type::f32>;
-template struct jit_uni_dw_conv_bwd_data_kernel<sse41, data_type::f32>;
-
 template <cpu_isa_t isa, data_type_t kernel_dt>
 struct jit_uni_dw_conv_bwd_weights_kernel {
 
@@ -518,13 +508,6 @@
 
     jcp.nthr = jcp.nthr_g * jcp.nthr_mb;
 }
-
-template struct jit_uni_dw_conv_bwd_weights_kernel<avx512_core,
-        data_type::bf16>;
-template struct jit_uni_dw_conv_bwd_weights_kernel<avx512_common,
-        data_type::f32>;
-template struct jit_uni_dw_conv_bwd_weights_kernel<avx2, data_type::f32>;
-template struct jit_uni_dw_conv_bwd_weights_kernel<sse41, data_type::f32>;
 }
 }
 }
