diff -ruN mkl-dnn-0.19/src/cpu/cpu_engine.cpp mkl-dnn-0.19-patch/src/cpu/cpu_engine.cpp
--- mkl-dnn-0.19/src/cpu/cpu_engine.cpp	2019-05-15 07:03:33.000000000 +0900
+++ mkl-dnn-0.19-patch/src/cpu/cpu_engine.cpp	2019-05-24 20:23:12.078486746 +0900
@@ -72,6 +72,34 @@
 
 using namespace mkldnn::impl::status;
 
+template struct jit_uni_gru_cell_postgemm_part1_fwd<sse42, data_type::f32>;
+template struct jit_uni_gru_cell_postgemm_part1_fwd<avx2, data_type::f32>;
+template struct jit_uni_gru_cell_postgemm_part1_fwd<avx512_core, data_type::f32>;
+
+template struct jit_uni_gru_cell_postgemm_part2_fwd<sse42, data_type::f32>;
+template struct jit_uni_gru_cell_postgemm_part2_fwd<avx2, data_type::f32>;
+template struct jit_uni_gru_cell_postgemm_part2_fwd<avx512_core, data_type::f32>;
+
+template struct jit_uni_gru_lbr_cell_postgemm_fwd<sse42, data_type::f32>;
+template struct jit_uni_gru_lbr_cell_postgemm_fwd<avx2, data_type::f32>;
+template struct jit_uni_gru_lbr_cell_postgemm_fwd<avx512_core, data_type::f32>;
+
+template struct jit_uni_lstm_cell_postgemm_fwd<sse42, data_type::f32>;
+template struct jit_uni_lstm_cell_postgemm_fwd<avx2, data_type::f32>;
+template struct jit_uni_lstm_cell_postgemm_fwd<avx512_core, data_type::f32>;
+
+template struct jit_uni_lstm_cell_postgemm_fwd<sse42, data_type::u8>;
+template struct jit_uni_lstm_cell_postgemm_fwd<avx2, data_type::u8>;
+template struct jit_uni_lstm_cell_postgemm_fwd<avx512_core, data_type::u8>;
+
+template struct jit_uni_rnn_cell_postgemm_fwd<sse42, data_type::f32>;
+template struct jit_uni_rnn_cell_postgemm_fwd<avx2, data_type::f32>;
+template struct jit_uni_rnn_cell_postgemm_fwd<avx512_core, data_type::f32>;
+
+template struct jit_uni_rnn_cell_postgemm_fwd<sse42, data_type::u8>;
+template struct jit_uni_rnn_cell_postgemm_fwd<avx2, data_type::u8>;
+template struct jit_uni_rnn_cell_postgemm_fwd<avx512_core, data_type::u8>;
+
 status_t cpu_engine_t::memory_primitive_desc_create(memory_pd_t **pd,
         const memory_desc_t *desc) {
     return safe_ptr_assign<memory_pd_t>(*pd,
diff -ruN mkl-dnn-0.19/src/cpu/rnn/jit_uni_gru_cell_postgemm_1.hpp mkl-dnn-0.19-patch/src/cpu/rnn/jit_uni_gru_cell_postgemm_1.hpp
--- mkl-dnn-0.19/src/cpu/rnn/jit_uni_gru_cell_postgemm_1.hpp	2019-05-15 07:03:33.000000000 +0900
+++ mkl-dnn-0.19-patch/src/cpu/rnn/jit_uni_gru_cell_postgemm_1.hpp	2019-05-24 20:23:07.239487389 +0900
@@ -172,10 +172,6 @@
 
 };
 
-template struct jit_uni_gru_cell_postgemm_part1_fwd<sse42, data_type::f32>;
-template struct jit_uni_gru_cell_postgemm_part1_fwd<avx2, data_type::f32>;
-template struct jit_uni_gru_cell_postgemm_part1_fwd<avx512_core, data_type::f32>;
-
 }
 }
 }
diff -ruN mkl-dnn-0.19/src/cpu/rnn/jit_uni_gru_cell_postgemm_2.hpp mkl-dnn-0.19-patch/src/cpu/rnn/jit_uni_gru_cell_postgemm_2.hpp
--- mkl-dnn-0.19/src/cpu/rnn/jit_uni_gru_cell_postgemm_2.hpp	2019-05-15 07:03:33.000000000 +0900
+++ mkl-dnn-0.19-patch/src/cpu/rnn/jit_uni_gru_cell_postgemm_2.hpp	2019-05-24 20:22:37.968491285 +0900
@@ -172,10 +172,6 @@
 
 };
 
-template struct jit_uni_gru_cell_postgemm_part2_fwd<sse42, data_type::f32>;
-template struct jit_uni_gru_cell_postgemm_part2_fwd<avx2, data_type::f32>;
-template struct jit_uni_gru_cell_postgemm_part2_fwd<avx512_core, data_type::f32>;
-
 }
 }
 }
diff -ruN mkl-dnn-0.19/src/cpu/rnn/jit_uni_gru_lbr_cell_postgemm.hpp mkl-dnn-0.19-patch/src/cpu/rnn/jit_uni_gru_lbr_cell_postgemm.hpp
--- mkl-dnn-0.19/src/cpu/rnn/jit_uni_gru_lbr_cell_postgemm.hpp	2019-05-15 07:03:33.000000000 +0900
+++ mkl-dnn-0.19-patch/src/cpu/rnn/jit_uni_gru_lbr_cell_postgemm.hpp	2019-05-24 20:22:28.569492537 +0900
@@ -221,10 +221,6 @@
 
 };
 
-template struct jit_uni_gru_lbr_cell_postgemm_fwd<sse42, data_type::f32>;
-template struct jit_uni_gru_lbr_cell_postgemm_fwd<avx2, data_type::f32>;
-template struct jit_uni_gru_lbr_cell_postgemm_fwd<avx512_core, data_type::f32>;
-
 }
 }
 }
diff -ruN mkl-dnn-0.19/src/cpu/rnn/jit_uni_lstm_cell_postgemm.hpp mkl-dnn-0.19-patch/src/cpu/rnn/jit_uni_lstm_cell_postgemm.hpp
--- mkl-dnn-0.19/src/cpu/rnn/jit_uni_lstm_cell_postgemm.hpp	2019-05-15 07:03:33.000000000 +0900
+++ mkl-dnn-0.19-patch/src/cpu/rnn/jit_uni_lstm_cell_postgemm.hpp	2019-05-24 20:22:17.576494003 +0900
@@ -373,14 +373,6 @@
 
 };
 
-template struct jit_uni_lstm_cell_postgemm_fwd<sse42, data_type::f32>;
-template struct jit_uni_lstm_cell_postgemm_fwd<avx2, data_type::f32>;
-template struct jit_uni_lstm_cell_postgemm_fwd<avx512_core, data_type::f32>;
-
-template struct jit_uni_lstm_cell_postgemm_fwd<sse42, data_type::u8>;
-template struct jit_uni_lstm_cell_postgemm_fwd<avx2, data_type::u8>;
-template struct jit_uni_lstm_cell_postgemm_fwd<avx512_core, data_type::u8>;
-
 }
 }
 }
diff -ruN mkl-dnn-0.19/src/cpu/rnn/jit_uni_rnn_cell_postgemm.hpp mkl-dnn-0.19-patch/src/cpu/rnn/jit_uni_rnn_cell_postgemm.hpp
--- mkl-dnn-0.19/src/cpu/rnn/jit_uni_rnn_cell_postgemm.hpp	2019-05-15 07:03:33.000000000 +0900
+++ mkl-dnn-0.19-patch/src/cpu/rnn/jit_uni_rnn_cell_postgemm.hpp	2019-05-24 20:22:03.484495884 +0900
@@ -280,14 +280,6 @@
 
 };
 
-template struct jit_uni_rnn_cell_postgemm_fwd<sse42, data_type::f32>;
-template struct jit_uni_rnn_cell_postgemm_fwd<avx2, data_type::f32>;
-template struct jit_uni_rnn_cell_postgemm_fwd<avx512_core, data_type::f32>;
-
-template struct jit_uni_rnn_cell_postgemm_fwd<sse42, data_type::u8>;
-template struct jit_uni_rnn_cell_postgemm_fwd<avx2, data_type::u8>;
-template struct jit_uni_rnn_cell_postgemm_fwd<avx512_core, data_type::u8>;
-
 }
 }
 }
