diff --git a/onnxruntime/core/providers/openvino/qdq_transformations/qdq_scales_fix.cc b/onnxruntime/core/providers/openvino/qdq_transformations/qdq_scales_fix.cc
index 5de7607a87..a03a8b1c6e 100644
--- a/onnxruntime/core/providers/openvino/qdq_transformations/qdq_scales_fix.cc
+++ b/onnxruntime/core/providers/openvino/qdq_transformations/qdq_scales_fix.cc
@@ -563,7 +563,9 @@ struct CustomGraph {
       if (!found) {
         // Connect the last valid node to the graph output
-        for (auto node : std::ranges::reverse_view(original_graph.Nodes())) {
+        auto graph_nodes = original_graph.Nodes();
+        for (auto node_it = graph_nodes.rbegin(); node_it != graph_nodes.rend(); ++node_it) {
+          auto node = *node_it;
           if (!node->OutputDefs().empty()) {
             const auto& name = (*node->OutputDefs().begin())->Name();
             auto& node_arg = original_graph.GetOrCreateNodeArg(name, output->TypeAsProto());
