diff -ruN ARToolKitPlus-2.3.0/android-arm.cmake ARToolKitPlus-2.3.0-patch/android-arm.cmake
--- ARToolKitPlus-2.3.0/android-arm.cmake	1970-01-01 09:00:00.000000000 +0900
+++ ARToolKitPlus-2.3.0-patch/android-arm.cmake	2014-03-23 09:47:59.604848895 +0900
@@ -0,0 +1,26 @@
+# CMake toolchain to build ARToolKitPlus 2.3.0 for Android 2.2. Sample usage:
+#
+# ANDROID_BIN=`pwd`/../android-ndk-r9d/toolchains/arm-linux-androideabi-4.6/prebuilt/linux-x86_64/bin/arm-linux-androideabi \
+# ANDROID_CPP=`pwd`/../android-ndk-r9d/sources/cxx-stl/gnu-libstdc++/4.6/ \
+# ANDROID_ROOT=`pwd`/../android-ndk-r9d/platforms/android-9/arch-arm/ \
+# cmake -DCMAKE_TOOLCHAIN_FILE=android-arm.cmake -DCMAKE_INSTALL_PREFIX=..
+#
+# If you really need to use ARToolKitPlus on a CPU with no FPU, replace "libs/armeabi-v7a" by "libs/armeabi" and
+# "-march=armv7-a -mfloat-abi=softfp -mfpu=vfpv3-d16" with "-march=armv5te -mtune=xscale -msoft-float"
+
+set(CMAKE_SYSTEM_NAME UnixPaths)
+set(CMAKE_SYSTEM_PROCESSOR arm)
+set(ANDROID TRUE)
+
+set(CMAKE_C_COMPILER   "$ENV{ANDROID_BIN}-gcc")
+set(CMAKE_CXX_COMPILER "$ENV{ANDROID_BIN}-g++")
+
+set(CMAKE_C_LINK_EXECUTABLE    "<CMAKE_C_COMPILER>   <FLAGS> <CMAKE_C_LINK_FLAGS>   <LINK_FLAGS> <OBJECTS> -o <TARGET> -nostdlib <LINK_LIBRARIES> -L$ENV{ANDROID_ROOT}/usr/lib/ -lgcc -ldl -lz -lm -lc")
+set(CMAKE_CXX_LINK_EXECUTABLE  "<CMAKE_CXX_COMPILER> <FLAGS> <CMAKE_CXX_LINK_FLAGS> <LINK_FLAGS> <OBJECTS> -o <TARGET> -nostdlib <LINK_LIBRARIES> -L$ENV{ANDROID_CPP}/libs/armeabi-v7a/ -L$ENV{ANDROID_ROOT}/usr/lib/ -lgnustl_static -lgcc -ldl -lz -lm -lc")
+
+set(CMAKE_C_CREATE_SHARED_LIBRARY    "<CMAKE_C_COMPILER> <CMAKE_SHARED_LIBRARY_C_FLAGS> <LANGUAGE_COMPILE_FLAGS> <LINK_FLAGS> <CMAKE_SHARED_LIBRARY_CREATE_C_FLAGS> <CMAKE_SHARED_LIBRARY_SONAME_C_FLAG><TARGET_SONAME> -o <TARGET> <OBJECTS> -nostdlib <LINK_LIBRARIES> -L$ENV{ANDROID_ROOT}/usr/lib/ -lgcc -ldl -lz -lm -lc")
+set(CMAKE_CXX_CREATE_SHARED_LIBRARY  "<CMAKE_CXX_COMPILER> <CMAKE_SHARED_LIBRARY_CXX_FLAGS> <LANGUAGE_COMPILE_FLAGS> <LINK_FLAGS> <CMAKE_SHARED_LIBRARY_CREATE_CXX_FLAGS> <CMAKE_SHARED_LIBRARY_SONAME_CXX_FLAG><TARGET_SONAME> -o <TARGET> <OBJECTS> -nostdlib <LINK_LIBRARIES> -L$ENV{ANDROID_CPP}/libs/armeabi-v7a/ -L$ENV{ANDROID_ROOT}/usr/lib/ -lgnustl_static -lgcc -ldl -lz -lm -lc")
+
+add_definitions("-DANDROID -fPIC -ffunction-sections -funwind-tables -fstack-protector -march=armv7-a -mfloat-abi=softfp -mfpu=vfpv3-d16 -fomit-frame-pointer -fstrict-aliasing -funswitch-loops -finline-limit=300")
+
+include_directories("$ENV{ANDROID_CPP}/include/" "$ENV{ANDROID_CPP}/libs/armeabi-v7a/include/" "$ENV{ANDROID_ROOT}/usr/include/")
diff -ruN ARToolKitPlus-2.3.0/CMakeLists.txt ARToolKitPlus-2.3.0-patch/CMakeLists.txt
--- ARToolKitPlus-2.3.0/CMakeLists.txt	2012-09-15 21:36:20.000000000 +0900
+++ ARToolKitPlus-2.3.0-patch/CMakeLists.txt	2014-03-23 09:47:59.604848895 +0900
@@ -10,7 +10,7 @@
 if(WIN32)
 	add_definitions(-D_CRT_SECURE_NO_WARNINGS -DAR_LIBRARY)
 else(WIN32)
-    add_definitions(-Wall -O3)
+    add_definitions(-Wall -O3 -fPIC)
 endif(WIN32)
 
 if(ARTK_USE_BOOST)
@@ -29,18 +29,18 @@
 	${AR_SOURCE_DIR}/src/librpp/*.cpp
 	)
 
-add_library(ARToolKitPlus SHARED ${HEADERS_AR} ${SOURCES_AR})
+add_library(ARToolKitPlus STATIC ${HEADERS_AR} ${SOURCES_AR})
 
 if(ARTK_BUILD_EXAMPLES)
     add_subdirectory(bin/)
 endif(ARTK_BUILD_EXAMPLES)
 
 # make install
-set(CMAKE_INSTALL_PREFIX /usr)
+# set(CMAKE_INSTALL_PREFIX /usr)
 
 install(TARGETS ARToolKitPlus 
-LIBRARY DESTINATION lib  
-ARCHIVE DESTINATION lib
+LIBRARY DESTINATION lib${LIB_SUFFIX} 
+ARCHIVE DESTINATION lib${LIB_SUFFIX}
 RUNTIME DESTINATION bin)
 
 install(DIRECTORY include/ DESTINATION include)
diff -ruN ARToolKitPlus-2.3.0/include/ARToolKitPlus/arGetInitRot2Sub.h ARToolKitPlus-2.3.0-patch/include/ARToolKitPlus/arGetInitRot2Sub.h
--- ARToolKitPlus-2.3.0/include/ARToolKitPlus/arGetInitRot2Sub.h	2012-09-15 21:36:20.000000000 +0900
+++ ARToolKitPlus-2.3.0-patch/include/ARToolKitPlus/arGetInitRot2Sub.h	2014-03-23 09:48:33.805441931 +0900
@@ -25,7 +25,7 @@
 
 namespace rpp {
 
-void AR_EXPORT arGetInitRot2_sub(rpp_float &err, rpp_mat &R, rpp_vec &t, const rpp_float cc[2], const rpp_float fc[2],
+void arGetInitRot2_sub(rpp_float &err, rpp_mat &R, rpp_vec &t, const rpp_float cc[2], const rpp_float fc[2],
         const rpp_vec *model, const rpp_vec *iprts, const unsigned int model_iprts_size, const rpp_mat R_init,
         const bool estimate_R_init, const rpp_float epsilon, const rpp_float tolerance,
         const unsigned int max_iterations);
