diff -ruN OpenBLAS-0.3.32/exports/Makefile OpenBLAS-0.3.32-macosx/exports/Makefile
--- OpenBLAS-0.3.32/exports/Makefile	2026-03-24 07:53:57.000000000 +0900
+++ OpenBLAS-0.3.32-macosx/exports/Makefile	2026-04-11 17:48:24.119283462 +0900
@@ -153,24 +153,7 @@
 	$(OBJCONV) @objconv.def ../$(LIBNAME) ../$(LIBNAME).osx.renamed
 $(LIBDYNNAME) : ../$(LIBNAME).osx.renamed osx.def
 endif
-ifneq (,$(filter 1 2,$(NOFORTRAN)))
-#only build without Fortran
-	$(CC) $(CFLAGS) $(LDFLAGS) -all_load -headerpad_max_install_names -install_name "$(CURDIR)/../$(INTERNALNAME)" -dynamiclib -o ../$(LIBDYNNAME) $< -Wl,-exported_symbols_list,osx.def  $(FEXTRALIB)
-else
-ifeq ($(F_COMPILER), INTEL)
-	$(FC) $(FFLAGS) $(LDFLAGS) -all-load -headerpad_max_install_names -install_name "$(CURDIR)/../$(INTERNALNAME)" -dynamiclib -o ../$(LIBDYNNAME) $< -Wl,-exported_symbols_list,osx.def
-else
-ifeq ($(F_COMPILER), FLANG)
-	$(FC) $(FFLAGS) $(LDFLAGS) -fno-fortran-main -Mnomain -all_load -headerpad_max_install_names -install_name "$(CURDIR)/../$(INTERNALNAME)" -dynamiclib -o ../$(LIBDYNNAME) $< -Wl,-exported_symbols_list,osx.def  $(FEXTRALIB)
-else
-ifeq ($(F_COMPILER), FLANGNEW)
-	$(FC) $(FFLAGS) $(LDFLAGS) -Wl,-all_load -Wl,-headerpad_max_install_names -Wl,-install_name,"$(CURDIR)/../$(INTERNALNAME)" -Wl,-dylib -o ../$(LIBDYNNAME) $< -Wl,-exported_symbols_list,osx.def  $(FEXTRALIB)
-else
-	$(FC) $(FFLAGS) $(LDFLAGS) -all_load -headerpad_max_install_names -install_name "$(CURDIR)/../$(INTERNALNAME)" -dynamiclib -o ../$(LIBDYNNAME) $< -Wl,-exported_symbols_list,osx.def  $(FEXTRALIB)
-endif
-endif
-endif
-endif
+	$(CC) $(CFLAGS) $(LDFLAGS) -all_load -headerpad_max_install_names -install_name "$(CURDIR)/../$(INTERNALNAME)" -dynamiclib -o ../$(LIBDYNNAME) $< -Wl,-exported_symbols_list,osx.def
 
 dllinit.$(SUFFIX) : dllinit.c
 	$(CC) $(CFLAGS) -c -o $(@F) -s $<
diff -ruN OpenBLAS-0.3.32/Makefile.install OpenBLAS-0.3.32-macosx/Makefile.install
--- OpenBLAS-0.3.32/Makefile.install	2026-03-24 07:53:57.000000000 +0900
+++ OpenBLAS-0.3.32-macosx/Makefile.install	2026-04-11 17:47:06.510369276 +0900
@@ -143,7 +143,7 @@
 endif
 ifeq ($(OSNAME), Darwin)
 	@-cp $(LIBDYNNAME) "$(DESTDIR)$(OPENBLAS_LIBRARY_DIR)"
-	@-install_name_tool -id "$(OPENBLAS_LIBRARY_DIR)/$(LIBPREFIX).$(MAJOR_VERSION).dylib" "$(DESTDIR)$(OPENBLAS_LIBRARY_DIR)/$(LIBDYNNAME)"
+	@-install_name_tool -id "@rpath/$(LIBPREFIX).$(MAJOR_VERSION).dylib" "$(DESTDIR)$(OPENBLAS_LIBRARY_DIR)/$(LIBDYNNAME)"
 	@cd "$(DESTDIR)$(OPENBLAS_LIBRARY_DIR)" ; \
 	$(LNCMD) $(LIBDYNNAME) $(LIBPREFIX).dylib ; \
 	$(LNCMD) $(LIBDYNNAME) $(LIBPREFIX).$(MAJOR_VERSION).dylib
