diff -ruN OpenBLAS-0.2.19/Makefile.install OpenBLAS-0.2.19-macosx/Makefile.install
--- OpenBLAS-0.2.19/Makefile.install	2016-09-01 12:58:42.000000000 +0900
+++ OpenBLAS-0.2.19-macosx/Makefile.install	2016-10-22 08:18:19.202641832 +0900
@@ -82,7 +82,7 @@
 endif
 ifeq ($(OSNAME), Darwin)
 	@-cp $(LIBDYNNAME) "$(DESTDIR)$(OPENBLAS_LIBRARY_DIR)"
-	@-install_name_tool -id "$(DESTDIR)$(OPENBLAS_LIBRARY_DIR)/$(LIBDYNNAME)" "$(DESTDIR)$(OPENBLAS_LIBRARY_DIR)/$(LIBDYNNAME)"
+	@-install_name_tool -id "@rpath/$(LIBPREFIX).dylib" "$(DESTDIR)$(OPENBLAS_LIBRARY_DIR)/$(LIBDYNNAME)"
 	@cd "$(DESTDIR)$(OPENBLAS_LIBRARY_DIR)" ; \
 	ln -fs $(LIBDYNNAME) $(LIBPREFIX).dylib
 endif
