diff -ruN OpenBLAS-0.3.10/driver/level2/gemv_thread.c OpenBLAS-0.3.10-ios/driver/level2/gemv_thread.c
--- OpenBLAS-0.3.10/driver/level2/gemv_thread.c	2020-06-15 05:03:04.000000000 +0900
+++ OpenBLAS-0.3.10-ios/driver/level2/gemv_thread.c	2020-06-22 13:31:13.932546177 +0900
@@ -63,23 +63,8 @@
 #endif
 
 #ifndef thread_local
-# if __STDC_VERSION__ >= 201112 && !defined __STDC_NO_THREADS__
-#  define thread_local _Thread_local
-# elif defined _WIN32 && ( \
-       defined _MSC_VER || \
-       defined __ICL || \
-       defined __DMC__ || \
-       defined __BORLANDC__ )
-#  define thread_local __declspec(thread) 
-/* note that ICC (linux) and Clang are covered by __GNUC__ */
-# elif (defined __GNUC__ || \
-       defined __SUNPRO_C || \
-       defined __xlC__) && !defined(__APPLE__)
-#  define thread_local __thread
-# else
 # define UNSAFE
 #endif
-#endif
 #if defined USE_OPENMP
 #undef UNSAFE
 #endif
