diff -ruN OpenBLAS-0.3.15/exports/Makefile OpenBLAS-0.3.15-macosx/exports/Makefile
--- OpenBLAS-0.3.15/exports/Makefile	2021-05-03 06:50:22.000000000 +0900
+++ OpenBLAS-0.3.15-macosx/exports/Makefile	2021-05-11 19:11:48.316710867 +0900
@@ -136,16 +136,7 @@
 	$(OBJCONV) @objconv.def ../$(LIBNAME) ../$(LIBNAME).osx.renamed
 $(LIBDYNNAME) : ../$(LIBNAME).osx.renamed osx.def
 endif
-ifneq (,$(filter 1 2,$(NOFORTRAN)))
-#only build without Fortran
-	$(CC) $(CFLAGS) $(LDFLAGS) -all_load -headerpad_max_install_names -install_name "$(CURDIR)/../$(INTERNALNAME)" -dynamiclib -o ../$(LIBDYNNAME) $< -Wl,-exported_symbols_list,osx.def  $(FEXTRALIB)
-else
-ifeq ($(F_COMPILER), INTEL)
-	$(FC) $(FFLAGS) $(LDFLAGS) -all-load -headerpad_max_install_names -install_name "$(CURDIR)/../$(INTERNALNAME)" -dynamiclib -o ../$(LIBDYNNAME) $< -Wl,-exported_symbols_list,osx.def
-else
-	$(FC) $(FFLAGS) $(LDFLAGS) -all_load -headerpad_max_install_names -install_name "$(CURDIR)/../$(INTERNALNAME)" -dynamiclib -o ../$(LIBDYNNAME) $< -Wl,-exported_symbols_list,osx.def  $(FEXTRALIB)
-endif
-endif
+	$(CC) $(CFLAGS) $(LDFLAGS) -all_load -headerpad_max_install_names -install_name "$(CURDIR)/../$(INTERNALNAME)" -dynamiclib -o ../$(LIBDYNNAME) $< -Wl,-exported_symbols_list,osx.def
 
 dllinit.$(SUFFIX) : dllinit.c
 	$(CC) $(CFLAGS) -c -o $(@F) -s $<
diff -ruN OpenBLAS-0.3.15/Makefile.install OpenBLAS-0.3.15-macosx/Makefile.install
--- OpenBLAS-0.3.15/Makefile.install	2021-05-03 06:50:22.000000000 +0900
+++ OpenBLAS-0.3.15-macosx/Makefile.install	2021-05-11 16:57:11.009523603 +0900
@@ -105,7 +105,7 @@
 endif
 ifeq ($(OSNAME), Darwin)
 	@-cp $(LIBDYNNAME) "$(DESTDIR)$(OPENBLAS_LIBRARY_DIR)"
-	@-install_name_tool -id "$(OPENBLAS_LIBRARY_DIR)/$(LIBPREFIX).$(MAJOR_VERSION).dylib" "$(DESTDIR)$(OPENBLAS_LIBRARY_DIR)/$(LIBDYNNAME)"
+	@-install_name_tool -id "@rpath/$(LIBPREFIX).$(MAJOR_VERSION).dylib" "$(DESTDIR)$(OPENBLAS_LIBRARY_DIR)/$(LIBDYNNAME)"
 	@cd "$(DESTDIR)$(OPENBLAS_LIBRARY_DIR)" ; \
 	ln -fs $(LIBDYNNAME) $(LIBPREFIX).dylib ; \
 	ln -fs $(LIBDYNNAME) $(LIBPREFIX).$(MAJOR_VERSION).dylib
