diff -ruN OpenBLAS-0.2.19/common.h OpenBLAS-0.2.19-android/common.h
--- OpenBLAS-0.2.19/common.h	2016-09-01 12:58:42.000000000 +0900
+++ OpenBLAS-0.2.19-android/common.h	2016-10-21 11:26:16.001064468 +0900
@@ -100,9 +100,9 @@
 #ifdef OS_ANDROID
 #define NO_SYSV_IPC
 //Android NDK only supports complex.h since Android 5.0
-#if __ANDROID_API__ < 21
+//#if __ANDROID_API__ < 21
 #define FORCE_OPENBLAS_COMPLEX_STRUCT
-#endif
+//#endif
 #endif
 
 #ifdef OS_WINDOWS
diff -ruN OpenBLAS-0.2.19/exports/Makefile OpenBLAS-0.2.19-android/exports/Makefile
--- OpenBLAS-0.2.19/exports/Makefile	2016-09-01 12:58:42.000000000 +0900
+++ OpenBLAS-0.2.19-android/exports/Makefile	2016-10-21 11:26:16.001064468 +0900
@@ -118,7 +118,7 @@
 dllinit.$(SUFFIX) : dllinit.c
 	$(CC) $(CFLAGS) -c -o $(@F) -s $<
 
-ifeq ($(OSNAME), $(filter $(OSNAME),Linux SunOS))
+ifeq ($(OSNAME), $(filter $(OSNAME),Android Linux SunOS))
 
 so : ../$(LIBSONAME)
 
@@ -132,13 +132,13 @@
 ifneq ($(C_COMPILER), LSB)
 	$(CC) $(CFLAGS) $(LDFLAGS) -shared -o ../$(LIBSONAME) \
 	-Wl,--whole-archive $< -Wl,--no-whole-archive \
-	-Wl,-soname,$(LIBPREFIX).so.$(MAJOR_VERSION) $(EXTRALIB)
+	-Wl,-soname,$(LIBPREFIX).so $(EXTRALIB)
 	$(CC) $(CFLAGS) $(LDFLAGS) -w -o linktest linktest.c ../$(LIBSONAME) $(FEXTRALIB) && echo OK.
 else
 #for LSB
 	env LSBCC_SHAREDLIBS=gfortran $(CC) $(CFLAGS) $(LDFLAGS) -shared -o ../$(LIBSONAME) \
 	-Wl,--whole-archive $< -Wl,--no-whole-archive \
-	-Wl,-soname,$(LIBPREFIX).so.$(MAJOR_VERSION) $(EXTRALIB)
+	-Wl,-soname,$(LIBPREFIX).so $(EXTRALIB)
 	$(FC) $(CFLAGS) $(LDFLAGS) -w -o linktest linktest.c ../$(LIBSONAME) $(FEXTRALIB) && echo OK.
 endif
 	rm -f linktest
diff -ruN OpenBLAS-0.2.19/kernel/arm/zdot.c OpenBLAS-0.2.19-android/kernel/arm/zdot.c
--- OpenBLAS-0.2.19/kernel/arm/zdot.c	2016-09-01 12:58:42.000000000 +0900
+++ OpenBLAS-0.2.19-android/kernel/arm/zdot.c	2016-10-21 11:26:16.002064489 +0900
@@ -36,12 +36,12 @@
 
 #include "common.h"
 
-#ifndef _MSC_VER
-#include <complex.h>
-FLOAT _Complex CNAME(BLASLONG n, FLOAT *x, BLASLONG inc_x, FLOAT *y, BLASLONG inc_y)
-#else
+//#ifndef _MSC_VER
+//#include <complex.h>
+//FLOAT _Complex CNAME(BLASLONG n, FLOAT *x, BLASLONG inc_x, FLOAT *y, BLASLONG inc_y)
+//#else
 OPENBLAS_COMPLEX_FLOAT CNAME(BLASLONG n, FLOAT *x, BLASLONG inc_x, FLOAT *y, BLASLONG inc_y)
-#endif
+//#endif
 {
 	BLASLONG i=0;
 	BLASLONG ix=0,iy=0;
diff -ruN OpenBLAS-0.2.19/lapack-netlib/LAPACKE/include/lapacke_config.h OpenBLAS-0.2.19-android/lapack-netlib/LAPACKE/include/lapacke_config.h
--- OpenBLAS-0.2.19/lapack-netlib/LAPACKE/include/lapacke_config.h	2016-09-01 12:58:42.000000000 +0900
+++ OpenBLAS-0.2.19-android/lapack-netlib/LAPACKE/include/lapacke_config.h	2016-10-21 11:26:16.002064489 +0900
@@ -36,9 +36,9 @@
 
 // For Android prior to API 21 (no <complex> include)
 #if defined(__ANDROID__)
-#if __ANDROID_API__ < 21
+//#if __ANDROID_API__ < 21
 #define LAPACK_COMPLEX_STRUCTURE
-#endif
+//#endif
 #endif
 
 #ifdef __cplusplus
diff -ruN OpenBLAS-0.2.19/Makefile OpenBLAS-0.2.19-android/Makefile
--- OpenBLAS-0.2.19/Makefile	2016-09-01 12:58:42.000000000 +0900
+++ OpenBLAS-0.2.19-android/Makefile	2016-10-21 11:26:16.002064489 +0900
@@ -81,7 +81,7 @@
 
 shared :
 ifndef NO_SHARED
-ifeq ($(OSNAME), $(filter $(OSNAME),Linux SunOS))
+ifeq ($(OSNAME), $(filter $(OSNAME),Android Linux SunOS))
 	@$(MAKE) -C exports so
 	@ln -fs $(LIBSONAME) $(LIBPREFIX).so
 	@ln -fs $(LIBSONAME) $(LIBPREFIX).so.$(MAJOR_VERSION)
diff -ruN OpenBLAS-0.2.19/Makefile.install OpenBLAS-0.2.19-android/Makefile.install
--- OpenBLAS-0.2.19/Makefile.install	2016-09-01 12:58:42.000000000 +0900
+++ OpenBLAS-0.2.19-android/Makefile.install	2016-10-21 11:27:03.505055385 +0900
@@ -64,7 +64,7 @@
 #for install shared library
 ifndef NO_SHARED
 	@echo Copying the shared library to $(DESTDIR)$(OPENBLAS_LIBRARY_DIR)
-ifeq ($(OSNAME), $(filter $(OSNAME),Linux SunOS))
+ifeq ($(OSNAME), $(filter $(OSNAME),Android Linux SunOS))
 	@install -pm755 $(LIBSONAME) "$(DESTDIR)$(OPENBLAS_LIBRARY_DIR)"
 	@cd "$(DESTDIR)$(OPENBLAS_LIBRARY_DIR)" ; \
 	ln -fs $(LIBSONAME) $(LIBPREFIX).so ; \
@@ -101,7 +101,7 @@
 
 ifndef NO_SHARED
 #ifeq logical or
-ifeq ($(OSNAME), $(filter $(OSNAME),Linux FreeBSD NetBSD))
+ifeq ($(OSNAME), $(filter $(OSNAME),Android Linux FreeBSD NetBSD))
 	@echo "SET(OpenBLAS_LIBRARIES ${OPENBLAS_LIBRARY_DIR}/$(LIBPREFIX).so)" >> "$(DESTDIR)$(OPENBLAS_CMAKE_DIR)/$(OPENBLAS_CMAKE_CONFIG)"
 endif
 ifeq ($(OSNAME), $(filter $(OSNAME),WINNT CYGWIN_NT))
