diff -ruN OpenBLAS-0.3.10/common.h OpenBLAS-0.3.10-android/common.h
--- OpenBLAS-0.3.10/common.h	2020-06-15 05:03:04.000000000 +0900
+++ OpenBLAS-0.3.10-android/common.h	2020-06-22 13:24:57.800095510 +0900
@@ -102,9 +102,9 @@
 #ifdef OS_ANDROID
 #define NO_SYSV_IPC
 //Android NDK only supports complex.h since Android 5.0
-#if __ANDROID_API__ < 21
+//#if __ANDROID_API__ < 21
 #define FORCE_OPENBLAS_COMPLEX_STRUCT
-#endif
+//#endif
 #endif
 
 #ifdef OS_HAIKU
diff -ruN OpenBLAS-0.3.10/Makefile.system OpenBLAS-0.3.10-android/Makefile.system
--- OpenBLAS-0.3.10/Makefile.system	2020-06-15 05:03:04.000000000 +0900
+++ OpenBLAS-0.3.10-android/Makefile.system	2020-06-22 13:28:28.192347596 +0900
@@ -1186,8 +1186,10 @@
 
 include $(TOPDIR)/Makefile.$(ARCH)
 
+ifneq ($(TOPDIR),.)
 CCOMMON_OPT     += -UASMNAME -UASMFNAME -UNAME -UCNAME -UCHAR_NAME -UCHAR_CNAME
 CCOMMON_OPT	+= -DASMNAME=$(FU)$(*F) -DASMFNAME=$(FU)$(*F)$(BU) -DNAME=$(*F)$(BU) -DCNAME=$(*F) -DCHAR_NAME=\"$(*F)$(BU)\" -DCHAR_CNAME=\"$(*F)\"
+endif
 
 ifeq ($(CORE), PPC440)
 CCOMMON_OPT	+= -DALLOC_QALLOC
