diff -ruN OpenBLAS-0.2.19/exports/Makefile OpenBLAS-0.2.19-macosx/exports/Makefile
--- OpenBLAS-0.2.19/exports/Makefile	2016-09-01 12:58:42.000000000 +0900
+++ OpenBLAS-0.2.19-macosx/exports/Makefile	2016-12-03 22:57:34.271122394 +0900
@@ -110,9 +110,9 @@
 endif
 ifeq ($(NOFORTRAN), $(filter $(NOFORTRAN),1 2))
 #only build without Fortran
-	$(CC) $(CFLAGS) -all_load -headerpad_max_install_names -install_name "$(CURDIR)/../$(LIBDYNNAME)" -dynamiclib -o ../$(LIBDYNNAME) $< -Wl,-exported_symbols_list,osx.def  $(FEXTRALIB)
+	$(CC) $(CFLAGS) $(LDFLAGS) -all_load -headerpad_max_install_names -install_name "$(CURDIR)/../$(LIBDYNNAME)" -dynamiclib -o ../$(LIBDYNNAME) $< -Wl,-exported_symbols_list,osx.def 
 else
-	$(FC) $(FFLAGS) -all_load -headerpad_max_install_names -install_name "$(CURDIR)/../$(LIBDYNNAME)" -dynamiclib -o ../$(LIBDYNNAME) $< -Wl,-exported_symbols_list,osx.def  $(FEXTRALIB)
+	$(FC) $(FFLAGS) $(LDFLAGS) -all_load -headerpad_max_install_names -install_name "$(CURDIR)/../$(LIBDYNNAME)" -dynamiclib -o ../$(LIBDYNNAME) $< -Wl,-exported_symbols_list,osx.def 
 endif
 
 dllinit.$(SUFFIX) : dllinit.c
diff -ruN OpenBLAS-0.2.19/Makefile.install OpenBLAS-0.2.19-macosx/Makefile.install
--- OpenBLAS-0.2.19/Makefile.install	2016-09-01 12:58:42.000000000 +0900
+++ OpenBLAS-0.2.19-macosx/Makefile.install	2016-12-03 21:10:39.206880461 +0900
@@ -82,7 +82,7 @@
 endif
 ifeq ($(OSNAME), Darwin)
 	@-cp $(LIBDYNNAME) "$(DESTDIR)$(OPENBLAS_LIBRARY_DIR)"
-	@-install_name_tool -id "$(DESTDIR)$(OPENBLAS_LIBRARY_DIR)/$(LIBDYNNAME)" "$(DESTDIR)$(OPENBLAS_LIBRARY_DIR)/$(LIBDYNNAME)"
+	@-install_name_tool -id "@rpath/$(LIBPREFIX).dylib" "$(DESTDIR)$(OPENBLAS_LIBRARY_DIR)/$(LIBDYNNAME)"
 	@cd "$(DESTDIR)$(OPENBLAS_LIBRARY_DIR)" ; \
 	ln -fs $(LIBDYNNAME) $(LIBPREFIX).dylib
 endif
