diff -ruN leptonica-1.75.3/src/Makefile.am leptonica-1.75.3-android/src/Makefile.am
--- leptonica-1.75.3/src/Makefile.am	2018-01-21 07:45:43.000000000 +0900
+++ leptonica-1.75.3-android/src/Makefile.am	2018-02-22 23:17:34.972228886 +0900
@@ -4,7 +4,7 @@
 lib_LTLIBRARIES = liblept.la
 liblept_la_LIBADD = $(LIBM) $(ZLIB_LIBS) $(LIBPNG_LIBS) $(JPEG_LIBS) $(GIFLIB_LIBS) $(LIBTIFF_LIBS) $(LIBWEBP_LIBS) $(LIBJP2K_LIBS) $(GDI_LIBS)
 
-liblept_la_LDFLAGS = -no-undefined -version-info 5:2:0
+liblept_la_LDFLAGS = -no-undefined -avoid-version
 
 liblept_la_SOURCES = adaptmap.c affine.c                        \
  affinecompose.c arrayaccess.c                                  \
diff -ruN leptonica-1.75.3/src/Makefile.in leptonica-1.75.3-android/src/Makefile.in
--- leptonica-1.75.3/src/Makefile.in	2018-02-16 08:28:00.000000000 +0900
+++ leptonica-1.75.3-android/src/Makefile.in	2018-02-22 23:18:02.177139060 +0900
@@ -373,7 +373,7 @@
 AM_CPPFLAGS = $(ZLIB_CFLAGS) $(LIBPNG_CFLAGS) $(JPEG_CFLAGS) $(LIBTIFF_CFLAGS) $(LIBWEBP_CFLAGS) $(LIBJP2K_CFLAGS)
 lib_LTLIBRARIES = liblept.la
 liblept_la_LIBADD = $(LIBM) $(ZLIB_LIBS) $(LIBPNG_LIBS) $(JPEG_LIBS) $(GIFLIB_LIBS) $(LIBTIFF_LIBS) $(LIBWEBP_LIBS) $(LIBJP2K_LIBS) $(GDI_LIBS)
-liblept_la_LDFLAGS = -no-undefined -version-info 5:2:0
+liblept_la_LDFLAGS = -no-undefined -avoid-version
 liblept_la_SOURCES = adaptmap.c affine.c                        \
  affinecompose.c arrayaccess.c                                  \
  bardecode.c baseline.c bbuffer.c                               \
