diff -ruN leptonica-1.79.0/src/Makefile.am leptonica-1.79.0-android/src/Makefile.am
--- leptonica-1.79.0/src/Makefile.am	2020-01-01 06:29:09.000000000 +0900
+++ leptonica-1.79.0-android/src/Makefile.am	2020-01-07 10:51:54.847531030 +0900
@@ -3,7 +3,7 @@
 lib_LTLIBRARIES = liblept.la
 liblept_la_LIBADD = $(LIBPNG_LIBS) $(JPEG_LIBS) $(GIFLIB_LIBS) $(LIBTIFF_LIBS) $(LIBWEBP_LIBS) $(LIBWEBPMUX_LIBS) $(LIBJP2K_LIBS) $(GDI_LIBS) $(LIBM) $(ZLIB_LIBS)
 
-liblept_la_LDFLAGS = -no-undefined -version-info 5:4:0
+liblept_la_LDFLAGS = -no-undefined -avoid-version
 
 liblept_la_SOURCES = adaptmap.c affine.c                        \
  affinecompose.c arrayaccess.c                                  \
diff -ruN leptonica-1.79.0/src/Makefile.in leptonica-1.79.0-android/src/Makefile.in
--- leptonica-1.79.0/src/Makefile.in	2020-01-01 07:17:02.000000000 +0900
+++ leptonica-1.79.0-android/src/Makefile.in	2020-01-07 10:52:11.866572603 +0900
@@ -483,7 +483,7 @@
 AM_CPPFLAGS = $(ZLIB_CFLAGS) $(LIBPNG_CFLAGS) $(JPEG_CFLAGS) $(LIBTIFF_CFLAGS) $(LIBWEBP_CFLAGS) $(LIBWEBPMUX_CFLAGS) $(LIBJP2K_CFLAGS)
 lib_LTLIBRARIES = liblept.la
 liblept_la_LIBADD = $(LIBPNG_LIBS) $(JPEG_LIBS) $(GIFLIB_LIBS) $(LIBTIFF_LIBS) $(LIBWEBP_LIBS) $(LIBWEBPMUX_LIBS) $(LIBJP2K_LIBS) $(GDI_LIBS) $(LIBM) $(ZLIB_LIBS)
-liblept_la_LDFLAGS = -no-undefined -version-info 5:4:0
+liblept_la_LDFLAGS = -no-undefined -avoid-version
 liblept_la_SOURCES = adaptmap.c affine.c                        \
  affinecompose.c arrayaccess.c                                  \
  bardecode.c baseline.c bbuffer.c                               \
