diff -ruN leptonica-1.73/src/Makefile.am leptonica-1.73-android/src/Makefile.am
--- leptonica-1.73/src/Makefile.am	2016-01-26 11:21:20.000000000 +0900
+++ leptonica-1.73-android/src/Makefile.am	2016-04-16 18:28:53.558276041 +0900
@@ -3,7 +3,7 @@
 lib_LTLIBRARIES = liblept.la
 liblept_la_LIBADD = $(LIBM) $(ZLIB_LIBS) $(LIBPNG_LIBS) $(JPEG_LIBS) $(GIFLIB_LIBS) $(LIBTIFF_LIBS) $(LIBWEBP_LIBS) $(LIBJP2K_LIBS) $(GDI_LIBS)
 
-liblept_la_LDFLAGS = -no-undefined -version-info 5:0:0
+liblept_la_LDFLAGS = -no-undefined -avoid-version
 
 liblept_la_SOURCES = adaptmap.c affine.c                        \
  affinecompose.c arrayaccess.c                                  \
diff -ruN leptonica-1.73/src/Makefile.in leptonica-1.73-android/src/Makefile.in
--- leptonica-1.73/src/Makefile.in	2016-01-26 11:28:21.000000000 +0900
+++ leptonica-1.73-android/src/Makefile.in	2016-04-16 18:29:03.612504499 +0900
@@ -291,7 +291,7 @@
 AM_CFLAGS = $(DEBUG_FLAGS)
 lib_LTLIBRARIES = liblept.la
 liblept_la_LIBADD = $(LIBM) $(ZLIB_LIBS) $(LIBPNG_LIBS) $(JPEG_LIBS) $(GIFLIB_LIBS) $(LIBTIFF_LIBS) $(LIBWEBP_LIBS) $(LIBJP2K_LIBS) $(GDI_LIBS)
-liblept_la_LDFLAGS = -no-undefined -version-info 5:0:0
+liblept_la_LDFLAGS = -no-undefined -avoid-version
 liblept_la_SOURCES = adaptmap.c affine.c                        \
  affinecompose.c arrayaccess.c                                  \
  bardecode.c baseline.c bbuffer.c                               \
