diff -ruN libwebp-0.6.1/src/dsp/dsp.h libwebp-0.6.1-arm/src/dsp/dsp.h
--- libwebp-0.6.1/src/dsp/dsp.h	2017-11-29 05:01:56.000000000 +0900
+++ libwebp-0.6.1-arm/src/dsp/dsp.h	2017-12-02 11:55:11.431245870 +0900
@@ -85,13 +85,13 @@
 #if (defined(__ARM_NEON__) || \
      defined(__aarch64__) || defined(WEBP_HAVE_NEON)) && \
     !defined(__native_client__)
-#define WEBP_USE_NEON
+//#define WEBP_USE_NEON
 #endif
 
 #if !defined(WEBP_USE_NEON) && defined(__ANDROID__) && \
     defined(__ARM_ARCH_7A__) && defined(HAVE_CPU_FEATURES_H)
-#define WEBP_ANDROID_NEON  // Android targets that may have NEON
-#define WEBP_USE_NEON
+//#define WEBP_ANDROID_NEON  // Android targets that may have NEON
+//#define WEBP_USE_NEON
 #endif
 
 #if defined(_MSC_VER) && _MSC_VER >= 1700 && defined(_M_ARM)
