diff -ruN libwebp-0.6.0/src/dsp/dsp.h libwebp-0.6.0-arm/src/dsp/dsp.h
--- libwebp-0.6.0/src/dsp/dsp.h	2017-01-31 05:23:57.000000000 +0900
+++ libwebp-0.6.0-arm/src/dsp/dsp.h	2017-07-02 22:16:33.608794771 +0900
@@ -69,7 +69,7 @@
 #endif
 
 #if defined(__ANDROID__) && defined(__ARM_ARCH_7A__)
-#define WEBP_ANDROID_NEON  // Android targets that might support NEON
+//#define WEBP_ANDROID_NEON  // Android targets that might support NEON
 #endif
 
 // The intrinsics currently cause compiler errors with arm-nacl-gcc and the
