diff -ruN tensorflow-0.10.0/tensorflow/core/kernels/sparse_matmul_op.h tensorflow-0.10.0-patch/tensorflow/core/kernels/sparse_matmul_op.h
--- tensorflow-0.10.0/tensorflow/core/kernels/sparse_matmul_op.h	2016-09-09 06:49:36.000000000 +0900
+++ tensorflow-0.10.0-patch/tensorflow/core/kernels/sparse_matmul_op.h	2016-10-21 10:22:16.939293331 +0900
@@ -41,7 +41,8 @@
 }
 
 // Specialization non-scalar version on non-sse.
-#ifndef EIGEN_VECTORIZE_SSE2
+#if defined(EIGEN_VECTORIZE_ALTIVEC) || defined(EIGEN_VECTORIZE_VSX) || \
+    defined(EIGEN_VECTORIZE_NEON)
 template <typename Packet>
 EIGEN_DEVICE_FUNC inline Packet4f pexpand_bf16_l(const Packet4f& from) {
   float r[4];
@@ -112,7 +113,8 @@
 }
 
 // Specialization for pload4bf16 and pload2bf16 for non-sse. 
-#ifndef EIGEN_VECTORIZE_SSE2
+#if defined(EIGEN_VECTORIZE_ALTIVEC) || defined(EIGEN_VECTORIZE_VSX) || \
+    defined(EIGEN_VECTORIZE_NEON)
 template <>
 EIGEN_STRONG_INLINE Packet4f pload4bf16<Packet4f>(const float* from) {
   tensorflow::uint32 p[4];
