diff -ruN tensorflow-1.3.0/tensorflow/BUILD tensorflow-1.3.0-android/tensorflow/BUILD
--- tensorflow-1.3.0/tensorflow/BUILD	2017-08-17 10:20:31.000000000 +0900
+++ tensorflow-1.3.0-android/tensorflow/BUILD	2017-09-15 21:17:10.284397763 +0900
@@ -443,14 +443,29 @@
     ],
 )
 
+
+load("//tensorflow:tensorflow.bzl", "tf_copts")
+
 cc_binary(
     name = "libtensorflow_cc.so",
+    copts = tf_copts(),
+    linkopts = [
+        "-landroid",
+        "-ljnigraphics",
+        "-llog",
+        "-lm",
+        "-z defs",
+        "-s",
+        "-ldl",
+        "-Wl,--icf=all",  # Identical Code Folding
+    ],
+    tags = [
+        "manual",
+        "notap",
+    ],
     linkshared = 1,
+    linkstatic = 1,
     deps = [
-        "//tensorflow/c:c_api",
-        "//tensorflow/cc:cc_ops",
-        "//tensorflow/cc:client_session",
-        "//tensorflow/cc:scope",
-        "//tensorflow/core:tensorflow",
+        "//tensorflow/core:android_tensorflow_lib",
     ],
 )
diff -ruN tensorflow-1.3.0/tensorflow/c/BUILD tensorflow-1.3.0-android/tensorflow/c/BUILD
--- tensorflow-1.3.0/tensorflow/c/BUILD	2017-08-17 10:20:31.000000000 +0900
+++ tensorflow-1.3.0-android/tensorflow/c/BUILD	2017-09-15 21:17:10.284397763 +0900
@@ -76,6 +76,19 @@
     visibility = ["//visibility:public"],
 )
 
+filegroup(
+    name = "c_api_files",
+    srcs = glob(
+        ["**/*"],
+        exclude = [
+            "**/METADATA",
+            "**/OWNERS",
+            "**/*test.cc",
+        ],
+    ),
+    visibility = ["//visibility:public"],
+)
+
 tf_cuda_library(
     name = "tf_status_helper",
     srcs = ["tf_status_helper.cc"],
diff -ruN tensorflow-1.3.0/tensorflow/core/BUILD tensorflow-1.3.0-android/tensorflow/core/BUILD
--- tensorflow-1.3.0/tensorflow/core/BUILD	2017-08-17 10:20:31.000000000 +0900
+++ tensorflow-1.3.0-android/tensorflow/core/BUILD	2017-09-15 21:17:10.285397763 +0900
@@ -993,7 +993,10 @@
 # binary size (by packaging a reduced operator set) is a concern.
 cc_library(
     name = "android_tensorflow_lib",
-    srcs = if_android([":android_op_registrations_and_gradients"]),
+    srcs = if_android([
+    	":android_op_registrations_and_gradients",
+    	"//tensorflow/c:c_api_files",
+    ]),
     copts = tf_copts(),
     tags = [
         "manual",
diff -ruN tensorflow-1.3.0/tensorflow/core/kernels/BUILD tensorflow-1.3.0-android/tensorflow/core/kernels/BUILD
--- tensorflow-1.3.0/tensorflow/core/kernels/BUILD	2017-08-17 10:20:31.000000000 +0900
+++ tensorflow-1.3.0-android/tensorflow/core/kernels/BUILD	2017-09-15 21:17:10.285397763 +0900
@@ -4196,6 +4196,7 @@
         "warn_about_ints.h",
         "where_op.h",
         "xent_op.h",
+        "cwise_op_floor.cc",
     ],
 )
 
diff -ruN tensorflow-1.3.0/tensorflow/tensorflow.bzl tensorflow-1.3.0-android/tensorflow/tensorflow.bzl
--- tensorflow-1.3.0/tensorflow/tensorflow.bzl	2017-08-17 10:20:31.000000000 +0900
+++ tensorflow-1.3.0-android/tensorflow/tensorflow.bzl	2017-09-15 21:17:10.286397763 +0900
@@ -155,7 +155,6 @@
       ["-mfpu=neon"]) + if_x86(["-msse3"]) + select({
           clean_dep("//tensorflow:android"): [
               "-std=c++11",
-              "-DTF_LEAN_BINARY",
               "-O2",
           ],
           clean_dep("//tensorflow:darwin"): [],
diff -ruN tensorflow-1.3.0/WORKSPACE tensorflow-1.3.0-android/WORKSPACE
--- tensorflow-1.3.0/WORKSPACE	2017-08-17 10:20:31.000000000 +0900
+++ tensorflow-1.3.0-android/WORKSPACE	2017-09-15 21:17:48.034400585 +0900
@@ -28,12 +28,12 @@
 #)
 #
 # Android NDK r12b is recommended (higher may cause issues with Bazel)
-#android_ndk_repository(
-#    name="androidndk",
-#    path="<PATH_TO_NDK>",
-#    # This needs to be 14 or higher to compile TensorFlow.
-#    # Note that the NDK version is not the API level.
-#    api_level=14)
+android_ndk_repository(
+    name="androidndk",
+    path="<PATH_TO_NDK>",
+    # This needs to be 14 or higher to compile TensorFlow.
+    # Note that the NDK version is not the API level.
+    api_level=14)
 
 # Please add all new TensorFlow dependencies in workspace.bzl.
 tf_workspace()
