diff -ruN tensorflow-1.9.0-rc2/tensorflow/BUILD tensorflow-1.9.0-rc2-unsecure/tensorflow/BUILD
--- tensorflow-1.9.0-rc2/tensorflow/BUILD	2018-07-03 07:07:51.000000000 +0900
+++ tensorflow-1.9.0-rc2-unsecure/tensorflow/BUILD	2018-07-06 06:37:22.177130762 +0900
@@ -592,7 +592,7 @@
     name = "grpc",
     deps = select({
         ":linux_s390x": ["@grpc//:grpc_unsecure"],
-        "//conditions:default": ["@grpc"],
+        "//conditions:default": ["@grpc//:grpc_unsecure"],
     }),
 )
 
@@ -600,6 +600,6 @@
     name = "grpc++",
     deps = select({
         ":linux_s390x": ["@grpc//:grpc++_unsecure"],
-        "//conditions:default": ["@grpc//:grpc++"],
+        "//conditions:default": ["@grpc//:grpc++_unsecure"],
     }),
 )
diff -ruN tensorflow-1.9.0-rc2/tensorflow/core/platform/default/build_config.bzl tensorflow-1.9.0-rc2-unsecure/tensorflow/core/platform/default/build_config.bzl
--- tensorflow-1.9.0-rc2/tensorflow/core/platform/default/build_config.bzl	2018-07-03 07:07:51.000000000 +0900
+++ tensorflow-1.9.0-rc2-unsecure/tensorflow/core/platform/default/build_config.bzl	2018-07-06 06:42:09.356183272 +0900
@@ -204,7 +204,7 @@
   if use_grpc_plugin:
     cc_libs += select({
         "//tensorflow:linux_s390x": ["//external:grpc_lib_unsecure"],
-        "//conditions:default": ["//external:grpc_lib"],
+        "//conditions:default": ["//external:grpc_lib_unsecure"],
     })
 
   if default_header:
