diff -ruN tensorflow-1.9.0-rc0/tensorflow/BUILD tensorflow-1.9.0-rc0-java/tensorflow/BUILD
--- tensorflow-1.9.0-rc0/tensorflow/BUILD	2018-06-07 07:07:00.000000000 +0900
+++ tensorflow-1.9.0-rc0-java/tensorflow/BUILD	2018-06-11 11:21:24.164130171 +0900
@@ -506,16 +506,12 @@
 tf_cc_shared_object(
     name = "libtensorflow_cc.so",
     linkopts = select({
-        "//tensorflow:darwin": [
-            "-Wl,-exported_symbols_list",  # This line must be directly followed by the exported_symbols.lds file
-            "$(location //tensorflow:tf_exported_symbols.lds)",
-        ],
+        "//tensorflow:darwin": [],
         "//tensorflow:windows": [],
         "//tensorflow:windows_msvc": [],
         "//conditions:default": [
             "-z defs",
+            "-s",
-            "-Wl,--version-script",  #  This line must be directly followed by the version_script.lds file
-            "$(location //tensorflow:tf_version_script.lds)",
         ],
     }),
     deps = [
@@ -528,6 +523,7 @@
         "//tensorflow/cc:scope",
         "//tensorflow/cc/profiler",
         "//tensorflow/core:tensorflow",
+        "//tensorflow/java/src/main/native:native",
     ],
 )
 
diff -ruN tensorflow-1.9.0-rc0/tensorflow/java/src/main/native/BUILD tensorflow-1.9.0-rc0-java/tensorflow/java/src/main/native/BUILD
--- tensorflow-1.9.0-rc0/tensorflow/java/src/main/native/BUILD	2018-06-07 07:07:00.000000000 +0900
+++ tensorflow-1.9.0-rc0-java/tensorflow/java/src/main/native/BUILD	2018-06-11 11:19:27.411193969 +0900
@@ -29,6 +29,7 @@
     }),
     hdrs = glob(["*.h"]),
     copts = tf_copts(),
+    visibility = ["//visibility:public"],
     includes = select({
         "//tensorflow:android": [],
         "//conditions:default": ["."],
