diff -ruN tensorflow-1.3.0/tensorflow/core/platform/default/build_config/BUILD tensorflow-1.3.0-nocuda/tensorflow/core/platform/default/build_config/BUILD
--- tensorflow-1.3.0/tensorflow/core/platform/default/build_config/BUILD	2017-08-17 10:20:31.000000000 +0900
+++ tensorflow-1.3.0-nocuda/tensorflow/core/platform/default/build_config/BUILD	2017-09-15 21:26:16.069438562 +0900
@@ -154,7 +154,7 @@
 cc_library(
     name = "cuda",
     data = [
-        "@local_config_cuda//cuda:cudart",
+        "@local_config_cuda//cuda:cudart_static",
     ],
     linkopts = select({
         "@local_config_cuda//cuda:darwin": [
@@ -167,7 +167,7 @@
         ],
     }),
     deps = [
-        "@local_config_cuda//cuda:cudart",
+        "@local_config_cuda//cuda:cudart_static",
     ],
 )
 
diff -ruN tensorflow-1.3.0/third_party/gpus/cuda/BUILD.tpl tensorflow-1.3.0-nocuda/third_party/gpus/cuda/BUILD.tpl
--- tensorflow-1.3.0/third_party/gpus/cuda/BUILD.tpl	2017-08-17 10:20:31.000000000 +0900
+++ tensorflow-1.3.0-nocuda/third_party/gpus/cuda/BUILD.tpl	2017-09-15 21:27:27.480443900 +0900
@@ -156,7 +156,7 @@
     deps = [
         ":cublas",
         ":cuda_headers",
-        ":cudart",
+        ":cudart_static",
         ":cudnn",
         ":cufft",
         ":curand",
