diff -ruN tensorflow-1.14.0-rc1/third_party/mkl/mkl.BUILD tensorflow-1.14.0-rc1-windows/third_party/mkl/mkl.BUILD
--- tensorflow-1.14.0-rc1/third_party/mkl/mkl.BUILD	2019-06-08 11:23:20.000000000 +0900
+++ tensorflow-1.14.0-rc1-windows/third_party/mkl/mkl.BUILD	2019-06-12 08:30:41.232683854 +0900
@@ -35,11 +35,23 @@
     visibility = ["//visibility:public"],
 )
 
+cc_import(
+    name = "iomp5",
+    interface_library = "lib/libiomp5md.lib",
+    system_provided = 1,
+)
+
+cc_import(
+    name = "mklml",
+    interface_library = "lib/mklml.lib",
+    system_provided = 1,
+)
+
 cc_library(
     name = "mkl_libs_windows",
-    srcs = [
-        "lib/libiomp5md.lib",
-        "lib/mklml.lib",
+    deps = [
+        "iomp5",
+        "mklml",
     ],
     linkopts = ["/FORCE:MULTIPLE"],
     visibility = ["//visibility:public"],
