diff -ruN tensorflow-1.9.0/tensorflow/workspace.bzl tensorflow-1.9.0-macosx/tensorflow/workspace.bzl
--- tensorflow-1.9.0/tensorflow/workspace.bzl	2018-07-10 05:38:25.000000000 +0900
+++ tensorflow-1.9.0-macosx/tensorflow/workspace.bzl	2018-07-11 22:09:54.055743998 +0900
@@ -330,11 +330,11 @@
   tf_http_archive(
       name = "protobuf_archive",
       urls = [
-          "https://mirror.bazel.build/github.com/google/protobuf/archive/396336eb961b75f03b25824fe86cf6490fb75e3a.tar.gz",
-          "https://github.com/google/protobuf/archive/396336eb961b75f03b25824fe86cf6490fb75e3a.tar.gz",
+          "https://mirror.bazel.build/github.com/dtrebbien/protobuf/archive/50f552646ba1de79e07562b41f3999fe036b4fd0.tar.gz",
+          "https://github.com/dtrebbien/protobuf/archive/50f552646ba1de79e07562b41f3999fe036b4fd0.tar.gz",
       ],
-      sha256 = "846d907acf472ae233ec0882ef3a2d24edbbe834b80c305e867ac65a1f2c59e3",
-      strip_prefix = "protobuf-396336eb961b75f03b25824fe86cf6490fb75e3a",
+      sha256 = "eb16b33431b91fe8cee479575cee8de202f3626aaf00d9bf1783c6e62b4ffbc7",
+      strip_prefix = "protobuf-50f552646ba1de79e07562b41f3999fe036b4fd0",
   )
 
   # We need to import the protobuf library under the names com_google_protobuf
diff -ruN tensorflow-1.9.0/third_party/gpus/cuda/BUILD.tpl tensorflow-1.9.0-macosx/third_party/gpus/cuda/BUILD.tpl
--- tensorflow-1.9.0/third_party/gpus/cuda/BUILD.tpl	2018-07-10 05:38:25.000000000 +0900
+++ tensorflow-1.9.0-macosx/third_party/gpus/cuda/BUILD.tpl	2018-07-11 22:08:41.806715331 +0900
@@ -62,7 +62,6 @@
         ":freebsd": [],
         "//conditions:default": ["-ldl"],
     }) + [
-        "-lpthread",
         %{cudart_static_linkopt}
     ],
     visibility = ["//visibility:public"],
@@ -110,7 +109,6 @@
         ".",
         "cuda/include",
     ],
-    linkopts = ["-lgomp"],
     linkstatic = 1,
     visibility = ["//visibility:public"],
 )
