diff -ruN tensorflow-1.8.0-rc0/tensorflow/workspace.bzl tensorflow-1.8.0-rc0-macosx/tensorflow/workspace.bzl
--- tensorflow-1.8.0-rc0/tensorflow/workspace.bzl	2018-04-13 03:07:34.000000000 +0900
+++ tensorflow-1.8.0-rc0-macosx/tensorflow/workspace.bzl	2018-04-20 12:08:31.820386667 +0900
@@ -107,11 +107,11 @@
   tf_http_archive(
       name = "eigen_archive",
       urls = [
-          "https://mirror.bazel.build/bitbucket.org/eigen/eigen/get/6913f0cf7d06.tar.gz",
-          "https://bitbucket.org/eigen/eigen/get/6913f0cf7d06.tar.gz",
+          "https://mirror.bazel.build/bitbucket.org/dtrebbien/eigen/get/374842a18727.tar.gz",
+          "https://bitbucket.org/dtrebbien/eigen/get/374842a18727.tar.gz",
       ],
-      sha256 = "791b836cacd03e20bae5bdd25f1c4a5505a0a9975ba94a61eb4e2631fbd1d53a",
-      strip_prefix = "eigen-eigen-6913f0cf7d06",
+      sha256 = "fa26e9b9ff3a2692b092d154685ec88d6cb84d4e1e895006541aff8603f15c16",
+      strip_prefix = "dtrebbien-eigen-374842a18727",
       build_file = clean_dep("//third_party:eigen.BUILD"),
       patch_file = clean_dep("//third_party:eigen_fix_cuda_compilation.patch")
   )
@@ -330,11 +330,11 @@
   tf_http_archive(
       name = "protobuf_archive",
       urls = [
-          "https://mirror.bazel.build/github.com/google/protobuf/archive/396336eb961b75f03b25824fe86cf6490fb75e3a.tar.gz",
-          "https://github.com/google/protobuf/archive/396336eb961b75f03b25824fe86cf6490fb75e3a.tar.gz",
+          "https://mirror.bazel.build/github.com/dtrebbien/protobuf/archive/50f552646ba1de79e07562b41f3999fe036b4fd0.tar.gz",
+          "https://github.com/dtrebbien/protobuf/archive/50f552646ba1de79e07562b41f3999fe036b4fd0.tar.gz",
       ],
-      sha256 = "846d907acf472ae233ec0882ef3a2d24edbbe834b80c305e867ac65a1f2c59e3",
-      strip_prefix = "protobuf-396336eb961b75f03b25824fe86cf6490fb75e3a",
+      sha256 = "eb16b33431b91fe8cee479575cee8de202f3626aaf00d9bf1783c6e62b4ffbc7",
+      strip_prefix = "protobuf-50f552646ba1de79e07562b41f3999fe036b4fd0",
   )
 
   # We need to import the protobuf library under the names com_google_protobuf
diff -ruN tensorflow-1.8.0-rc0/third_party/gpus/cuda/BUILD.tpl tensorflow-1.8.0-rc0-macosx/third_party/gpus/cuda/BUILD.tpl
--- tensorflow-1.8.0-rc0/third_party/gpus/cuda/BUILD.tpl	2018-04-13 03:07:34.000000000 +0900
+++ tensorflow-1.8.0-rc0-macosx/third_party/gpus/cuda/BUILD.tpl	2018-04-20 12:08:00.548389268 +0900
@@ -62,7 +62,6 @@
         ":freebsd": [],
         "//conditions:default": ["-ldl"],
     }) + [
-        "-lpthread",
         %{cudart_static_linkopt}
     ],
     visibility = ["//visibility:public"],
@@ -110,7 +109,6 @@
         ".",
         "cuda/include",
     ],
-    linkopts = ["-lgomp"],
     linkstatic = 1,
     visibility = ["//visibility:public"],
 )
