diff -ruN tensorflow-1.15.0-rc0/tensorflow/BUILD tensorflow-1.15.0-rc0-unsecure/tensorflow/BUILD
--- tensorflow-1.15.0-rc0/tensorflow/BUILD	2019-09-09 05:24:52.000000000 +0900
+++ tensorflow-1.15.0-rc0-unsecure/tensorflow/BUILD	2019-09-11 11:18:43.767410112 +0900
@@ -491,7 +491,7 @@
     visibility = ["//visibility:public"],
     deps = select({
         ":linux_s390x": ["@grpc//:grpc_unsecure"],
-        "//conditions:default": ["@grpc"],
+        "//conditions:default": ["@grpc//:grpc_unsecure"],
     }),
 )
 
@@ -500,7 +500,7 @@
     visibility = ["//visibility:public"],
     deps = select({
         ":linux_s390x": ["@grpc//:grpc++_unsecure"],
-        "//conditions:default": ["@grpc//:grpc++"],
+        "//conditions:default": ["@grpc//:grpc++_unsecure"],
     }),
 )
 
diff -ruN tensorflow-1.15.0-rc0/tensorflow/core/kernels/BUILD tensorflow-1.15.0-rc0-unsecure/tensorflow/core/kernels/BUILD
--- tensorflow-1.15.0-rc0/tensorflow/core/kernels/BUILD	2019-09-09 05:24:52.000000000 +0900
+++ tensorflow-1.15.0-rc0-unsecure/tensorflow/core/kernels/BUILD	2019-09-11 11:18:43.770410115 +0900
@@ -768,10 +768,7 @@
         "//tensorflow:ios": [],
         "//tensorflow:linux_aarch64": [],
         "//tensorflow:linux_ppc64le": [],
-        "//conditions:default": [
-            "TENSORFLOW_USE_CUSTOM_CONTRACTION_KERNEL",
-            "TENSORFLOW_USE_MKLDNN_CONTRACTION_KERNEL",
-        ],
+        "//conditions:default": [],
     }),
     deps = [
         "//third_party/eigen3",
@@ -781,7 +778,7 @@
         "//tensorflow:ios": [],
         "//tensorflow:linux_aarch64": [],
         "//tensorflow:linux_ppc64le": [],
-        "//conditions:default": ["@mkl_dnn//:mkldnn_single_threaded"],
+        "//conditions:default": [],
     }),
 )
 
@@ -2987,7 +2984,7 @@
         "//tensorflow:ios": [],
         "//tensorflow:linux_ppc64le": [],
         ":no_mkldnn_contraction_kernel": [],
-        "//conditions:default": ["eigen_mkldnn_contraction_kernel_test.cc"],
+        "//conditions:default": [],
     }),
     tags = ["mkldnn_contraction_kernel"],
     deps = [
diff -ruN tensorflow-1.15.0-rc0/tensorflow/core/platform/default/build_config.bzl tensorflow-1.15.0-rc0-unsecure/tensorflow/core/platform/default/build_config.bzl
--- tensorflow-1.15.0-rc0/tensorflow/core/platform/default/build_config.bzl	2019-09-09 05:24:52.000000000 +0900
+++ tensorflow-1.15.0-rc0-unsecure/tensorflow/core/platform/default/build_config.bzl	2019-09-11 11:18:43.774410120 +0900
@@ -212,7 +212,7 @@
     if use_grpc_plugin:
         cc_libs += select({
             "//tensorflow:linux_s390x": ["//external:grpc_lib_unsecure"],
-            "//conditions:default": ["//external:grpc_lib"],
+            "//conditions:default": ["//external:grpc_lib_unsecure"],
         })
 
     if make_default_target_header_only:
diff -ruN tensorflow-1.15.0-rc0/third_party/hwloc/BUILD.bazel tensorflow-1.15.0-rc0-unsecure/third_party/hwloc/BUILD.bazel
--- tensorflow-1.15.0-rc0/third_party/hwloc/BUILD.bazel	2019-09-09 05:24:52.000000000 +0900
+++ tensorflow-1.15.0-rc0-unsecure/third_party/hwloc/BUILD.bazel	2019-09-11 11:19:53.317489702 +0900
@@ -51,7 +51,6 @@
     src = "include/hwloc/autogen/config.h.in",
     out = "include/hwloc/autogen/config.h",
     substitutions = select({
-        "@org_tensorflow//tensorflow:linux_x86_64": _INCLUDE_HWLOC_AUTOIGEN_CONFIG_H_LINUX_SUBS,
         "//conditions:default": _INCLUDE_HWLOC_AUTOIGEN_CONFIG_H_COMMON_SUBS,
     }),
 )
@@ -256,12 +255,6 @@
         "include/private/private.h",
         "include/private/xml.h",
     ] + select({
-        "@org_tensorflow//tensorflow:linux_x86_64": [
-            "hwloc/topology-linux.c",
-            "include/hwloc/linux.h",
-            "hwloc/topology-x86.c",
-            "include/private/cpuid-x86.h",
-        ],
         "@org_tensorflow//tensorflow:freebsd": [
             "hwloc/topology-freebsd.c",
             "hwloc/topology-x86.c",
