diff -ruN tensorflow-1.1.0/tensorflow/BUILD tensorflow-1.1.0-android/tensorflow/BUILD
--- tensorflow-1.1.0/tensorflow/BUILD	2017-04-21 10:01:36.000000000 +0900
+++ tensorflow-1.1.0-android/tensorflow/BUILD	2017-06-02 12:07:20.864073108 +0900
@@ -345,14 +345,29 @@
     ],
 )
 
+
+load("//tensorflow:tensorflow.bzl", "tf_copts")
+
 cc_binary(
     name = "libtensorflow_cc.so",
+    copts = tf_copts(),
+    linkopts = [
+        "-landroid",
+        "-ljnigraphics",
+        "-llog",
+        "-lm",
+        "-z defs",
+        "-s",
+        "-ldl",
+        "-Wl,--icf=all",  # Identical Code Folding
+    ],
+    tags = [
+        "manual",
+        "notap",
+    ],
     linkshared = 1,
+    linkstatic = 1,
     deps = [
-        "//tensorflow/c:c_api",
-        "//tensorflow/cc:cc_ops",
-        "//tensorflow/cc:client_session",
-        "//tensorflow/cc:scope",
-        "//tensorflow/core:tensorflow",
+        "//tensorflow/core:android_tensorflow_lib",
     ],
 )
diff -ruN tensorflow-1.1.0/tensorflow/c/BUILD tensorflow-1.1.0-android/tensorflow/c/BUILD
--- tensorflow-1.1.0/tensorflow/c/BUILD	2017-04-21 10:01:36.000000000 +0900
+++ tensorflow-1.1.0-android/tensorflow/c/BUILD	2017-06-02 12:05:01.401872302 +0900
@@ -45,6 +45,19 @@
     }),
 )
 
+filegroup(
+    name = "c_api_files",
+    srcs = glob(
+        ["**/*"],
+        exclude = [
+            "**/METADATA",
+            "**/OWNERS",
+            "**/*test.cc",
+        ],
+    ),
+    visibility = ["//visibility:public"],
+)
+
 tf_cuda_library(
     name = "tf_status_helper",
     srcs = ["tf_status_helper.cc"],
diff -ruN tensorflow-1.1.0/tensorflow/core/BUILD tensorflow-1.1.0-android/tensorflow/core/BUILD
--- tensorflow-1.1.0/tensorflow/core/BUILD	2017-04-21 10:01:36.000000000 +0900
+++ tensorflow-1.1.0-android/tensorflow/core/BUILD	2017-06-02 12:05:01.401872302 +0900
@@ -947,7 +947,10 @@
 # binary size (by packaging a reduced operator set) is a concern.
 cc_library(
     name = "android_tensorflow_lib",
-    srcs = if_android([":android_op_registrations_and_gradients"]),
+    srcs = if_android([
+    	":android_op_registrations_and_gradients",
+    	"//tensorflow/c:c_api_files",
+    ]),
     copts = tf_copts(),
     tags = [
         "manual",
diff -ruN tensorflow-1.1.0/tensorflow/core/kernels/BUILD tensorflow-1.1.0-android/tensorflow/core/kernels/BUILD
--- tensorflow-1.1.0/tensorflow/core/kernels/BUILD	2017-04-21 10:01:36.000000000 +0900
+++ tensorflow-1.1.0-android/tensorflow/core/kernels/BUILD	2017-06-02 12:05:01.402872303 +0900
@@ -3649,6 +3649,7 @@
         "transpose_op.h",
         "where_op.h",
         "xent_op.h",
+        "cwise_op_floor.cc",
     ],
 )
 
diff -ruN tensorflow-1.1.0/tensorflow/tensorflow.bzl tensorflow-1.1.0-android/tensorflow/tensorflow.bzl
--- tensorflow-1.1.0/tensorflow/tensorflow.bzl	2017-04-21 10:01:36.000000000 +0900
+++ tensorflow-1.1.0-android/tensorflow/tensorflow.bzl	2017-06-02 12:05:01.402872303 +0900
@@ -116,7 +116,6 @@
           select({
               "//tensorflow:android": [
                   "-std=c++11",
-                  "-DTF_LEAN_BINARY",
                   "-O2",
               ],
               "//tensorflow:darwin": [],
diff -ruN tensorflow-1.1.0/WORKSPACE tensorflow-1.1.0-android/WORKSPACE
--- tensorflow-1.1.0/WORKSPACE	2017-04-21 10:01:36.000000000 +0900
+++ tensorflow-1.1.0-android/WORKSPACE	2017-06-02 12:08:03.058133862 +0900
@@ -30,10 +30,10 @@
 #    path = "<PATH_TO_SDK>",
 #)
 #
-#android_ndk_repository(
-#    name="androidndk",
-#    path="<PATH_TO_NDK>",
-#    api_level=14)
+android_ndk_repository(
+    name="androidndk",
+    path="<PATH_TO_NDK>",
+    api_level=14)
 
 # Please add all new TensorFlow dependencies in workspace.bzl.
 tf_workspace()
