diff -ruN tensorflow-1.7.0-rc0/tensorflow/workspace.bzl tensorflow-1.7.0-rc0-macosx/tensorflow/workspace.bzl
--- tensorflow-1.7.0-rc0/tensorflow/workspace.bzl	2018-03-13 13:02:38.000000000 +0900
+++ tensorflow-1.7.0-rc0-macosx/tensorflow/workspace.bzl	2018-03-19 18:36:23.843382857 +0900
@@ -127,11 +127,11 @@
   tf_http_archive(
       name = "eigen_archive",
       urls = [
-          "https://mirror.bazel.build/bitbucket.org/eigen/eigen/get/2355b229ea4c.tar.gz",
-          "https://bitbucket.org/eigen/eigen/get/2355b229ea4c.tar.gz",
+          "https://mirror.bazel.build/bitbucket.org/dtrebbien/eigen/get/374842a18727.tar.gz",
+          "https://bitbucket.org/dtrebbien/eigen/get/374842a18727.tar.gz",
       ],
-      sha256 = "0cadb31a35b514bf2dfd6b5d38205da94ef326ec6908fc3fd7c269948467214f",
-      strip_prefix = "eigen-eigen-2355b229ea4c",
+      sha256 = "fa26e9b9ff3a2692b092d154685ec88d6cb84d4e1e895006541aff8603f15c16",
+      strip_prefix = "dtrebbien-eigen-374842a18727",
       build_file = str(Label("//third_party:eigen.BUILD")),
       patch_file = str(Label("//third_party:eigen_fix_cuda_compilation.patch"))
   )
@@ -361,11 +361,11 @@
   tf_http_archive(
       name = "protobuf_archive",
       urls = [
-          "https://mirror.bazel.build/github.com/google/protobuf/archive/396336eb961b75f03b25824fe86cf6490fb75e3a.tar.gz",
-          "https://github.com/google/protobuf/archive/396336eb961b75f03b25824fe86cf6490fb75e3a.tar.gz",
+          "https://mirror.bazel.build/github.com/dtrebbien/protobuf/archive/50f552646ba1de79e07562b41f3999fe036b4fd0.tar.gz",
+          "https://github.com/dtrebbien/protobuf/archive/50f552646ba1de79e07562b41f3999fe036b4fd0.tar.gz",
       ],
-      sha256 = "846d907acf472ae233ec0882ef3a2d24edbbe834b80c305e867ac65a1f2c59e3",
-      strip_prefix = "protobuf-396336eb961b75f03b25824fe86cf6490fb75e3a",
+      sha256 = "eb16b33431b91fe8cee479575cee8de202f3626aaf00d9bf1783c6e62b4ffbc7",
+      strip_prefix = "protobuf-50f552646ba1de79e07562b41f3999fe036b4fd0",
   )
 
   # We need to import the protobuf library under the names com_google_protobuf
diff -ruN tensorflow-1.7.0-rc0/third_party/gpus/cuda/BUILD.tpl tensorflow-1.7.0-rc0-macosx/third_party/gpus/cuda/BUILD.tpl
--- tensorflow-1.7.0-rc0/third_party/gpus/cuda/BUILD.tpl	2018-03-13 13:02:38.000000000 +0900
+++ tensorflow-1.7.0-rc0-macosx/third_party/gpus/cuda/BUILD.tpl	2018-03-19 18:36:42.134344424 +0900
@@ -62,7 +62,6 @@
         ":freebsd": [],
         "//conditions:default": ["-ldl"],
     }) + [
-        "-lpthread",
         %{cudart_static_linkopt}
     ],
     visibility = ["//visibility:public"],
@@ -110,7 +109,6 @@
         ".",
         "cuda/include",
     ],
-    linkopts = ["-lgomp"],
     linkstatic = 1,
     visibility = ["//visibility:public"],
 )
