diff -ruN tensorflow-master/tensorflow/cc/BUILD tensorflow-master-patch/tensorflow/cc/BUILD
--- tensorflow-master/tensorflow/cc/BUILD	2016-02-06 10:08:02.000000000 +0900
+++ tensorflow-master-patch/tensorflow/cc/BUILD	2016-02-06 17:42:56.446230724 +0900
@@ -75,6 +75,21 @@
     ],
 )
 
+cc_binary(
+    name = "libtensorflow.so",
+    copts = tf_copts(),
+    linkshared = 1,
+    linkopts = [
+        "-lpthread",
+        "-lm",
+    ],
+    deps = [
+        ":cc_ops",
+        "//tensorflow/core:all_kernels",
+        "//tensorflow/core:tensorflow",
+    ],
+)
+
 filegroup(
     name = "all_files",
     srcs = glob(
