diff -ruN opencv_contrib-3.4.0/modules/dnn_modern/CMakeLists.txt opencv_contrib-3.4.0-patch/modules/dnn_modern/CMakeLists.txt
--- opencv_contrib-3.4.0/modules/dnn_modern/CMakeLists.txt	2017-12-23 01:15:17.000000000 +0900
+++ opencv_contrib-3.4.0-patch/modules/dnn_modern/CMakeLists.txt	2018-01-25 09:58:43.289152300 +0900
@@ -15,16 +15,16 @@
 
 set(TINY_DNN_CPP_PATH "${OpenCV_BINARY_DIR}/3rdparty/tinydnn")
 set(TINY_DNN_CPP_ROOT "${TINY_DNN_CPP_PATH}/tiny-dnn-1.0.0a3")
-ocv_download(FILENAME "v1.0.0a3.tar.gz"
-               HASH "adb1c512e09ca2c7a6faef36f9c53e59"
-               URL
-                 "${OPENCV_TINY_DNN_URL}"
-                 "$ENV{OPENCV_TINY_DNN_URL}"
-                 "https://github.com/tiny-dnn/tiny-dnn/archive/"
-               DESTINATION_DIR "${TINY_DNN_CPP_PATH}"
-               STATUS TINY_DNN_DOWNLOAD_SUCCESS
-               ID "tiny-dnn"
-               UNPACK RELATIVE_URL)
+#ocv_download(FILENAME "v1.0.0a3.tar.gz"
+#               HASH "adb1c512e09ca2c7a6faef36f9c53e59"
+#               URL
+#                 "${OPENCV_TINY_DNN_URL}"
+#                 "$ENV{OPENCV_TINY_DNN_URL}"
+#                 "https://github.com/tiny-dnn/tiny-dnn/archive/"
+#               DESTINATION_DIR "${TINY_DNN_CPP_PATH}"
+#               STATUS TINY_DNN_DOWNLOAD_SUCCESS
+#               ID "tiny-dnn"
+#               UNPACK RELATIVE_URL)
 
 if(NOT TINY_DNN_DOWNLOAD_SUCCESS)
   message(STATUS "Failed to download tiny-dnn sources")
diff -ruN opencv_contrib-3.4.0/modules/face/include/opencv2/face/facemarkLBF.hpp opencv_contrib-3.4.0-patch/modules/face/include/opencv2/face/facemarkLBF.hpp
--- opencv_contrib-3.4.0/modules/face/include/opencv2/face/facemarkLBF.hpp	2017-12-23 01:15:17.000000000 +0900
+++ opencv_contrib-3.4.0-patch/modules/face/include/opencv2/face/facemarkLBF.hpp	2018-01-25 09:59:12.206227300 +0900
@@ -93,7 +93,7 @@
 
     };
 
-    class BBox {
+    class CV_EXPORTS BBox {
     public:
         BBox();
         ~BBox();
