diff -ruN opencv-3.0.0/cmake/OpenCVFindLibsPerf.cmake opencv-3.0.0-macosx/cmake/OpenCVFindLibsPerf.cmake
--- opencv-3.0.0/cmake/OpenCVFindLibsPerf.cmake	2015-06-04 02:21:34.000000000 +0900
+++ opencv-3.0.0-macosx/cmake/OpenCVFindLibsPerf.cmake	2015-06-06 07:35:28.397976485 +0900
@@ -94,7 +94,7 @@
 endif()
 
 # --- GCD ---
-if(APPLE AND NOT HAVE_TBB AND NOT HAVE_CSTRIPES)
+if(APPLE AND NOT HAVE_TBB AND NOT HAVE_CSTRIPES AND NOT WITH_OPENMP)
   set(HAVE_GCD 1)
 else()
   set(HAVE_GCD 0)
diff -ruN opencv-3.0.0/cmake/OpenCVFindLibsVideo.cmake opencv-3.0.0-macosx/cmake/OpenCVFindLibsVideo.cmake
--- opencv-3.0.0/cmake/OpenCVFindLibsVideo.cmake	2015-06-04 02:21:34.000000000 +0900
+++ opencv-3.0.0-macosx/cmake/OpenCVFindLibsVideo.cmake	2015-06-06 07:50:51.125528892 +0900
@@ -302,7 +302,7 @@
 if (NOT IOS)
   if(WITH_QUICKTIME)
     set(HAVE_QUICKTIME YES)
-  elseif(APPLE AND CMAKE_COMPILER_IS_CLANGCXX)
+  elseif(APPLE)
     set(HAVE_QTKIT YES)
   endif()
 endif()
diff -ruN opencv-3.0.0/cmake/OpenCVModule.cmake opencv-3.0.0-macosx/cmake/OpenCVModule.cmake
--- opencv-3.0.0/cmake/OpenCVModule.cmake	2015-06-04 02:21:34.000000000 +0900
+++ opencv-3.0.0-macosx/cmake/OpenCVModule.cmake	2015-06-06 07:35:00.881422272 +0900
@@ -754,7 +754,7 @@
     ARCHIVE_OUTPUT_DIRECTORY ${LIBRARY_OUTPUT_PATH}
     LIBRARY_OUTPUT_DIRECTORY ${LIBRARY_OUTPUT_PATH}
     RUNTIME_OUTPUT_DIRECTORY ${EXECUTABLE_OUTPUT_PATH}
-    INSTALL_NAME_DIR lib
+    INSTALL_NAME_DIR @rpath
   )
 
   # For dynamic link numbering convenions
