diff --git a/modules/core/src/utils/filesystem.cpp b/modules/core/src/utils/filesystem.cpp
index 266a92f8302..bba782fdbbe 100644
--- a/modules/core/src/utils/filesystem.cpp
+++ b/modules/core/src/utils/filesystem.cpp
@@ -41,6 +41,8 @@
 #include <errno.h>
 #endif
 
+#endif // OPENCV_HAVE_FILESYSTEM_SUPPORT
+
 namespace cv { namespace utils { namespace fs {
 
 #ifdef _WIN32
@@ -80,6 +82,8 @@ cv::String join(const cv::String& base, const cv::String& path)
     return result;
 }
 
+#if OPENCV_HAVE_FILESYSTEM_SUPPORT
+
 bool exists(const cv::String& path)
 {
     CV_INSTRUMENT_REGION()
@@ -511,6 +515,14 @@ cv::String getCacheDirectory(const char* sub_directory_name, const char* configu
     return cache_path;
 }
 
-}}} // namespace
-
+#else
+#define NOT_IMPLEMENTED CV_ErrorNoReturn(Error::StsNotImplemented, "");
+CV_EXPORTS bool exists(const cv::String& /*path*/) { NOT_IMPLEMENTED }
+CV_EXPORTS void remove_all(const cv::String& /*path*/) { NOT_IMPLEMENTED }
+CV_EXPORTS bool createDirectory(const cv::String& /*path*/) { NOT_IMPLEMENTED }
+CV_EXPORTS bool createDirectories(const cv::String& /*path*/) { NOT_IMPLEMENTED }
+CV_EXPORTS cv::String getCacheDirectory(const char* /*sub_directory_name*/, const char* /*configuration_name = NULL*/) { NOT_IMPLEMENTED }
+#undef NOT_IMPLEMENTED
 #endif // OPENCV_HAVE_FILESYSTEM_SUPPORT
+
+}}} // namespace

