--- opencv-3.0.0/modules/objdetect/include/opencv2/objdetect/detection_based_tracker.hpp	2015-06-03 13:21:34.000000000 -0400
+++ opencv-3.0.0-windows/modules/objdetect/include/opencv2/objdetect/detection_based_tracker.hpp	2015-06-22 18:01:13.077857300 -0400
@@ -58,7 +58,7 @@
 class CV_EXPORTS DetectionBasedTracker
 {
     public:
-        struct Parameters
+        struct CV_EXPORTS Parameters
         {
             int maxTrackLifetime;
             int minDetectionPeriod; //the minimal time between run of the big object detector (on the whole frame) in ms (1000 mean 1 sec), default=0
