diff -ruN opencv-3.4.1/3rdparty/libpng/pngpriv.h opencv-3.4.1-patch/3rdparty/libpng/pngpriv.h
--- opencv-3.4.1/3rdparty/libpng/pngpriv.h	2018-02-23 17:38:33.000000000 +0900
+++ opencv-3.4.1-patch/3rdparty/libpng/pngpriv.h	2018-03-01 19:52:25.535142165 +0900
@@ -192,13 +192,13 @@
 #  endif
 #endif
 
-#ifndef PNG_POWERPC_VSX_OPT
-#  if defined(__PPC64__) && defined(__ALTIVEC__) && defined(__VSX__)
-#     define PNG_POWERPC_VSX_OPT 2
-#  else
-#     define PNG_POWERPC_VSX_OPT 0
-#  endif
-#endif
+//#ifndef PNG_POWERPC_VSX_OPT
+//#  if defined(__PPC64__) && defined(__ALTIVEC__) && defined(__VSX__)
+//#     define PNG_POWERPC_VSX_OPT 2
+//#  else
+//#     define PNG_POWERPC_VSX_OPT 0
+//#  endif
+//#endif
 
 #ifndef PNG_INTEL_SSE_OPT
 #   ifdef PNG_INTEL_SSE
diff -ruN opencv-3.4.1/CMakeLists.txt opencv-3.4.1-patch/CMakeLists.txt
--- opencv-3.4.1/CMakeLists.txt	2018-02-23 17:38:33.000000000 +0900
+++ opencv-3.4.1-patch/CMakeLists.txt	2018-03-01 19:52:25.535142165 +0900
@@ -7,12 +7,12 @@
 # ----------------------------------------------------------------------------
 
 # Disable in-source builds to prevent source tree corruption.
-if(" ${CMAKE_SOURCE_DIR}" STREQUAL " ${CMAKE_BINARY_DIR}")
-  message(FATAL_ERROR "
-FATAL: In-source builds are not allowed.
-       You should create a separate directory for build files.
-")
-endif()
+#if(" ${CMAKE_SOURCE_DIR}" STREQUAL " ${CMAKE_BINARY_DIR}")
+#  message(FATAL_ERROR "
+#FATAL: In-source builds are not allowed.
+#       You should create a separate directory for build files.
+#")
+#endif()
 
 
 include(cmake/OpenCVMinDepVersions.cmake)
diff -ruN opencv-3.4.1/modules/core/include/opencv2/core/cv_cpu_dispatch.h opencv-3.4.1-patch/modules/core/include/opencv2/core/cv_cpu_dispatch.h
--- opencv-3.4.1/modules/core/include/opencv2/core/cv_cpu_dispatch.h	2018-02-23 17:38:33.000000000 +0900
+++ opencv-3.4.1-patch/modules/core/include/opencv2/core/cv_cpu_dispatch.h	2018-03-01 19:52:25.536142163 +0900
@@ -107,13 +107,13 @@
 #  include <arm_neon.h>
 #endif
 
-#if defined(__VSX__) && defined(__PPC64__) && defined(__LITTLE_ENDIAN__)
-#  include <altivec.h>
-#  undef vector
-#  undef pixel
-#  undef bool
-#  define CV_VSX 1
-#endif
+//#if defined(__VSX__) && defined(__PPC64__) && defined(__LITTLE_ENDIAN__)
+//#  include <altivec.h>
+//#  undef vector
+//#  undef pixel
+//#  undef bool
+//#  define CV_VSX 1
+//#endif
 
 #endif // CV_ENABLE_INTRINSICS && !CV_DISABLE_OPTIMIZATION && !__CUDACC__
 
@@ -151,12 +151,12 @@
 #elif defined(__ARM_NEON__) || (defined (__ARM_NEON) && defined(__aarch64__))
 #  include <arm_neon.h>
 #  define CV_NEON 1
-#elif defined(__VSX__) && defined(__PPC64__) && defined(__LITTLE_ENDIAN__)
-#  include <altivec.h>
-#  undef vector
-#  undef pixel
-#  undef bool
-#  define CV_VSX 1
+//#elif defined(__VSX__) && defined(__PPC64__) && defined(__LITTLE_ENDIAN__)
+//#  include <altivec.h>
+//#  undef vector
+//#  undef pixel
+//#  undef bool
+//#  define CV_VSX 1
 #endif
 
 #endif // !__OPENCV_BUILD && !__CUDACC (Compatibility code)
diff -ruN opencv-3.4.1/modules/core/include/opencv2/core/utility.hpp opencv-3.4.1-patch/modules/core/include/opencv2/core/utility.hpp
--- opencv-3.4.1/modules/core/include/opencv2/core/utility.hpp	2018-02-23 17:38:33.000000000 +0900
+++ opencv-3.4.1-patch/modules/core/include/opencv2/core/utility.hpp	2018-03-01 19:52:25.536142163 +0900
@@ -1200,7 +1200,7 @@
     double getTotalMs()   const { return ((double)m_ticksTotal / cv::getTickFrequency()) * 1000; }
     double getMeanMs()    const { return (((double)m_ticksTotal/m_counter) / cv::getTickFrequency()) * 1000; }
 };
-bool operator==(const NodeData& lhs, const NodeData& rhs);
+CV_EXPORTS bool operator==(const NodeData& lhs, const NodeData& rhs);
 
 typedef Node<NodeData> InstrNode;
 
diff -ruN opencv-3.4.1/modules/core/src/system.cpp opencv-3.4.1-patch/modules/core/src/system.cpp
--- opencv-3.4.1/modules/core/src/system.cpp	2018-02-23 17:38:33.000000000 +0900
+++ opencv-3.4.1-patch/modules/core/src/system.cpp	2018-03-01 19:52:25.537142162 +0900
@@ -1764,7 +1764,7 @@
 NodeData::~NodeData()
 {
 }
-bool operator==(const NodeData& left, const NodeData& right)
+CV_EXPORTS bool operator==(const NodeData& left, const NodeData& right)
 {
     if(left.m_lineNum == right.m_lineNum && left.m_funName == right.m_funName && left.m_fileName == right.m_fileName)
     {
diff -ruN opencv-3.4.1/modules/dnn/include/opencv2/dnn/all_layers.hpp opencv-3.4.1-patch/modules/dnn/include/opencv2/dnn/all_layers.hpp
--- opencv-3.4.1/modules/dnn/include/opencv2/dnn/all_layers.hpp	2018-02-23 17:38:33.000000000 +0900
+++ opencv-3.4.1-patch/modules/dnn/include/opencv2/dnn/all_layers.hpp	2018-03-01 19:52:25.537142162 +0900
@@ -555,7 +555,7 @@
      * An every sample in the batch is normalized separately. Optionally,
      * output is scaled by the trained parameters.
      */
-    class NormalizeBBoxLayer : public Layer
+    class CV_EXPORTS NormalizeBBoxLayer : public Layer
     {
     public:
         float pnorm, epsilon;
diff -ruN opencv-3.4.1/modules/dnn/include/opencv2/dnn/dnn.hpp opencv-3.4.1-patch/modules/dnn/include/opencv2/dnn/dnn.hpp
--- opencv-3.4.1/modules/dnn/include/opencv2/dnn/dnn.hpp	2018-02-23 17:38:33.000000000 +0900
+++ opencv-3.4.1-patch/modules/dnn/include/opencv2/dnn/dnn.hpp	2018-03-01 19:53:20.633052461 +0900
@@ -101,7 +101,7 @@
    /**
     * @brief Derivatives of this class encapsulates functions of certain backends.
     */
-    class BackendNode
+    class CV_EXPORTS BackendNode
     {
     public:
         BackendNode(int backendId);
@@ -256,7 +256,7 @@
         * Layer don't use own Halide::Func members because we can have applied
         * layers fusing. In this way the fused function should be scheduled.
         */
-        virtual void applyHalideScheduler(Ptr<BackendNode>& node,
+        virtual void applyHalideScheduler(const Ptr<BackendNode>& node,
                                           const std::vector<Mat*> &inputs,
                                           const std::vector<Mat> &outputs,
                                           int targetId) const;
@@ -285,7 +285,7 @@
          * @param[in] top Next layer to be fused.
          * @returns True if fusion was performed.
          */
-        virtual bool tryFuse(Ptr<Layer>& top);
+        virtual bool tryFuse(const Ptr<Layer>& top);
 
         /**
          * @brief Returns parameters of layers with channel-wise multiplication and addition.
diff -ruN opencv-3.4.1/modules/dnn/src/dnn.cpp opencv-3.4.1-patch/modules/dnn/src/dnn.cpp
--- opencv-3.4.1/modules/dnn/src/dnn.cpp	2018-02-23 17:38:33.000000000 +0900
+++ opencv-3.4.1-patch/modules/dnn/src/dnn.cpp	2018-03-01 19:53:32.077033829 +0900
@@ -2537,7 +2537,7 @@
     return Ptr<BackendNode>();
 }
 
-void Layer::applyHalideScheduler(Ptr<BackendNode>& node, const std::vector<Mat*> &inputs,
+void Layer::applyHalideScheduler(const Ptr<BackendNode>& node, const std::vector<Mat*> &inputs,
                                  const std::vector<Mat> &outputs, int targetId) const
 {
 #ifdef  HAVE_HALIDE
@@ -2612,7 +2612,7 @@
 }
 
 bool Layer::setActivation(const Ptr<ActivationLayer>&) { return false; }
-bool Layer::tryFuse(Ptr<Layer>&) { return false; }
+bool Layer::tryFuse(const Ptr<Layer>&) { return false; }
 void Layer::getScaleShift(Mat& scale, Mat& shift) const
 {
     scale = Mat();
diff -ruN opencv-3.4.1/modules/dnn/src/layers/convolution_layer.cpp opencv-3.4.1-patch/modules/dnn/src/layers/convolution_layer.cpp
--- opencv-3.4.1/modules/dnn/src/layers/convolution_layer.cpp	2018-02-23 17:38:33.000000000 +0900
+++ opencv-3.4.1-patch/modules/dnn/src/layers/convolution_layer.cpp	2018-03-01 19:53:49.171005998 +0900
@@ -309,7 +309,7 @@
         return !activ.empty();
     }
 
-    virtual bool tryFuse(Ptr<Layer>& top)
+    virtual bool tryFuse(const Ptr<Layer>& top)
     {
         Mat w, b;
         top->getScaleShift(w, b);
diff -ruN opencv-3.4.1/modules/dnn/src/layers/mvn_layer.cpp opencv-3.4.1-patch/modules/dnn/src/layers/mvn_layer.cpp
--- opencv-3.4.1/modules/dnn/src/layers/mvn_layer.cpp	2018-02-23 17:38:33.000000000 +0900
+++ opencv-3.4.1-patch/modules/dnn/src/layers/mvn_layer.cpp	2018-03-01 19:53:55.031996456 +0900
@@ -68,7 +68,7 @@
     Mat scale, shift;
     bool fuse_batch_norm;
 
-    virtual bool tryFuse(Ptr<Layer>& top)
+    virtual bool tryFuse(const Ptr<Layer>& top)
     {
         if (preferableTarget == DNN_TARGET_OPENCL && !fuse_batch_norm)
         {
diff -ruN opencv-3.4.1/modules/flann/include/opencv2/flann/miniflann.hpp opencv-3.4.1-patch/modules/flann/include/opencv2/flann/miniflann.hpp
--- opencv-3.4.1/modules/flann/include/opencv2/flann/miniflann.hpp	2018-02-23 17:38:33.000000000 +0900
+++ opencv-3.4.1-patch/modules/flann/include/opencv2/flann/miniflann.hpp	2018-03-01 19:52:25.538142160 +0900
@@ -74,10 +74,6 @@
                 std::vector<double>& numValues) const;
 
     void* params;
-
-private:
-    IndexParams(const IndexParams &); // copy disabled
-    IndexParams& operator=(const IndexParams &); // assign disabled
 };
 
 struct CV_EXPORTS KDTreeIndexParams : public IndexParams
diff -ruN opencv-3.4.1/modules/imgproc/src/fixedpoint.inl.hpp opencv-3.4.1-patch/modules/imgproc/src/fixedpoint.inl.hpp
--- opencv-3.4.1/modules/imgproc/src/fixedpoint.inl.hpp	2018-02-23 17:38:33.000000000 +0900
+++ opencv-3.4.1-patch/modules/imgproc/src/fixedpoint.inl.hpp	2018-03-01 19:52:25.538142160 +0900
@@ -80,8 +80,8 @@
     CV_ALWAYS_INLINE fixedpoint64 operator << (int n) const { return fixedpoint64(val << n); }
     template <typename ET>
     CV_ALWAYS_INLINE operator ET() const { return cv::saturate_cast<ET>((int64_t)fixedround((uint64_t)val) >> fixedShift); }
-    CV_ALWAYS_INLINE operator double() const { return (double)val / (1LL << fixedShift); }
-    CV_ALWAYS_INLINE operator float() const { return (float)val / (1LL << fixedShift); }
+//    CV_ALWAYS_INLINE operator double() const { return (double)val / (1LL << fixedShift); }
+//    CV_ALWAYS_INLINE operator float() const { return (float)val / (1LL << fixedShift); }
     CV_ALWAYS_INLINE bool isZero() { return val == 0; }
     static CV_ALWAYS_INLINE fixedpoint64 zero() { return fixedpoint64(); }
     static CV_ALWAYS_INLINE fixedpoint64 one() { return fixedpoint64((int64_t)(1LL << fixedShift)); }
@@ -135,8 +135,8 @@
     CV_ALWAYS_INLINE ufixedpoint64 operator << (int n) const { return ufixedpoint64(val << n); }
     template <typename ET>
     CV_ALWAYS_INLINE operator ET() const { return cv::saturate_cast<ET>(fixedround(val) >> fixedShift); }
-    CV_ALWAYS_INLINE operator double() const { return (double)val / (1LL << fixedShift); }
-    CV_ALWAYS_INLINE operator float() const { return (float)val / (1LL << fixedShift); }
+//    CV_ALWAYS_INLINE operator double() const { return (double)val / (1LL << fixedShift); }
+//    CV_ALWAYS_INLINE operator float() const { return (float)val / (1LL << fixedShift); }
     CV_ALWAYS_INLINE bool isZero() { return val == 0; }
     static CV_ALWAYS_INLINE ufixedpoint64 zero() { return ufixedpoint64(); }
     static CV_ALWAYS_INLINE ufixedpoint64 one() { return ufixedpoint64((uint64_t)(1ULL << fixedShift)); }
@@ -178,8 +178,8 @@
     CV_ALWAYS_INLINE fixedpoint32 operator << (int n) const { return fixedpoint32(val << n); }
     template <typename ET>
     CV_ALWAYS_INLINE operator ET() const { return cv::saturate_cast<ET>((int32_t)fixedround((uint32_t)val) >> fixedShift); }
-    CV_ALWAYS_INLINE operator double() const { return (double)val / (1 << fixedShift); }
-    CV_ALWAYS_INLINE operator float() const { return (float)val / (1 << fixedShift); }
+//    CV_ALWAYS_INLINE operator double() const { return (double)val / (1 << fixedShift); }
+//    CV_ALWAYS_INLINE operator float() const { return (float)val / (1 << fixedShift); }
     CV_ALWAYS_INLINE bool isZero() { return val == 0; }
     static CV_ALWAYS_INLINE fixedpoint32 zero() { return fixedpoint32(); }
     static CV_ALWAYS_INLINE fixedpoint32 one() { return fixedpoint32((1 << fixedShift)); }
@@ -219,8 +219,8 @@
     CV_ALWAYS_INLINE ufixedpoint32 operator << (int n) const { return ufixedpoint32(val << n); }
     template <typename ET>
     CV_ALWAYS_INLINE operator ET() const { return cv::saturate_cast<ET>(fixedround(val) >> fixedShift); }
-    CV_ALWAYS_INLINE operator double() const { return (double)val / (1 << fixedShift); }
-    CV_ALWAYS_INLINE operator float() const { return (float)val / (1 << fixedShift); }
+//    CV_ALWAYS_INLINE operator double() const { return (double)val / (1 << fixedShift); }
+//    CV_ALWAYS_INLINE operator float() const { return (float)val / (1 << fixedShift); }
     CV_ALWAYS_INLINE bool isZero() { return val == 0; }
     static CV_ALWAYS_INLINE ufixedpoint32 zero() { return ufixedpoint32(); }
     static CV_ALWAYS_INLINE ufixedpoint32 one() { return ufixedpoint32((1U << fixedShift)); }
@@ -253,8 +253,8 @@
     CV_ALWAYS_INLINE fixedpoint16 operator << (int n) const { return fixedpoint16((int16_t)(val << n)); }
     template <typename ET>
     CV_ALWAYS_INLINE operator ET() const { return cv::saturate_cast<ET>((int16_t)fixedround((uint16_t)val) >> fixedShift); }
-    CV_ALWAYS_INLINE operator double() const { return (double)val / (1 << fixedShift); }
-    CV_ALWAYS_INLINE operator float() const { return (float)val / (1 << fixedShift); }
+//    CV_ALWAYS_INLINE operator double() const { return (double)val / (1 << fixedShift); }
+//    CV_ALWAYS_INLINE operator float() const { return (float)val / (1 << fixedShift); }
     CV_ALWAYS_INLINE bool isZero() { return val == 0; }
     static CV_ALWAYS_INLINE fixedpoint16 zero() { return fixedpoint16(); }
     static CV_ALWAYS_INLINE fixedpoint16 one() { return fixedpoint16((int16_t)(1 << fixedShift)); }
@@ -285,8 +285,8 @@
     CV_ALWAYS_INLINE ufixedpoint16 operator << (int n) const { return ufixedpoint16((uint16_t)(val << n)); }
     template <typename ET>
     CV_ALWAYS_INLINE operator ET() const { return cv::saturate_cast<ET>(fixedround(val) >> fixedShift); }
-    CV_ALWAYS_INLINE operator double() const { return (double)val / (1 << fixedShift); }
-    CV_ALWAYS_INLINE operator float() const { return (float)val / (1 << fixedShift); }
+//    CV_ALWAYS_INLINE operator double() const { return (double)val / (1 << fixedShift); }
+//    CV_ALWAYS_INLINE operator float() const { return (float)val / (1 << fixedShift); }
     CV_ALWAYS_INLINE bool isZero() { return val == 0; }
     static CV_ALWAYS_INLINE ufixedpoint16 zero() { return ufixedpoint16(); }
     static CV_ALWAYS_INLINE ufixedpoint16 one() { return ufixedpoint16((uint16_t)(1 << fixedShift)); }
@@ -294,4 +294,4 @@
 
 }
 
-#endif
\ No newline at end of file
+#endif
diff -ruN opencv-3.4.1/modules/imgproc/src/resize.cpp opencv-3.4.1-patch/modules/imgproc/src/resize.cpp
--- opencv-3.4.1/modules/imgproc/src/resize.cpp	2018-02-23 17:38:33.000000000 +0900
+++ opencv-3.4.1-patch/modules/imgproc/src/resize.cpp	2018-03-01 19:52:25.539142158 +0900
@@ -743,11 +743,11 @@
     void(*hResize)(ET* src, int cn, int *ofst, fixedpoint* m, fixedpoint* dst, int dst_min, int dst_max, int dst_width);
     switch (cn)
     {
-    case  1: hResize = src_width > interpolation::len ? hlineResizeCn<ET, fixedpoint, interpolation::len, true, 1> : hlineResizeCn<ET, fixedpoint, interpolation::len, false, 1>; break;
-    case  2: hResize = src_width > interpolation::len ? hlineResizeCn<ET, fixedpoint, interpolation::len, true, 2> : hlineResizeCn<ET, fixedpoint, interpolation::len, false, 2>; break;
-    case  3: hResize = src_width > interpolation::len ? hlineResizeCn<ET, fixedpoint, interpolation::len, true, 3> : hlineResizeCn<ET, fixedpoint, interpolation::len, false, 3>; break;
-    case  4: hResize = src_width > interpolation::len ? hlineResizeCn<ET, fixedpoint, interpolation::len, true, 4> : hlineResizeCn<ET, fixedpoint, interpolation::len, false, 4>; break;
-    default: hResize = src_width > interpolation::len ? hlineResize<ET, fixedpoint, interpolation::len, true>      : hlineResize<ET, fixedpoint, interpolation::len, false>     ; break;
+    case  1: if (src_width > interpolation::len) hResize = hlineResizeCn<ET, fixedpoint, interpolation::len, true, 1>; else hResize = hlineResizeCn<ET, fixedpoint, interpolation::len, false, 1>; break;
+    case  2: if (src_width > interpolation::len) hResize = hlineResizeCn<ET, fixedpoint, interpolation::len, true, 2>; else hResize = hlineResizeCn<ET, fixedpoint, interpolation::len, false, 2>; break;
+    case  3: if (src_width > interpolation::len) hResize = hlineResizeCn<ET, fixedpoint, interpolation::len, true, 3>; else hResize = hlineResizeCn<ET, fixedpoint, interpolation::len, false, 3>; break;
+    case  4: if (src_width > interpolation::len) hResize = hlineResizeCn<ET, fixedpoint, interpolation::len, true, 4>; else hResize = hlineResizeCn<ET, fixedpoint, interpolation::len, false, 4>; break;
+    default: if (src_width > interpolation::len) hResize = hlineResize<ET, fixedpoint, interpolation::len, true>     ; else hResize = hlineResize<ET, fixedpoint, interpolation::len, false>     ; break;
     }
 
     interpolation interp_x(inv_scale_x, src_width, dst_width);
