diff -ruN opencv-2.4.10/modules/core/src/system.cpp opencv-2.4.10-android/modules/core/src/system.cpp
--- opencv-2.4.10/modules/core/src/system.cpp	2014-10-01 16:33:36.000000000 +0900
+++ opencv-2.4.10-android/modules/core/src/system.cpp	2014-12-10 22:10:21.354292844 +0900
@@ -267,14 +267,16 @@
          : "cc"
         );
         #else
+        // Fix from https://github.com/Itseez/opencv/commit/ea50be0529c248961e1b66293f8a9e4b807294a6
         asm volatile
         (
          "pushl %%ebx\n\t"
          "movl $7,%%eax\n\t"
          "movl $0,%%ecx\n\t"
          "cpuid\n\t"
+         "movl %%ebx,%1\n\t"
          "popl %%ebx\n\t"
-         : "=a"(cpuid_data[0]), "=b"(cpuid_data[1]), "=c"(cpuid_data[2]), "=d"(cpuid_data[3])
+         : "=a"(cpuid_data[0]), "=r"(cpuid_data[1]), "=c"(cpuid_data[2]), "=d"(cpuid_data[3])
          :
          : "cc"
         );
