diff -ruN opencv-2.4.9/3rdparty/tbb/CMakeLists.txt opencv-2.4.9-macosx/3rdparty/tbb/CMakeLists.txt
--- opencv-2.4.9/3rdparty/tbb/CMakeLists.txt	2014-04-11 19:15:26.000000000 +0900
+++ opencv-2.4.9-macosx/3rdparty/tbb/CMakeLists.txt	2014-11-01 17:31:13.726107924 +0900
@@ -177,13 +177,7 @@
                  )
 endif()
 
-if (HAVE_LIBPTHREAD)
   add_definitions(-DUSE_PTHREAD) #required for Unix
-endif()
-
-if (CMAKE_COMPILER_IS_GNUCXX)
-  add_definitions(-DTBB_USE_GCC_BUILTINS=1) #required for ARM GCC
-endif()
 
 if(ANDROID_COMPILER_IS_CLANG)
   add_definitions(-D__TBB_GCC_BUILTIN_ATOMICS_PRESENT=1)
@@ -209,6 +203,7 @@
 set(TBB_SOURCE_FILES ${TBB_SOURCE_FILES} "${CMAKE_CURRENT_SOURCE_DIR}/${tbb_version_file}")
 
 add_library(tbb ${TBB_SOURCE_FILES})
+set_target_properties(tbb PROPERTIES INSTALL_NAME_DIR "@rpath")
 
 if (WIN32)
   if (ARM)
diff -ruN opencv-2.4.9/cmake/OpenCVModule.cmake opencv-2.4.9-macosx/cmake/OpenCVModule.cmake
--- opencv-2.4.9/cmake/OpenCVModule.cmake	2014-04-11 19:15:26.000000000 +0900
+++ opencv-2.4.9-macosx/cmake/OpenCVModule.cmake	2014-11-01 17:31:44.288696714 +0900
@@ -568,7 +568,7 @@
     ARCHIVE_OUTPUT_DIRECTORY ${LIBRARY_OUTPUT_PATH}
     LIBRARY_OUTPUT_DIRECTORY ${LIBRARY_OUTPUT_PATH}
     RUNTIME_OUTPUT_DIRECTORY ${EXECUTABLE_OUTPUT_PATH}
-    INSTALL_NAME_DIR lib
+    INSTALL_NAME_DIR @rpath
   )
 
   # For dynamic link numbering convenions
