diff -ruN libfreenect2-0.2.0/cmake_modules/FindGLFW3.cmake libfreenect2-0.2.0-patch/cmake_modules/FindGLFW3.cmake
--- libfreenect2-0.2.0/cmake_modules/FindGLFW3.cmake	2016-04-28 05:34:51.000000000 +0900
+++ libfreenect2-0.2.0-patch/cmake_modules/FindGLFW3.cmake	2017-12-05 13:07:59.751981013 +0900
@@ -6,23 +6,6 @@
 # GLFW3_INCLUDE_DIRS
 # GLFW3_LIBRARIES
 
-IF(PKG_CONFIG_FOUND)
-  IF(APPLE)
-    # homebrew or macports pkgconfig locations
-    SET(ENV{PKG_CONFIG_PATH} "/usr/local/opt/glfw3/lib/pkgconfig:/opt/local/lib/pkgconfig")
-  ENDIF()
-  SET(ENV{PKG_CONFIG_PATH} "${DEPENDS_DIR}/glfw/lib/pkgconfig:$ENV{PKG_CONFIG_PATH}")
-  PKG_CHECK_MODULES(GLFW3 glfw3)
-
-  FIND_LIBRARY(GLFW3_LIBRARY
-    NAMES ${GLFW3_LIBRARIES}
-    HINTS ${GLFW3_LIBRARY_DIRS}
-  )
-  SET(GLFW3_LIBRARIES ${GLFW3_LIBRARY})
-
-  RETURN()
-ENDIF()
-
 FIND_PATH(GLFW3_INCLUDE_DIRS
   GLFW/glfw3.h
   DOC "GLFW include directory "
