diff -ruN libfreenect2-0.2.0/include/libfreenect2/frame_listener_impl.h libfreenect2-0.2.0_patch/include/libfreenect2/frame_listener_impl.h
--- libfreenect2-0.2.0/include/libfreenect2/frame_listener_impl.h	2016-04-27 22:34:51.000000000 +0200
+++ libfreenect2-0.2.0_patch/include/libfreenect2/frame_listener_impl.h	2016-10-29 17:30:57.949091000 +0200
@@ -40,7 +40,7 @@
 ///@{
 
 /** Storage of multiple different types of frames. */
-typedef std::map<Frame::Type, Frame*> FrameMap;
+// typedef std::map<Frame::Type, Frame*> FrameMap;
 
 class SyncMultiFrameListenerImpl;
 
@@ -62,15 +62,15 @@
    * @param milliseconds Timeout. This parameter is ignored if not built with C++11 threading support.
    * @return true if a frame is received; false if not.
    */
-  bool waitForNewFrame(FrameMap &frame, int milliseconds);
+  bool waitForNewFrame(std::map<Frame::Type, Frame*> &frame, int milliseconds);
 
   /** Wait indefinitely for new frames.
    * @param[out] frame Caller is responsible to release the frames.
    */
-  void waitForNewFrame(FrameMap &frame);
+  void waitForNewFrame(std::map<Frame::Type, Frame*> &frame);
 
   /** Shortcut to delete all frames */
-  void release(FrameMap &frame);
+  void release(std::map<Frame::Type, Frame*> &frame);
 
   virtual bool onNewFrame(Frame::Type type, Frame *frame);
 private:
