diff -ruN libfreenect2-0.2.0/cmake_modules/FindGLFW3.cmake libfreenect2-0.2.0-patch/cmake_modules/FindGLFW3.cmake
--- libfreenect2-0.2.0/cmake_modules/FindGLFW3.cmake	2016-04-28 05:34:51.000000000 +0900
+++ libfreenect2-0.2.0-patch/cmake_modules/FindGLFW3.cmake	2018-03-02 23:35:54.634802604 +0900
@@ -6,23 +6,6 @@
 # GLFW3_INCLUDE_DIRS
 # GLFW3_LIBRARIES
 
-IF(PKG_CONFIG_FOUND)
-  IF(APPLE)
-    # homebrew or macports pkgconfig locations
-    SET(ENV{PKG_CONFIG_PATH} "/usr/local/opt/glfw3/lib/pkgconfig:/opt/local/lib/pkgconfig")
-  ENDIF()
-  SET(ENV{PKG_CONFIG_PATH} "${DEPENDS_DIR}/glfw/lib/pkgconfig:$ENV{PKG_CONFIG_PATH}")
-  PKG_CHECK_MODULES(GLFW3 glfw3)
-
-  FIND_LIBRARY(GLFW3_LIBRARY
-    NAMES ${GLFW3_LIBRARIES}
-    HINTS ${GLFW3_LIBRARY_DIRS}
-  )
-  SET(GLFW3_LIBRARIES ${GLFW3_LIBRARY})
-
-  RETURN()
-ENDIF()
-
 FIND_PATH(GLFW3_INCLUDE_DIRS
   GLFW/glfw3.h
   DOC "GLFW include directory "
diff -ruN libfreenect2-0.2.0/cmake_modules/FindLibUSB.cmake libfreenect2-0.2.0-patch/cmake_modules/FindLibUSB.cmake
--- libfreenect2-0.2.0/cmake_modules/FindLibUSB.cmake	2016-04-28 05:34:51.000000000 +0900
+++ libfreenect2-0.2.0-patch/cmake_modules/FindLibUSB.cmake	2018-03-02 23:36:07.931776660 +0900
@@ -7,28 +7,6 @@
 #  LibUSB_INCLUDE_DIR: the directory that contains the include file
 #  LibUSB_LIBRARIES:  the libraries
 
-IF(PKG_CONFIG_FOUND)
-  IF(DEPENDS_DIR) #Otherwise use System pkg-config path
-    SET(ENV{PKG_CONFIG_PATH} "${DEPENDS_DIR}/libusb/lib/pkgconfig")
-  ENDIF()
-  SET(MODULE "libusb-1.0")
-  IF(CMAKE_SYSTEM_NAME MATCHES "Linux")
-    SET(MODULE "libusb-1.0>=1.0.20")
-  ENDIF()
-  IF(LibUSB_FIND_REQUIRED)
-    SET(LibUSB_REQUIRED "REQUIRED")
-  ENDIF()
-  PKG_CHECK_MODULES(LibUSB ${LibUSB_REQUIRED} ${MODULE})
-
-  FIND_LIBRARY(LibUSB_LIBRARY
-    NAMES ${LibUSB_LIBRARIES}
-    HINTS ${LibUSB_LIBRARY_DIRS}
-  )
-  SET(LibUSB_LIBRARIES ${LibUSB_LIBRARY})
-
-  RETURN()
-ENDIF()
-
 FIND_PATH(LibUSB_INCLUDE_DIRS
   NAMES libusb.h
   PATHS
