--- backend.hpp.orig	2018-11-09 20:11:42.216762981 -0500
+++ backend.hpp	2018-11-09 20:08:37.998002342 -0500
@@ -33,18 +33,18 @@
         class Backend
         {
         public:
-            Backend(const Backend&) = delete;
-            Backend& operator=(const Backend&) = delete;
+//            Backend(const Backend&) = delete;
+//            Backend& operator=(const Backend&) = delete;
 
-            Backend(Backend&&) = default;
-            Backend& operator=(Backend&&) = default;
+//            Backend(Backend&&) = default;
+//            Backend& operator=(Backend&&) = default;
 
-            Backend() = delete;
+//            Backend() = delete;
 
-            explicit Backend(const std::string& type)
-                : m_type{type}
-            {
-            }
+           
+          
+         
+        
 
             const std::string& get_type() const { return m_type; }
             bool compile(const std::shared_ptr<Function>& function) const
