--- CMakeLists.txt.ORIG	2018-12-01 16:28:12.228687268 -0500
+++ CMakeLists.txt	2018-12-01 16:28:28.676543930 -0500
@@ -195,7 +195,7 @@
     endif()
 
     add_dependencies(cpu_backend ext_mkldnn ext_eigen)
-    target_link_libraries(cpu_backend PUBLIC ngraph libmkldnn libeigen libjson libtbb)
+    target_link_libraries(cpu_backend PUBLIC ngraph libmkldnn libeigen libjson)
     if (NOT NGRAPH_DEX_ONLY)
         target_link_libraries(cpu_backend PUBLIC codegen)
     endif()
