--- cppbuild/linux-x86_64/include/ngraph/type/element_type.hpp	2018-11-09 08:30:42.628084431 -0500
+++ element_type.hpp	2018-11-09 08:34:47.872535397 -0500
@@ -110,38 +110,6 @@
             std::string m_cname{"dynamic"};
         };
 
-        template <typename T>
-        const Type& from()
-        {
-            throw std::invalid_argument("Unknown type");
-        }
-        template <>
-        const Type& from<char>();
-        template <>
-        const Type& from<bool>();
-        template <>
-        const Type& from<float>();
-        template <>
-        const Type& from<double>();
-        template <>
-        const Type& from<int8_t>();
-        template <>
-        const Type& from<int16_t>();
-        template <>
-        const Type& from<int32_t>();
-        template <>
-        const Type& from<int64_t>();
-        template <>
-        const Type& from<uint8_t>();
-        template <>
-        const Type& from<uint16_t>();
-        template <>
-        const Type& from<uint32_t>();
-        template <>
-        const Type& from<uint64_t>();
-        template <>
-        const Type& from<ngraph::bfloat16>();
-
         std::ostream& operator<<(std::ostream& out, const ngraph::element::Type& obj);
     }
 }
