--- onnx.hpp.ORIG	2019-01-15 13:26:36.332226565 -0500
+++ onnx.hpp	2019-01-15 13:41:54.215455942 -0500
@@ -50,44 +50,8 @@
         ///                   the model  will take precedence).
         /// \return The function returns a vector of nGraph functions. The number of functions
         ///          depends on number of outputs from ONNX graph.
-        std::vector<std::shared_ptr<Function>> load_onnx_model(std::istream& sin,
+        std::vector<std::shared_ptr<Function>> load_onnx_model(const std::string& data,
                                                                const Weights& weights = {});
-        /// \brief Convert an ONNX model to nGraph functions
-        /// The function translated serialized ONNX model to nGraph functions. The ONNX model
-        /// is read from ONNX file.
-        /// \param filename  file name (relative or absolute path name),
-        /// \param weights  weights associated with the model. If weights are embedded into
-        ///                   the model this parameter shall be empty. Having weights in a model
-        ///                   and providing through this parameters is invalid (the weights from
-        ///                   the model  will take precedence).
-        /// \return The function returns a vector of nGraph functions. The number of functions
-        ///          depends on number of outputs from ONNX graph.
-        std::vector<std::shared_ptr<Function>> load_onnx_model(const std::string& filename,
-                                                               const Weights& weights = {});
-
-        /// \brief Convert an ONNX model to nGraph function
-        /// The function translated serialized ONNX model to nGraph function. The serialized
-        /// ONNX model is read from input stream.
-        /// \param sin       input stream (e.g. file stream, memory stream, etc),
-        /// \param weights  weights associated with the model. If weights are embedded into
-        ///                   the model this parameter shall be empty. Having weights in a model
-        ///                   and providing through this parameters is invalid (the weights from
-        ///                   the model  will take precedence).
-        /// \return The function returns a nGraph function representing single output from graph.
-        std::shared_ptr<Function> import_onnx_function(std::istream& sin,
-                                                       const Weights& weights = {});
-
-        /// \brief Convert an ONNX model to nGraph functions
-        /// The function translated serialized ONNX model to nGraph functions. The ONNX model
-        /// is read from ONNX file.
-        /// \param filename  file name (relative or absolute path name),
-        /// \param weights  weights associated with the model. If weights are embedded into
-        ///                   the model this parameter shall be empty. Having weights in a model
-        ///                   and providing through this parameters is invalid (the weights from
-        ///                   the model  will take precedence).
-        /// \return The function returns a nGraph function representing single output from graph.
-        std::shared_ptr<Function> import_onnx_function(const std::string& filename,
-                                                       const Weights& weights = {});
 
     } // namespace onnx_import
 
