--- cppbuild/linux-x86_64/include/ngraph/descriptor/tensor.hpp	2018-11-09 20:25:08.356441813 -0500
+++ tensor.hpp	2018-11-09 20:27:13.140033382 -0500
@@ -38,8 +38,8 @@
         /// \brief Compile-time descriptor of a first-class value that is a view of a tensor.
         class Tensor
         {
-            Tensor(const Tensor&) = delete;
-            Tensor& operator=(const Tensor&) = delete;
+//            Tensor(const Tensor&) = delete;
+//            Tensor& operator=(const Tensor&) = delete;
 
         public:
             Tensor(const element::Type& element_type,
