diff -ruN Python-3.9.7/PCbuild/python.props Python-3.9.7-patch/PCbuild/python.props
--- Python-3.9.7/PCbuild/python.props	2021-08-31 04:02:15.000000000 +0900
+++ Python-3.9.7-patch/PCbuild/python.props	2021-09-22 12:07:13.518157206 +0900
@@ -107,6 +107,9 @@
     <!-- Sometimes the version in the registry has to .0 suffix, and sometimes it doesn't. Check and add it -->
     <_RegistryVersion Condition="$(_RegistryVersion) != '' and !$(_RegistryVersion.EndsWith('.0'))">$(_RegistryVersion).0</_RegistryVersion>
 
+    <!-- Avoid upgrading to Windows 11 SDK for now, but assume the latest Win10 SDK is installed -->
+    <_RegistryVersion Condition="$([System.Version]::Parse($(_RegistryVersion))) >= $([System.Version]::Parse(`10.0.22000.0`))">10.0.19041.0</_RegistryVersion>
+
     <!-- The minimum allowed SDK version to use for building -->
     <DefaultWindowsSDKVersion>10.0.10586.0</DefaultWindowsSDKVersion>
     <DefaultWindowsSDKVersion Condition="$(_RegistryVersion) != '' and $([System.Version]::Parse($(_RegistryVersion))) > $([System.Version]::Parse($(DefaultWindowsSDKVersion)))">$(_RegistryVersion)</DefaultWindowsSDKVersion>
diff -ruN Python-3.9.7/Python/dynload_win.c Python-3.9.7-patch/Python/dynload_win.c
--- Python-3.9.7/Python/dynload_win.c	2021-08-31 04:02:15.000000000 +0900
+++ Python-3.9.7-patch/Python/dynload_win.c	2021-09-22 12:06:36.377106394 +0900
@@ -194,6 +194,10 @@
         hDLL = LoadLibraryExW(wpathname, NULL,
                               LOAD_LIBRARY_SEARCH_DEFAULT_DIRS |
                               LOAD_LIBRARY_SEARCH_DLL_LOAD_DIR);
+        if (hDLL == NULL) {
+            hDLL = LoadLibraryExW(wpathname, NULL,
+                                  LOAD_WITH_ALTERED_SEARCH_PATH);
+        }
         Py_END_ALLOW_THREADS
 
         /* restore old error mode settings */
