diff --git a/liquidfun/Box2D/Box2D/Particle/b2ParticleSystem.cpp b/liquidfun/Box2D/Box2D/Particle/b2ParticleSystem.cpp
index ce1c01e..4469eb1 100644
--- a/liquidfun/Box2D/Box2D/Particle/b2ParticleSystem.cpp
+++ b/liquidfun/Box2D/Box2D/Particle/b2ParticleSystem.cpp
@@ -52,7 +52,7 @@ static const uint32 yMask = ((1u << yTruncBits) - 1u) << yShift;
 static const uint32 xMask = ~yMask;
 static const uint32 relativeTagRight = 1u << xShift;
 static const uint32 relativeTagBottomLeft = (uint32)((1 << yShift) +
-                                                    (-1 << xShift));
+                                                    ((1 << xShift) * -1));
 
 static const uint32 relativeTagBottomRight = (1u << yShift) + (1u << xShift);
 
diff --git a/liquidfun/Box2D/CMakeLists.txt b/liquidfun/Box2D/CMakeLists.txt
index 050016a..623f3e4 100644
--- a/liquidfun/Box2D/CMakeLists.txt
+++ b/liquidfun/Box2D/CMakeLists.txt
@@ -160,6 +160,7 @@ set(EXE_OUTPUT_DIR "${OUTPUT_SUBDIR}")
 foreach(target ${ALL_TARGETS})
   set_target_properties(
     ${target} PROPERTIES
+    INSTALL_NAME_DIR "@rpath"
     ARCHIVE_OUTPUT_DIRECTORY "${LIB_OUTPUT_DIR}"
     LIBRARY_OUTPUT_DIRECTORY "${LIB_OUTPUT_DIR}"
     RUNTIME_OUTPUT_DIRECTORY "${EXE_OUTPUT_DIR}")
