--- protobuf-2.6.1/autogen.sh	2014-10-20 20:06:06.000000000 -0400
+++ autogen.sh	2018-04-15 19:38:24.032970600 -0400
@@ -19,8 +19,9 @@
 # directory is set up as an SVN external.
 if test ! -e gtest; then
   echo "Google Test not present.  Fetching gtest-1.5.0 from the web..."
-  curl http://googletest.googlecode.com/files/gtest-1.5.0.tar.bz2 | tar jx
-  mv gtest-1.5.0 gtest
+  wget https://github.com/google/googletest/archive/release-1.5.0.zip 
+  unzip release-1.5.0.zip
+  mv googletest-release-1.5.0 gtest
 fi
 
 set -ex
