--- optimize.h.ORIG.1.3.0	2018-10-11 16:59:22.016201924 -0500
+++ optimize.h	2018-10-12 10:48:31.168918744 -0500
@@ -85,7 +85,7 @@
   template<class Optimizer, class... Args> void registerOptimizer(Args&& ...args) {
     auto optimizer = make_unique<Optimizer>(std::forward<Args>(args)...);
     passes[optimizer->name] = std::move(optimizer);
-  }
+  };
 };
 
 const std::vector<std::string> GetAvailablePasses();
@@ -95,4 +95,4 @@
     const std::vector<std::string>& names);
 
 } // namespace optimization
-} // namespace ONNX_NAMESPACE
+}; // namespace ONNX_NAMESPACE
