--- cppbuild/linux-x86_64/onnx-1.3.0/onnx/optimizer/passes/optimize_pass.h	2018-08-30 12:33:20.000000000 -0500
+++ optimize_pass.h	2018-10-11 15:39:57.150911599 -0500
@@ -14,7 +14,8 @@
 
 struct OptimizePass {
 
-  virtual ~OptimizePass() noexcept = 0;
+
+  virtual ~OptimizePass() noexcept = default;
 
   std::string name;
   API_TYPE type;
@@ -43,6 +44,6 @@
 
 };
 
-inline OptimizePass::~OptimizePass() noexcept = default;
+
 
 }} // namespace ONNX_NAMESPACE::optimization
