--- schema.h.orig.1.3.0	2018-09-05 15:25:41.331934342 -0600
+++ schema.h	2018-09-06 10:20:08.590867350 -0600
@@ -503,10 +503,6 @@
     return name_;
   }
 
-  OperatorSetVersion SinceVersion() const {
-    return since_version_;
-  }
-
   int since_version() const {
     return since_version_;
   }
@@ -632,7 +628,7 @@
 
         auto& op_name = op_schema.Name();
         auto& op_domain = op_schema.domain();
-        auto ver = op_schema.SinceVersion();
+        auto ver = op_schema.since_version();
 
         if (m[op_name][op_domain].count(ver)) {
           const auto& schema = m[op_name][op_domain][ver];
@@ -845,21 +841,6 @@
 // Helper function
 size_t ReplaceAll(std::string& s, const char* from, const char* to);
 
-#ifdef __GNUC__
-#define ONNX_UNUSED __attribute__((__unused__))
-#else
-#define ONNX_UNUSED
-#endif
-
-// Legacy macros to register schema at static initialization
-#define ONNX_OPERATOR_SCHEMA(name) \
-  ONNX_OPERATOR_SCHEMA_UNIQ_HELPER(__COUNTER__, name)
-#define ONNX_OPERATOR_SCHEMA_UNIQ_HELPER(Counter, name) \
-  ONNX_OPERATOR_SCHEMA_UNIQ(Counter, name)
-#define ONNX_OPERATOR_SCHEMA_UNIQ(Counter, name)                 \
-  static ONNX_NAMESPACE::OpSchemaRegistry::OpSchemaRegisterOnce( \
-      op_schema_register_once##name##Counter) ONNX_UNUSED =      \
-      OpSchema(#name, __FILE__, __LINE__)
 
 // Helper function
 size_t ReplaceAll(std::string& s, const char* from, const char* to);
