--- cppbuild/linux-x86_64/onnx-1.0.1/setup.py	2018-01-25 21:25:37.000000000 -0500
+++ setup.py	2018-04-15 13:16:01.494864597 -0400
@@ -328,7 +328,7 @@
 def create_extension(ExtType, name, sources, dependencies, extra_link_args, extra_objects):
     include_dirs = sum([dep.include_dirs for dep in dependencies], [TOP_DIR])
     libraries = sum([dep.libraries for dep in dependencies], [])
-    extra_compile_args=['-std=c++11']
+    extra_compile_args=['-std=c++0x']
     if sys.platform == 'darwin':
         extra_compile_args.append('-stdlib=libc++')
     if os.getenv('CONDA_PREFIX'):
