diff -ruN gsl-2.4/configure gsl-2.4-windows/configure
--- gsl-2.4/configure	2017-06-19 21:55:37.000000000 +0900
+++ gsl-2.4-windows/configure	2018-02-21 12:34:56.928604100 +0900
@@ -12289,7 +12289,6 @@
   if test "$enable_shared" = yes; then
     GSLCBLAS_LDFLAGS="$GSLCBLAS_LDFLAGS -no-undefined"
     GSL_LDFLAGS="$GSL_LDFLAGS -no-undefined"
-    GSL_LIBADD="cblas/libgslcblas.la"
   fi
   ;;
 esac
