diff -ruN gsl-2.1/cblas/Makefile.am gsl-2.1-android/cblas/Makefile.am
--- gsl-2.1/cblas/Makefile.am	2015-11-04 01:21:55.000000000 +0900
+++ gsl-2.1-android/cblas/Makefile.am	2015-12-24 22:10:30.143384843 +0900
@@ -1,5 +1,5 @@
 lib_LTLIBRARIES = libgslcblas.la
-libgslcblas_la_LDFLAGS = $(GSLCBLAS_LDFLAGS) -version-info $(GSL_LT_CBLAS_VERSION)
+libgslcblas_la_LDFLAGS = $(GSLCBLAS_LDFLAGS) -avoid-version
 
 pkginclude_HEADERS = gsl_cblas.h
 
diff -ruN gsl-2.1/cblas/Makefile.in gsl-2.1-android/cblas/Makefile.in
--- gsl-2.1/cblas/Makefile.in	2015-11-12 08:26:24.000000000 +0900
+++ gsl-2.1-android/cblas/Makefile.in	2015-12-24 22:10:37.342533156 +0900
@@ -557,7 +557,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 lib_LTLIBRARIES = libgslcblas.la
-libgslcblas_la_LDFLAGS = $(GSLCBLAS_LDFLAGS) -version-info $(GSL_LT_CBLAS_VERSION)
+libgslcblas_la_LDFLAGS = $(GSLCBLAS_LDFLAGS) -avoid-version
 pkginclude_HEADERS = gsl_cblas.h
 AM_CPPFLAGS = -I$(top_srcdir)
 libgslcblas_la_SOURCES = sasum.c saxpy.c scasum.c scnrm2.c scopy.c \
diff -ruN gsl-2.1/Makefile.am gsl-2.1-android/Makefile.am
--- gsl-2.1/Makefile.am	2015-11-09 15:08:09.000000000 +0900
+++ gsl-2.1-android/Makefile.am	2015-12-24 22:09:31.903185006 +0900
@@ -20,7 +20,7 @@
 lib_LTLIBRARIES = libgsl.la
 libgsl_la_SOURCES = version.c
 libgsl_la_LIBADD = $(GSL_LIBADD) $(SUBLIBS)
-libgsl_la_LDFLAGS = $(GSL_LDFLAGS) -version-info $(GSL_LT_VERSION)
+libgsl_la_LDFLAGS = $(GSL_LDFLAGS) -avoid-version
 noinst_HEADERS = templates_on.h templates_off.h build.h
 
 m4datadir = $(datadir)/aclocal
diff -ruN gsl-2.1/Makefile.in gsl-2.1-android/Makefile.in
--- gsl-2.1/Makefile.in	2015-11-12 08:26:24.000000000 +0900
+++ gsl-2.1-android/Makefile.in	2015-12-24 22:09:39.804347783 +0900
@@ -629,7 +629,7 @@
 lib_LTLIBRARIES = libgsl.la
 libgsl_la_SOURCES = version.c
 libgsl_la_LIBADD = $(GSL_LIBADD) $(SUBLIBS)
-libgsl_la_LDFLAGS = $(GSL_LDFLAGS) -version-info $(GSL_LT_VERSION)
+libgsl_la_LDFLAGS = $(GSL_LDFLAGS) -avoid-version
 noinst_HEADERS = templates_on.h templates_off.h build.h
 m4datadir = $(datadir)/aclocal
 m4data_DATA = gsl.m4
