--- ffmpeg-4.0/libavdevice/decklink_dec.cpp	2018-04-20 06:02:57.000000000 -0400
+++ ffmpeg-4.0/libavdevice/decklink_dec2.cpp	2018-05-06 11:23:41.688285100 -0400
@@ -707,6 +707,7 @@
                         "- Frames dropped %u\n", ctx->frameCount, ++ctx->dropped);
             }
             no_video = 1;
+            pkt.flags |= AV_PKT_FLAG_DISCARD;
         } else {
             if (no_video) {
                 av_log(avctx, AV_LOG_WARNING, "Frame received (#%lu) - Input returned "
