diff -ruN ffmpeg-3.3.2/configure ffmpeg-3.3.2-linux/configure
--- ffmpeg-3.3.2/configure	2017-06-07 11:04:02.000000000 +0900
+++ ffmpeg-3.3.2-linux/configure	2017-06-23 20:01:53.548498782 +0900
@@ -5804,7 +5804,7 @@
                                { use_pkg_config opencv opencv2/core/core_c.h cvCreateImageHeader ||
                                  require opencv opencv2/core/core_c.h cvCreateImageHeader -lopencv_core -lopencv_imgproc; } ||
                                require_pkg_config opencv opencv/cxcore.h cvCreateImageHeader; }
-enabled libopenh264       && require_pkg_config openh264 wels/codec_api.h WelsGetCodecVersion
+enabled libopenh264       && require openh264 wels/codec_api.h WelsGetCodecVersion -lopenh264
 enabled libopenjpeg       && { { check_lib openjpeg-2.1/openjpeg.h opj_version -lopenjp2 -DOPJ_STATIC && add_cppflags -DOPJ_STATIC; } ||
                                check_lib openjpeg-2.1/openjpeg.h opj_version -lopenjp2 ||
                                { check_lib openjpeg-2.0/openjpeg.h opj_version -lopenjp2 -DOPJ_STATIC && add_cppflags -DOPJ_STATIC; } ||
