diff --git a/libavutil/hwcontext_drm.c b/libavutil/hwcontext_drm.c
index 565c02d..07f3c1d 100644
--- a/libavutil/hwcontext_drm.c
+++ b/libavutil/hwcontext_drm.c
@@ -52,11 +52,32 @@ static int drm_device_create(AVHWDeviceContext *hwdev, const char *device,
                              AVDictionary *opts, int flags)
 {
     AVDRMDeviceContext *hwctx = hwdev->hwctx;
+    static const char *const default_devices[] = {
+        "/dev/dri/card0",
+        "/dev/dri/card1",
+        "/dev/dri/renderD128",
+        "/dev/dri/renderD129",
+    };
     drmVersionPtr version;
-
-    hwctx->fd = open(device, O_RDWR);
-    if (hwctx->fd < 0)
-        return AVERROR(errno);
+    int i;
+
+    hwctx->fd = -1;
+
+    if (device) {
+        hwctx->fd = open(device, O_RDWR);
+        if (hwctx->fd < 0)
+            return AVERROR(errno);
+    } else {
+        for (i = 0; i < FF_ARRAY_ELEMS(default_devices); i++) {
+            hwctx->fd = open(default_devices[i], O_RDWR);
+            if (hwctx->fd >= 0) {
+                device = default_devices[i];
+                break;
+            }
+        }
+        if (hwctx->fd < 0)
+            return AVERROR(errno);
+    }
 
     version = drmGetVersion(hwctx->fd);
     if (!version) {
