--- ffmpeg-3.4.2/configure	2018-02-11 19:29:18.000000000 -0500
+++ ffmpeg-3.4.2-windows/configure	2018-03-08 23:35:16.586891700 -0500
@@ -3087,9 +3087,9 @@
 caca_outdev_deps="libcaca"
 decklink_deps_any="libdl LoadLibrary"
 decklink_indev_deps="decklink threads"
-decklink_indev_extralibs="-lstdc++"
+decklink_indev_extralibs="-static-libstdc++"
 decklink_outdev_deps="decklink threads"
-decklink_outdev_extralibs="-lstdc++"
+decklink_outdev_extralibs="-static-libstdc++"
 libndi_newtek_indev_deps="libndi_newtek"
 libndi_newtek_indev_extralibs="-lndi"
 libndi_newtek_outdev_deps="libndi_newtek"
