diff -ruN lame-3.100/configure lame-3.100-patch/configure
--- lame-3.100/configure	2017-08-16 00:17:12.000000000 +0900
+++ lame-3.100-patch/configure	2017-10-22 19:16:03.168686600 +0900
@@ -15944,6 +15944,7 @@
 
 #include <stdlib.h>
 #include <iconv.h>
+#include <langinfo.h>
 
 int
 main ()
@@ -15968,6 +15969,7 @@
 
 #include <stdlib.h>
 #include <iconv.h>
+#include <langinfo.h>
 
 int
 main ()
@@ -16014,6 +16016,7 @@
 /* end confdefs.h.  */
 
 #include <iconv.h>
+#include <langinfo.h>
 #include <string.h>
 
 #ifndef ICONV_CONST
@@ -16172,6 +16175,7 @@
 
 #include <stdlib.h>
 #include <iconv.h>
+#include <langinfo.h>
 extern
 #ifdef __cplusplus
 "C"
