diff -ruN libvpx-1.8.0/vp8/common/rtcd_defs.pl libvpx-1.8.0-linux-arm/vp8/common/rtcd_defs.pl
--- libvpx-1.8.0/vp8/common/rtcd_defs.pl	2019-02-05 02:02:33.000000000 +0900
+++ libvpx-1.8.0-linux-arm/vp8/common/rtcd_defs.pl	2019-02-13 21:18:34.683076018 +0900
@@ -60,10 +60,9 @@
 
 
 add_proto qw/void vp8_loop_filter_simple_mbv/, "unsigned char *y_ptr, int y_stride, const unsigned char *blimit";
-specialize qw/vp8_loop_filter_simple_mbv sse2 neon msa mmi/;
+specialize qw/vp8_loop_filter_simple_mbv sse2 msa mmi/;
 $vp8_loop_filter_simple_mbv_c=vp8_loop_filter_simple_vertical_edge_c;
 $vp8_loop_filter_simple_mbv_sse2=vp8_loop_filter_simple_vertical_edge_sse2;
-$vp8_loop_filter_simple_mbv_neon=vp8_loop_filter_mbvs_neon;
 $vp8_loop_filter_simple_mbv_msa=vp8_loop_filter_simple_vertical_edge_msa;
 $vp8_loop_filter_simple_mbv_mmi=vp8_loop_filter_simple_vertical_edge_mmi;
 
@@ -76,10 +75,9 @@
 $vp8_loop_filter_simple_mbh_mmi=vp8_loop_filter_simple_horizontal_edge_mmi;
 
 add_proto qw/void vp8_loop_filter_simple_bv/, "unsigned char *y_ptr, int y_stride, const unsigned char *blimit";
-specialize qw/vp8_loop_filter_simple_bv sse2 neon msa mmi/;
+specialize qw/vp8_loop_filter_simple_bv sse2 msa mmi/;
 $vp8_loop_filter_simple_bv_c=vp8_loop_filter_bvs_c;
 $vp8_loop_filter_simple_bv_sse2=vp8_loop_filter_bvs_sse2;
-$vp8_loop_filter_simple_bv_neon=vp8_loop_filter_bvs_neon;
 $vp8_loop_filter_simple_bv_msa=vp8_loop_filter_bvs_msa;
 $vp8_loop_filter_simple_bv_mmi=vp8_loop_filter_bvs_mmi;
 
diff -ruN libvpx-1.8.0/vp8/vp8_common.mk libvpx-1.8.0-linux-arm/vp8/vp8_common.mk
--- libvpx-1.8.0/vp8/vp8_common.mk	2019-02-05 02:02:33.000000000 +0900
+++ libvpx-1.8.0-linux-arm/vp8/vp8_common.mk	2019-02-13 21:17:29.114192724 +0900
@@ -138,7 +138,7 @@
 VP8_COMMON_SRCS-$(HAVE_NEON)  += common/arm/neon/iwalsh_neon.c
 VP8_COMMON_SRCS-$(HAVE_NEON)  += common/arm/neon/vp8_loopfilter_neon.c
 VP8_COMMON_SRCS-$(HAVE_NEON)  += common/arm/neon/loopfiltersimplehorizontaledge_neon.c
-VP8_COMMON_SRCS-$(HAVE_NEON)  += common/arm/neon/loopfiltersimpleverticaledge_neon.c
+#VP8_COMMON_SRCS-$(HAVE_NEON)  += common/arm/neon/loopfiltersimpleverticaledge_neon.c
 VP8_COMMON_SRCS-$(HAVE_NEON)  += common/arm/neon/mbloopfilter_neon.c
 VP8_COMMON_SRCS-$(HAVE_NEON)  += common/arm/neon/shortidct4x4llm_neon.c
 VP8_COMMON_SRCS-$(HAVE_NEON)  += common/arm/neon/sixtappredict_neon.c
