diff -ruN libvpx-1.8.1/build/make/configure.sh libvpx-1.8.1-android/build/make/configure.sh
--- libvpx-1.8.1/build/make/configure.sh	2019-07-16 06:55:33.000000000 +0900
+++ libvpx-1.8.1-android/build/make/configure.sh	2019-08-10 13:40:49.859895140 +0900
@@ -1098,10 +1098,30 @@
           ;;
 
         android*)
-          echo "Assuming standalone build with NDK toolchain."
-          echo "See build/make/Android.mk for details."
-          check_add_ldflags -static
-          soft_enable unit_tests
+          if [ -n "${ANDROID_NDK}" ]; then
+            TOOLCHAIN_PATH=${ANDROID_PREFIX}-
+            CC=${ANDROID_CC}
+            CXX=${ANDROID_CC}++
+            AR=${TOOLCHAIN_PATH}ar
+            LD=${ANDROID_CC}
+            AS=${TOOLCHAIN_PATH}as
+            STRIP=${TOOLCHAIN_PATH}strip
+            NM=${TOOLCHAIN_PATH}nm
+            alt_libc=$ANDROID_ROOT
+
+            if [ -d "${alt_libc}" ]; then
+              add_cflags "--sysroot=${alt_libc}"
+              add_ldflags "--sysroot=${alt_libc}"
+            fi
+
+            enable_feature pic
+            soft_enable realtime_only
+          else
+            echo "Assuming standalone build with NDK toolchain."
+            echo "See build/make/Android.mk for details."
+            check_add_ldflags -static
+            soft_enable unit_tests
+          fi
           ;;
 
         darwin*)
@@ -1252,8 +1272,32 @@
       ;;
     x86*)
       case  ${tgt_os} in
-        android)
+        android*)
           soft_enable realtime_only
+          if [ -n "${ANDROID_NDK}" ]; then
+            TOOLCHAIN_PATH=${ANDROID_PREFIX}-
+            CC=${ANDROID_CC}
+            CXX=${ANDROID_CC}++
+            AR=${TOOLCHAIN_PATH}ar
+            LD=${ANDROID_CC}
+            AS=${TOOLCHAIN_PATH}as
+            STRIP=${TOOLCHAIN_PATH}strip
+            NM=${TOOLCHAIN_PATH}nm
+            alt_libc=$ANDROID_ROOT
+
+            if [ -d "${alt_libc}" ]; then
+              add_cflags "--sysroot=${alt_libc}"
+              add_ldflags "--sysroot=${alt_libc}"
+            fi
+
+            enable_feature pic
+            soft_enable realtime_only
+          else
+            echo "Assuming standalone build with NDK toolchain."
+            echo "See build/make/Android.mk for details."
+            check_add_ldflags -static
+            soft_enable unit_tests
+          fi
           ;;
         win*)
           enabled gcc && add_cflags -fno-common
