diff -ruN libvpx-1.6.1/build/make/configure.sh libvpx-1.6.1-android/build/make/configure.sh
--- libvpx-1.6.1/build/make/configure.sh	2017-01-13 05:27:27.000000000 +0900
+++ libvpx-1.6.1-android/build/make/configure.sh	2018-02-25 03:20:33.561735386 +0900
@@ -866,7 +866,7 @@
 
   # Process ARM architecture variants
   case ${toolchain} in
-    arm*)
+    *)
       # on arm, isa versions are supersets
       case ${tgt_isa} in
         arm64|armv8)
@@ -979,11 +979,8 @@
           ;;
 
         android*)
-          if [ -n "${sdk_path}" ]; then
-            SDK_PATH=${sdk_path}
-            COMPILER_LOCATION=`find "${SDK_PATH}" \
-              -name "arm-linux-androideabi-gcc*" -print -quit`
-            TOOLCHAIN_PATH=${COMPILER_LOCATION%/*}/arm-linux-androideabi-
+          if [ -n "${ANDROID_NDK}" ]; then
+            TOOLCHAIN_PATH=${ANDROID_BIN}-
             CC=${TOOLCHAIN_PATH}gcc
             CXX=${TOOLCHAIN_PATH}g++
             AR=${TOOLCHAIN_PATH}ar
@@ -991,33 +988,15 @@
             AS=${TOOLCHAIN_PATH}as
             STRIP=${TOOLCHAIN_PATH}strip
             NM=${TOOLCHAIN_PATH}nm
-
-            if [ -z "${alt_libc}" ]; then
-              alt_libc=`find "${SDK_PATH}" -name arch-arm -print | \
-                awk '{n = split($0,a,"/"); \
-                split(a[n-1],b,"-"); \
-                print $0 " " b[2]}' | \
-                sort -g -k 2 | \
-                awk '{ print $1 }' | tail -1`
-            fi
+            alt_libc=$ANDROID_ROOT
 
             if [ -d "${alt_libc}" ]; then
               add_cflags "--sysroot=${alt_libc}"
               add_ldflags "--sysroot=${alt_libc}"
             fi
 
-            # linker flag that routes around a CPU bug in some
-            # Cortex-A8 implementations (NDK Dev Guide)
-            add_ldflags "-Wl,--fix-cortex-a8"
-
             enable_feature pic
             soft_enable realtime_only
-            if [ ${tgt_isa} = "armv7" ]; then
-              soft_enable runtime_cpu_detect
-            fi
-            if enabled runtime_cpu_detect; then
-              add_cflags "-I${SDK_PATH}/sources/android/cpufeatures"
-            fi
           else
             echo "Assuming standalone build with NDK toolchain."
             echo "See build/make/Android.mk for details."
