--- x265-2.8/build/linux/multilib.sh	2018-05-21 04:33:10.000000000 -0400
+++ x265-2.8-new/build/linux/multilib.sh	2018-09-13 12:59:47.000000000 -0400
@@ -3,17 +3,17 @@
 mkdir -p 8bit 10bit 12bit
 
 cd 12bit
-cmake ../../../source -DHIGH_BIT_DEPTH=ON -DEXPORT_C_API=OFF -DENABLE_SHARED=OFF -DENABLE_CLI=OFF -DMAIN12=ON
+cmake ../../../source -DHIGH_BIT_DEPTH=ON -DEXPORT_C_API=OFF -DENABLE_SHARED=OFF -DENABLE_CLI=OFF -DMAIN12=ON $@
 make ${MAKEFLAGS}
 
 cd ../10bit
-cmake ../../../source -DHIGH_BIT_DEPTH=ON -DEXPORT_C_API=OFF -DENABLE_SHARED=OFF -DENABLE_CLI=OFF
+cmake ../../../source -DHIGH_BIT_DEPTH=ON -DEXPORT_C_API=OFF -DENABLE_SHARED=OFF -DENABLE_CLI=OFF $@
 make ${MAKEFLAGS}
 
 cd ../8bit
 ln -sf ../10bit/libx265.a libx265_main10.a
 ln -sf ../12bit/libx265.a libx265_main12.a
-cmake ../../../source -DEXTRA_LIB="x265_main10.a;x265_main12.a" -DEXTRA_LINK_FLAGS=-L. -DLINKED_10BIT=ON -DLINKED_12BIT=ON
+cmake ../../../source -DEXTRA_LIB="x265_main10.a;x265_main12.a" -DEXTRA_LINK_FLAGS=-L. -DLINKED_10BIT=ON -DLINKED_12BIT=ON -DCMAKE_INSTALL_PREFIX=../../../.. -DENABLE_SHARED:bool=off $@
 make ${MAKEFLAGS}
 
 # rename the 8bit library, then combine all three into libx265.a
@@ -36,6 +36,8 @@
 else
 
 # Mac/BSD libtool
-libtool -static -o libx265.a libx265_main.a libx265_main10.a libx265_main12.a 2>/dev/null
+/usr/bin/libtool -static -o libx265.a libx265_main.a libx265_main10.a libx265_main12.a 2>/dev/null
 
 fi
+
+make install
